Index: dev.saucy/src/chrome/browser/extensions/api/tld/tld_api.cc
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dev.saucy/src/chrome/browser/extensions/api/tld/tld_api.cc	2013-11-09 13:10:56.338455340 -0500
@@ -0,0 +1,91 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "chrome/browser/extensions/api/tld/tld_api.h"
+
+#include "base/time/time.h"
+#include "base/values.h"
+#include "chrome/browser/extensions/extension_service.h"
+#include "chrome/common/extensions/extension.h"
+#include "chrome/common/extensions/extension_constants.h"
+#include "url/gurl.h"
+#include "url/url_parse.h"
+#include "url/url_util.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+
+namespace extensions {
+
+namespace {
+const char kInvalidUriError[] = "Invalid or unsupported URI";
+
+bool IsValidUrl(const GURL& url) {
+  return url.is_valid() && url.IsStandard() &&
+     !url.SchemeIsFile() && !url.SchemeIsFileSystem();
+}
+}  // namespace
+
+bool TldGetBaseDomainFunction::RunImpl() {
+  EXTENSION_FUNCTION_VALIDATE(args_->GetSize() == 1);
+
+  std::string uri;
+  EXTENSION_FUNCTION_VALIDATE(args_->GetString(0, &uri));
+
+  // We might delegate all the url validity check to GetDomainAndRegistry
+  // but try to be more explicit about the errors in some cases.
+  const GURL url(uri);
+  if (!IsValidUrl(url)) {
+    error_ = kInvalidUriError;
+    return false;
+  }
+
+  std::string tld =
+      net::registry_controlled_domains::GetDomainAndRegistry(url, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
+
+  SetResult(Value::CreateStringValue(tld));
+
+  return true;
+}
+
+
+bool TldGetPublicSuffixFunction::RunImpl() {
+  EXTENSION_FUNCTION_VALIDATE(args_->GetSize() == 2);
+
+  std::string uri;
+  EXTENSION_FUNCTION_VALIDATE(!args_->GetString(0, &uri));
+
+  // We might delegate all the url validity check to GetDomainAndRegistry
+  // or GetRegistryLength but try to be more explicit about the errors in
+  // some cases.
+  const GURL url(uri);
+  if (!IsValidUrl(url)) {
+    error_ = kInvalidUriError;
+    return false;
+  }
+
+  std::string suffix;
+
+  size_t reg_len =
+      net::registry_controlled_domains::GetRegistryLength(url, net::registry_controlled_domains::INCLUDE_UNKNOWN_REGISTRIES, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
+
+  if (std::string::npos == reg_len || 0 == reg_len)
+    return false;
+
+  std::string tld =
+      net::registry_controlled_domains::GetDomainAndRegistry(url, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
+
+  if (!tld.empty () && tld.size() > reg_len)
+    suffix = tld.substr(tld.size() - reg_len - 1, reg_len);
+
+  // An empty suffix is a acceptable value since the URL might be
+  // "properly formed" but might fall into cases such as:
+  // * http://co.uk/file.html
+  // * http://<IP address>/file.html
+  // which are described by GetRegistryLength to return 0 (i.e. a valid value).
+  SetResult(Value::CreateStringValue(suffix));
+
+  return true;
+}
+
+}
+
Index: dev.saucy/src/chrome/browser/extensions/api/tld/tld_api.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dev.saucy/src/chrome/browser/extensions/api/tld/tld_api.h	2013-11-09 13:10:56.338455340 -0500
@@ -0,0 +1,32 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CHROME_BROWSER_EXTENSIONS_API_TLD_TLD_API_H_
+#define CHROME_BROWSER_EXTENSIONS_API_TLD_TLD_API_H_
+#pragma once
+
+#include "chrome/browser/extensions/extension_function.h"
+
+namespace extensions {
+
+class TldGetBaseDomainFunction : public SyncExtensionFunction {
+ public:
+  DECLARE_EXTENSION_FUNCTION("tld.getBaseDomain", TLD_GETBASEDOMAIN);
+ protected:
+  virtual ~TldGetBaseDomainFunction() {}
+  virtual bool RunImpl() OVERRIDE;
+};
+
+class TldGetPublicSuffixFunction : public SyncExtensionFunction {
+ public:
+  DECLARE_EXTENSION_FUNCTION("tld.getPublicSuffix", TLD_GETPUBLICSUFFIX);
+ protected:
+  virtual ~TldGetPublicSuffixFunction() {}
+  virtual bool RunImpl() OVERRIDE;
+};
+
+}
+
+#endif  // CHROME_BROWSER_EXTENSIONS_API_TLD_TLD_API_H_
+
Index: dev.saucy/src/chrome/common/extensions/api/api.gyp
===================================================================
--- dev.saucy.orig/src/chrome/common/extensions/api/api.gyp	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/common/extensions/api/api.gyp	2013-11-09 13:10:56.338455340 -0500
@@ -113,6 +113,7 @@
               'tabs.json',
               'terminal_private.json',
               'test.json',
+              'tld.json',
               'usb.idl',
               'virtual_keyboard_private.json',
               'web_navigation.json',
Index: dev.saucy/src/chrome/common/extensions/api/tld.json
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dev.saucy/src/chrome/common/extensions/api/tld.json	2013-11-09 13:10:56.338455340 -0500
@@ -0,0 +1,55 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+[
+  {
+    "namespace": "tld",
+    "nodoc": "true",
+    "description": "Use the <code>chrome.infobars</code> API to add a horizontal panel just above a tab's contents. See the screenshot below.",
+    "types": [],
+    "functions": [
+      {
+        "name": "getBaseDomain",
+        "type": "function",
+        "description": "Returns the base domain of a URI; that is, the public suffix with a given number of additional domain name parts.",
+        "nodoc": "true",
+        "parameters": [
+          {
+            "name": "uri",
+            "type": "string"
+          },
+          {
+            "type": "function",
+            "name": "callback",
+            "parameters": [
+              {"name": "domain", "type": "string", "description": "The base domain name of the associated URI if any"}
+            ]
+          }
+        ]
+      }
+      ,
+      {
+        "name": "getPublicSuffix",
+        "type": "function",
+        "description": "Returns the public suffix of the domain part of a URI.",
+        "nodoc": "true",
+        "parameters": [
+          {
+            "name": "uri",
+            "type": "string"
+          },
+          {
+            "type": "function",
+            "name": "callback",
+            "parameters": [
+              {"name": "suffix", "type": "string", "description": "The public suffix of the associated URI if any"}
+            ]
+          }
+        ]
+      }
+    ]
+  }
+
+]
+
Index: dev.saucy/src/chrome/common/extensions/api/_api_features.json
===================================================================
--- dev.saucy.orig/src/chrome/common/extensions/api/_api_features.json	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/common/extensions/api/_api_features.json	2013-11-09 13:10:56.342455485 -0500
@@ -620,6 +620,10 @@
     "extension_types": "all",
     "contexts": ["blessed_extension", "unblessed_extension", "content_script"]
   },
+  "tld": {
+    "dependencies": ["permission:tld"],
+    "contexts": ["blessed_extension"]
+  },
   "topSites": {
     "dependencies": ["permission:topSites"],
     "contexts": ["blessed_extension"]
Index: dev.saucy/src/chrome/common/extensions/api/_permission_features.json
===================================================================
--- dev.saucy.orig/src/chrome/common/extensions/api/_permission_features.json	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/common/extensions/api/_permission_features.json	2013-11-09 13:10:56.342455485 -0500
@@ -726,6 +726,10 @@
       "okddffdblfhhnmhodogpojmfkjmhinfp"   // HTerm dev
     ]
   },
+  "tld": {
+    "channel": "stable",
+    "extension_types": ["extension", "packaged_app", "platform_app"]
+  },
   "topSites": {
     "channel": "stable",
     "extension_types": ["extension", "legacy_packaged_app"]
Index: dev.saucy/src/chrome/browser/extensions/extension_function_histogram_value.h
===================================================================
--- dev.saucy.orig/src/chrome/browser/extensions/extension_function_histogram_value.h	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/browser/extensions/extension_function_histogram_value.h	2013-11-09 13:10:56.342455485 -0500
@@ -556,6 +556,8 @@
   DIAGNOSTICS_SENDPACKET,
   METRICSPRIVATE_GETFIELDTRIAL,
   FILEBROWSERPRIVATE_ZOOM,
+  TLD_GET_BASE_DOMAIN,
+  TLD_GET_PUBLIC_SUFFIX,
   WEBVIEW_GO,
   WEBSTOREPRIVATE_ISININCOGNITOMODEFUNCTION,
   FILEBROWSERPRIVATE_REQUESTACCESSTOKEN,
Index: dev.saucy/src/chrome/browser/extensions/extension_function_registry.cc
===================================================================
--- dev.saucy.orig/src/chrome/browser/extensions/extension_function_registry.cc	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/browser/extensions/extension_function_registry.cc	2013-11-09 13:10:56.354455919 -0500
@@ -8,6 +8,7 @@
 #include "chrome/browser/extensions/api/preference/chrome_direct_setting.h"
 #include "chrome/browser/extensions/api/preference/preference_api.h"
 #include "chrome/browser/extensions/api/runtime/runtime_api.h"
+#include "chrome/browser/extensions/api/tld/tld_api.h"
 #include "chrome/browser/extensions/api/web_request/web_request_api.h"
 #include "chrome/common/extensions/api/generated_api.h"
 
@@ -54,6 +55,10 @@
   RegisterFunction<extensions::ExperimentalIdentityGetAuthTokenFunction>();
   RegisterFunction<extensions::ExperimentalIdentityLaunchWebAuthFlowFunction>();
 
+  // TLD.
+  RegisterFunction<TldGetBaseDomainFunction>();
+  RegisterFunction<TldGetPublicSuffix>();
+
   // Generated APIs
   extensions::api::GeneratedFunctionRegistry::RegisterAll(this);
 #endif  // defined(ENABLE_EXTENSIONS)
Index: dev.saucy/src/chrome/chrome_browser_extensions.gypi
===================================================================
--- dev.saucy.orig/src/chrome/chrome_browser_extensions.gypi	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/chrome_browser_extensions.gypi	2013-11-09 13:10:56.354455919 -0500
@@ -578,6 +578,8 @@
         'browser/extensions/api/tabs/windows_util.h',
         'browser/extensions/api/test/test_api.cc',
         'browser/extensions/api/test/test_api.h',
+        'browser/extensions/api/tld/tld_api.cc',
+        'browser/extensions/api/tld/tld_api.h',
         'browser/extensions/api/top_sites/top_sites_api.cc',
         'browser/extensions/api/top_sites/top_sites_api.h',
         'browser/extensions/api/usb/usb_api.cc',
Index: dev.saucy/src/chrome/common/extensions/api/_manifest_features.json
===================================================================
--- dev.saucy.orig/src/chrome/common/extensions/api/_manifest_features.json	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/common/extensions/api/_manifest_features.json	2013-11-09 13:10:56.354455919 -0500
@@ -379,6 +379,10 @@
     "channel": "stable",
     "extension_types": ["theme"]
   },
+  "tld": {
+    "channel": "stable",
+    "extension_types": ["blessed_extension"]
+  },
   "tts_engine": {
     "channel": "stable",
     "extension_types": ["extension", "legacy_packaged_app"]
Index: dev.saucy/src/chrome/common/extensions/api/extension_api.cc
===================================================================
--- dev.saucy.orig/src/chrome/common/extensions/api/extension_api.cc	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/common/extensions/api/extension_api.cc	2013-11-09 13:10:56.354455919 -0500
@@ -255,6 +255,7 @@
       IDR_EXTENSION_API_JSON_STREAMSPRIVATE);
   RegisterSchemaResource("ttsEngine", IDR_EXTENSION_API_JSON_TTSENGINE);
   RegisterSchemaResource("tts", IDR_EXTENSION_API_JSON_TTS);
+  RegisterSchemaResource("tld", IDR_EXTENSION_API_JSON_TLD);
   RegisterSchemaResource("types", IDR_EXTENSION_API_JSON_TYPES);
   RegisterSchemaResource("types.private", IDR_EXTENSION_API_JSON_TYPES_PRIVATE);
   RegisterSchemaResource("webRequestInternal",
Index: dev.saucy/src/chrome/common/extensions_api_resources.grd
===================================================================
--- dev.saucy.orig/src/chrome/common/extensions_api_resources.grd	2013-11-09 13:10:56.370456498 -0500
+++ dev.saucy/src/chrome/common/extensions_api_resources.grd	2013-11-09 13:10:56.362456208 -0500
@@ -28,6 +28,7 @@
       <include name="IDR_EXTENSION_API_JSON_RUNTIME" file="extensions\api\runtime.json" type="BINDATA" />
       <include name="IDR_EXTENSION_API_JSON_SCRIPTBADGE" file="extensions\api\script_badge.json" type="BINDATA" />
       <include name="IDR_EXTENSION_API_JSON_TTS" file="extensions\api\tts.json" type="BINDATA" />
+      <include name="IDR_EXTENSION_API_JSON_TLD" file="extensions\api\tld.json" type="BINDATA" />
       <include name="IDR_EXTENSION_API_JSON_TTSENGINE" file="extensions\api\tts_engine.json" type="BINDATA" />
       <include name="IDR_EXTENSION_API_JSON_TYPES" file="extensions\api\types.json" type="BINDATA" />
       <include name="IDR_EXTENSION_API_JSON_TYPES_PRIVATE" file="extensions\api\types_private.json" type="BINDATA" />
