Index: dev.wily/chrome/browser/ui/views/frame/browser_frame.cc
===================================================================
--- dev.wily.orig/chrome/browser/ui/views/frame/browser_frame.cc
+++ dev.wily/chrome/browser/ui/views/frame/browser_frame.cc
@@ -27,6 +27,7 @@
 #include "ui/gfx/screen.h"
 #include "ui/views/controls/menu/menu_runner.h"
 #include "ui/views/widget/native_widget.h"
+#include "ui/views/widget/native_widget_aura.h"
 
 #if defined(OS_CHROMEOS)
 #include "ash/session/session_state_delegate.h"
@@ -126,6 +127,12 @@ views::View* BrowserFrame::GetFrameView(
   return browser_frame_view_;
 }
 
+void BrowserFrame::SendSessionIdToWindow() {
+  const Browser& browser = *browser_view_->browser();
+  views::NativeWidget* a = native_browser_frame_->GetWidgetParams().native_widget;
+  a->SetSessionID(browser.session_id().id());
+}
+
 bool BrowserFrame::UseCustomFrame() const {
   return native_browser_frame_->UseCustomFrame();
 }
Index: dev.wily/chrome/browser/ui/views/frame/browser_frame.h
===================================================================
--- dev.wily.orig/chrome/browser/ui/views/frame/browser_frame.h
+++ dev.wily/chrome/browser/ui/views/frame/browser_frame.h
@@ -94,6 +94,8 @@ class BrowserFrame
   void GetWindowPlacement(gfx::Rect* bounds,
                           ui::WindowShowState* show_state) const;
 
+  void SendSessionIdToWindow();
+
   // Overridden from views::Widget:
   views::internal::RootView* CreateRootView() override;
   views::NonClientFrameView* CreateNonClientFrameView() override;
Index: dev.wily/chrome/browser/ui/views/frame/browser_view.cc
===================================================================
--- dev.wily.orig/chrome/browser/ui/views/frame/browser_view.cc
+++ dev.wily/chrome/browser/ui/views/frame/browser_view.cc
@@ -703,6 +703,7 @@ void BrowserView::Show() {
   force_location_bar_focus_ = false;
 
   browser()->OnWindowDidShow();
+  frame_->SendSessionIdToWindow();
 
   chrome::MaybeShowInvertBubbleView(this);
 }
Index: dev.wily/ui/aura/client/aura_constants.cc
===================================================================
--- dev.wily.orig/ui/aura/client/aura_constants.cc
+++ dev.wily/ui/aura/client/aura_constants.cc
@@ -8,6 +8,7 @@
 #include "ui/gfx/geometry/rect.h"
 
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, bool)
+DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, uint32)
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, ui::ModalType)
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, gfx::Rect*)
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, ui::InputMethod*)
@@ -36,6 +37,7 @@ DEFINE_WINDOW_PROPERTY_KEY(
     ui::WindowShowState, kRestoreShowStateKey, ui::SHOW_STATE_DEFAULT);
 DEFINE_WINDOW_PROPERTY_KEY(
     ui::WindowShowState, kShowStateKey, ui::SHOW_STATE_DEFAULT);
+DEFINE_WINDOW_PROPERTY_KEY(uint32, kSessionIdKey, 0);
 
 }  // namespace client
 }  // namespace aura
Index: dev.wily/ui/aura/client/aura_constants.h
===================================================================
--- dev.wily.orig/ui/aura/client/aura_constants.h
+++ dev.wily/ui/aura/client/aura_constants.h
@@ -61,6 +61,9 @@ AURA_EXPORT extern const WindowProperty<
 AURA_EXPORT extern const WindowProperty<ui::WindowShowState>* const
     kShowStateKey;
 
+AURA_EXPORT extern const WindowProperty<uint32>* const kSessionIdKey;
+
+
 // Alphabetical sort.
 
 }  // namespace client
Index: dev.wily/ui/aura/window_tree_host_x11.cc
===================================================================
--- dev.wily.orig/ui/aura/window_tree_host_x11.cc
+++ dev.wily/ui/aura/window_tree_host_x11.cc
@@ -458,6 +458,26 @@ gfx::AcceleratedWidget WindowTreeHostX11
   return xwindow_;
 }
 
+#if 0
+void WindowTreeHostX11::SetSessionID(uint32 session_id) {
+  VLOG(1) << "BROKEN delete this session writer.";
+
+  if (session_id != 0) {
+    XChangeProperty(xdisplay_,
+                    xwindow_,
+                    atom_cache_.GetAtom("CHROMIUM_LINUX_SESSION_ID_PROPERTY"),
+                    XA_CARDINAL,
+                    32,
+                    PropModeReplace,
+                    reinterpret_cast<unsigned char*>(session_id),
+                    1);
+    VLOG(1) << " DID  set session_id as window property because we have a session " << session_id;
+  } else {
+    VLOG(1) << "can't set session_id as window property because we don't have a session yet.";
+  }
+}
+#endif
+
 void WindowTreeHostX11::ShowImpl() {
   if (!window_mapped_) {
     // Before we map the window, set size hints. Otherwise, some window managers
Index: dev.wily/ui/aura/window_tree_host_x11.h
===================================================================
--- dev.wily.orig/ui/aura/window_tree_host_x11.h
+++ dev.wily/ui/aura/window_tree_host_x11.h
@@ -39,6 +39,8 @@ class AURA_EXPORT WindowTreeHostX11 : pu
   bool CanDispatchEvent(const ui::PlatformEvent& event) override;
   uint32_t DispatchEvent(const ui::PlatformEvent& event) override;
 
+//  void SetSessionID(uint32 s) OVERRIDE;
+
   // WindowTreeHost:
   ui::EventSource* GetEventSource() override;
   gfx::AcceleratedWidget GetAcceleratedWidget() override;
Index: dev.wily/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
===================================================================
--- dev.wily.orig/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
+++ dev.wily/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
@@ -597,6 +597,11 @@ void DesktopNativeWidgetAura::ViewRemove
   drop_helper_->ResetTargetViewIfEquals(view);
 }
 
+void DesktopNativeWidgetAura::SetSessionID(unsigned int sessionid) {
+  desktop_window_tree_host_->SetSessionID(sessionid);
+  VLOG(1) << "session id Sent to treehost";
+}
+
 void DesktopNativeWidgetAura::SetNativeWindowProperty(const char* name,
                                                       void* value) {
   if (content_window_)
Index: dev.wily/ui/views/widget/desktop_aura/desktop_native_widget_aura.h
===================================================================
--- dev.wily.orig/ui/views/widget/desktop_aura/desktop_native_widget_aura.h
+++ dev.wily/ui/views/widget/desktop_aura/desktop_native_widget_aura.h
@@ -87,6 +87,8 @@ class VIEWS_EXPORT DesktopNativeWidgetAu
   // we are being activated/deactivated.
   void HandleActivationChanged(bool active);
 
+  void SetSessionID(uint32 sessionid);
+
  protected:
   // Overridden from internal::NativeWidgetPrivate:
   void InitNativeWidget(const Widget::InitParams& params) override;
Index: dev.wily/ui/views/widget/desktop_aura/desktop_window_tree_host.h
===================================================================
--- dev.wily.orig/ui/views/widget/desktop_aura/desktop_window_tree_host.h
+++ dev.wily/ui/views/widget/desktop_aura/desktop_window_tree_host.h
@@ -80,6 +80,8 @@ class VIEWS_EXPORT DesktopWindowTreeHost
 
   virtual bool IsVisible() const = 0;
 
+  virtual void SetSessionID(uint32 sessionid) = 0;
+
   virtual void SetSize(const gfx::Size& size) = 0;
   virtual void StackAbove(aura::Window* window) = 0;
   virtual void StackAtTop() = 0;
Index: dev.wily/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
===================================================================
--- dev.wily.orig/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
+++ dev.wily/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
@@ -10,6 +10,8 @@
 #include <X11/Xregion.h>
 #include <X11/Xutil.h>
 
+#include <signal.h>
+
 #include "base/basictypes.h"
 #include "base/command_line.h"
 #include "base/strings/stringprintf.h"
@@ -106,6 +108,7 @@ const char* kAtomsToCache[] = {
   "_NET_WM_WINDOW_TYPE_NORMAL",
   "_NET_WM_WINDOW_TYPE_NOTIFICATION",
   "_NET_WM_WINDOW_TYPE_TOOLTIP",
+  "CHROMIUM_LINUX_SESSION_ID_PROPERTY",
   "XdndActionAsk",
   "XdndActionCopy"
   "XdndActionLink",
@@ -1412,6 +1415,22 @@ void DesktopWindowTreeHostX11::UpdateMin
   XSetWMNormalHints(xdisplay_, xwindow_, &hints);
 }
 
+void DesktopWindowTreeHostX11::SetSessionID(uint32 session_id) {
+  if (session_id != 0) {
+    XChangeProperty(xdisplay_,
+                    xwindow_,
+                    atom_cache_.GetAtom("CHROMIUM_LINUX_SESSION_ID_PROPERTY"),
+                    XA_CARDINAL,
+                    32,
+                    PropModeReplace,
+                    reinterpret_cast<unsigned char*>(&session_id),
+                    1);
+    VLOG(1) << "Sent session id " << session_id << " to X window property";
+  } else {
+    VLOG(1) << "Session id is unlikely number.";
+  }
+}
+
 void DesktopWindowTreeHostX11::UpdateWMUserTime(
     const ui::PlatformEvent& event) {
   if (!IsActive())
Index: dev.wily/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
===================================================================
--- dev.wily.orig/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
+++ dev.wily/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
@@ -85,6 +85,8 @@ class VIEWS_EXPORT DesktopWindowTreeHost
   // internal list of open windows.
   static void CleanUpWindowList(void (*func)(aura::Window* window));
 
+  void SetSessionID(uint32 sessionid);
+
  protected:
   // Overridden from DesktopWindowTreeHost:
   void Init(aura::Window* content_window,
Index: dev.wily/ui/views/widget/native_widget.h
===================================================================
--- dev.wily.orig/ui/views/widget/native_widget.h
+++ dev.wily/ui/views/widget/native_widget.h
@@ -26,6 +26,8 @@ class VIEWS_EXPORT NativeWidget {
  public:
   virtual ~NativeWidget() {}
 
+  virtual void SetSessionID(uint32 sessionid) = 0;
+
  private:
   friend class Widget;
 
Index: dev.wily/ui/views/widget/native_widget_aura.cc
===================================================================
--- dev.wily.orig/ui/views/widget/native_widget_aura.cc
+++ dev.wily/ui/views/widget/native_widget_aura.cc
@@ -58,6 +58,8 @@
 #include "ui/views/widget/desktop_aura/desktop_window_tree_host.h"
 #endif
 
+#include <signal.h>
+
 namespace views {
 
 namespace {
@@ -322,6 +324,11 @@ void NativeWidgetAura::CenterWindow(cons
   window_->SetBounds(window_bounds);
 }
 
+void NativeWidgetAura::SetSessionID(uint32 sessionid) {
+  VLOG(1) << "I have no idea.";
+  raise(SIGABRT);
+}
+
 void NativeWidgetAura::GetWindowPlacement(
     gfx::Rect* bounds,
     ui::WindowShowState* show_state) const {
Index: dev.wily/ui/views/widget/native_widget_aura.h
===================================================================
--- dev.wily.orig/ui/views/widget/native_widget_aura.h
+++ dev.wily/ui/views/widget/native_widget_aura.h
@@ -48,6 +48,8 @@ class VIEWS_EXPORT NativeWidgetAura
       internal::NativeWidgetPrivate* native_widget,
       aura::Window* window);
 
+  virtual void SetSessionID(uint32 sessionid) override;
+
   // Overridden from internal::NativeWidgetPrivate:
   void InitNativeWidget(const Widget::InitParams& params) override;
   NonClientFrameView* CreateNonClientFrameView() override;
