Index: dev.xenial/chrome/app/generated_resources.grd
===================================================================
--- dev.xenial.orig/chrome/app/generated_resources.grd
+++ dev.xenial/chrome/app/generated_resources.grd
@@ -7739,6 +7739,35 @@ Keep your key file in a safe place. You
 I don't think this site should be blocked!
       </message>
 
+      <!-- Desktop integration -->
+      <message name="IDS_PROMPT_INTEGRATION_ENABLE_BUTTON_LABEL" desc="The text to be displayed on the checkbox that enables desktop integration.">
+        Notify me when the website I am visiting can run as an application
+      </message>
+      <message name="IDS_DESKTOP_INTEGRATION_NOTE_DESCRIPTION" desc="The text to be displayed below the Exceptions button in the Integrated Websites section.">
+        Websites which have been installed can be removed, like any other application, using Ubuntu Software Center.
+      </message>
+      <message name="IDS_DESKTOP_INTEGRATION_SETTINGS_EXCEPTIONS_BUTTON_LABEL" desc="The text to be displayed on the button that drives integration exceptions.">
+        Exceptions...
+      </message>
+      <message name="IDS_DESKTOP_INTEGRATION_SETTINGS_OVERLAY_TITLE" desc="Title for the desktop integration overlay.">
+        Integration prompt exceptions
+      </message>
+      <message name="IDS_DESKTOP_INTEGRATION_SETTINGS_OVERLAY_DESCRIPTION" desc="Description for the desktop integration overlay.">
+        These websites won't prompt to be installed:
+      </message>
+      <message name="IDS_OPTIONS_SETTINGS_INTEGRATION_WEBSITE_ADD_BUTTON"
+               desc="The label for the add button for adding a integration websites">
+        Add
+      </message>
+      <message name="IDS_OPTIONS_SETTINGS_INTEGRATION_WEBSITE_ADD_TEXT"
+               desc="The description text for the add overlay for adding a integration websites">
+        Website host-name
+      </message>
+      <message name="IDS_OPTIONS_SETTINGS_INTEGRATION_WEBSITE_ADD_TITLE"
+               desc="Title for add overlay for adding a integration websites">
+        Add Integrated Website
+      </message>
+
       <!-- Content Settings dialog data -->
       <message name="IDS_CONTENT_SETTINGS_TITLE" desc="Title for content settings dialog">
         Content settings
@@ -11103,6 +11132,9 @@ Some features may be unavailable.  Pleas
       <message name="IDS_OPTIONS_ADVANCED_SECTION_TITLE_CONTENT">
         Web content
       </message>
+      <message name="IDS_OPTIONS_ADVANCED_SECTION_TITLE_INTEGRATION">
+        Installed Websites
+      </message>
       <message name="IDS_OPTIONS_ADVANCED_SECTION_TITLE_CERTIFICATES">
         HTTPS/SSL
       </message>
Index: dev.xenial/chrome/browser/resources/options/browser_options.html
===================================================================
--- dev.xenial.orig/chrome/browser/resources/options/browser_options.html
+++ dev.xenial/chrome/browser/resources/options/browser_options.html
@@ -542,6 +542,20 @@
     </div>
   </section>
 </if>
+
+  <section id="desktop-integration-section">
+    <h3 i18n-content="advancedSectionTitleIntegration"></h3>
+    <div class="checkbox">
+      <label>
+        <input id="promptIntegrationForAnyWebsite"
+            metric="Options_PromptIntegration" type="checkbox">
+        <span i18n-content="promptIntegrationEnableIntegration"></span>
+      </label>
+    </div>
+    <button id="desktop-integration-button"
+        i18n-content="desktopIntegrationExceptionsSettingsButton"></button>
+  </section>
+
   <section id="passwords-and-autofill-section">
     <h3 i18n-content="passwordsAndAutofillGroupName"></h3>
     <div class="checkbox controlled-setting-with-label">
Index: dev.xenial/chrome/browser/resources/options/browser_options.js
===================================================================
--- dev.xenial.orig/chrome/browser/resources/options/browser_options.js
+++ dev.xenial/chrome/browser/resources/options/browser_options.js
@@ -674,6 +674,23 @@ cr.define('options', function() {
             [String(event.target.options[event.target.selectedIndex].value)]);
       };
 
+      // Desktop integration section.
+      if (cr.isLinux) {
+        var updateButtonState = function () {
+          $('desktop-integration-button').disabled =
+            ! $('promptIntegrationForAnyWebsite').checked;
+        };
+        $('promptIntegrationForAnyWebsite').onchange = function () {
+          updateButtonState();
+          chrome.send('setDesktopIntegrationAllowed',
+                      [$('promptIntegrationForAnyWebsite').checked]);
+        };
+        updateButtonState();
+        $('desktop-integration-button').onclick = function(event) {
+          PageManager.showPageByName('desktopIntegrationOverlay')
+        };
+      }
+
       // Languages section.
       var showLanguageOptions = function(event) {
         PageManager.showPageByName('languages');
@@ -1818,6 +1835,24 @@ cr.define('options', function() {
     },
 
     /**
+     * Disable the desktop integration settings if needed.
+     * @private
+     */
+    disableDesktopIntegration_: function() {
+      $('desktop-integration-section').style.display = 'none';
+    },
+
+    /**
+     * Disable the desktop integration settings if needed.
+     * @private
+     */
+    setDesktopIntegrationIsAllowed_: function(enabled) {
+      $('promptIntegrationForAnyWebsite').checked = enabled;
+      $('desktop-integration-button').disabled =
+        ! $('promptIntegrationForAnyWebsite').checked;
+    },
+
+    /**
      * Set the checked state of the metrics reporting checkbox.
      * @private
      */
@@ -2289,6 +2324,7 @@ cr.define('options', function() {
   // Forward public APIs to private implementations.
   cr.makePublic(BrowserOptions, [
     'deleteCurrentProfile',
+    'disableDesktopIntegration',
     'enableCertificateButton',
     'enableDisplaySettings',
     'enableFactoryResetSection',
@@ -2307,6 +2343,7 @@ cr.define('options', function() {
     'setNetworkPredictionValue',
     'setNowSectionVisible',
     'setHighContrastCheckboxState',
+    'setDesktopIntegrationIsAllowed',
     'setAllHotwordSectionsVisible',
     'setMetricsReportingCheckboxState',
     'setMetricsReportingSettingVisibility',
Index: dev.xenial/chrome/browser/resources/options/desktop_integration_add_website_overlay.html
===================================================================
--- /dev/null
+++ dev.xenial/chrome/browser/resources/options/desktop_integration_add_website_overlay.html
@@ -0,0 +1,15 @@
+<div id="add-desktop-integration-website-overlay-page" class="page" hidden>
+  <div class="close-button"></div>
+  <h1 i18n-content="add_desktop_website_title"></h1>
+
+  <div class="content-area">
+    <span i18n-content="add_desktop_website_input_label"></span>
+    <input id="integration-domain-url-field" type="url" data-type="url" placeholder="tube.example.com"
+           class="weakrtl hbox stretch">
+  </div>
+
+  <div class="action-area button-strip">
+    <button id="add-integration-website-overlay-cancel-button" i18n-content="cancel"></button>
+    <button id="add-integration-website-overlay-ok-button" i18n-content="ok"></button>
+  </div>
+</div>
Index: dev.xenial/chrome/browser/resources/options/desktop_integration_add_website_overlay.js
===================================================================
--- /dev/null
+++ dev.xenial/chrome/browser/resources/options/desktop_integration_add_website_overlay.js
@@ -0,0 +1,48 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+///////////////////////////////////////////////////////////////////////////////
+// AddDesktopIntegrationWebsiteOverlay class:
+
+cr.define('options', function() {
+  var Page = cr.ui.pageManager.Page;
+  var PageManager = cr.ui.pageManager.PageManager;
+
+  /**
+   * @constructor
+   */
+  function AddDesktopIntegrationWebsiteOverlay() {
+    Page.call(this, 'addDesktopIntegrationWebsite',
+              loadTimeData.getString('add_button'),
+              'add-desktop-integration-website-overlay-page');
+  }
+
+  cr.addSingletonGetter(AddDesktopIntegrationWebsiteOverlay);
+
+  AddDesktopIntegrationWebsiteOverlay.prototype = {
+    // Inherit AddDesktopIntegrationWebsiteOverlay from Page.
+    __proto__: Page.prototype,
+
+    /**
+     * Initializes AddDesktopIntegrationWebsiteOverlay page.
+     * Calls base class implementation to starts preference initialization.
+     */
+    initializePage: function() {
+      // Call base class implementation to starts preference initialization.
+      Page.prototype.initializePage.call(this);
+      
+      // Cleanup any previously entered text.
+      $('integration-domain-url-field').value = '';
+
+      // Set up the cancel button.
+      $('add-integration-website-overlay-cancel-button').onclick = function(e) {
+        PageManager.closeOverlay();
+      };
+    },
+  };
+
+  return {
+    AddDesktopIntegrationWebsiteOverlay: AddDesktopIntegrationWebsiteOverlay
+  };
+});
Index: dev.xenial/chrome/browser/resources/options/desktop_integration_overlay.css
===================================================================
--- /dev/null
+++ dev.xenial/chrome/browser/resources/options/desktop_integration_overlay.css
@@ -0,0 +1,75 @@
+/* Copyright (c) 2012 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file. */
+
+#desktop-integration-area {
+  min-width: 550px;
+}
+
+#desktop-integration-area list {
+  min-height: 172px;
+}
+
+#desktop-integration-area span {
+  -webkit-padding-start: 18px;
+}
+
+.desktop-integration-list-item {
+  -webkit-box-flex: 1;
+  -webkit-padding-start: 8px;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.desktop-integration-lower-left button {
+  min-width: 70px;
+}
+
+.desktop-integration-lower-left {
+  -webkit-box-flex: 0;
+  -webkit-padding-start: 12px;
+  padding-bottom: 10px;
+}
+
+#desktop-integration-column-headers {
+  -webkit-margin-start: 17px;
+  display: -webkit-box;
+  margin-top: 17px;
+}
+
+#desktop-integration-column-headers > div {
+  font-weight: bold;
+}
+
+#desktop-integration-pattern-column {
+  -webkit-box-flex: 1;
+}
+
+#desktop-integration-behavior-column {
+  width: 145px;
+}
+
+#desktop-integration-area list {
+  margin-bottom: 10px;
+  margin-top: 4px;
+}
+
+#domains-list {
+  -webkit-box-flex: 1;
+  outline: none;
+  padding: 1px 0 0;
+  width: 100%;
+}
+
+#domains-list .domain-name {
+  -webkit-box-flex: 1;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+#domains-list li {
+  -webkit-padding-start: 12px;
+  padding-bottom: 2px;
+  padding-top: 2px;
+}
Index: dev.xenial/chrome/browser/resources/options/desktop_integration_overlay.html
===================================================================
--- /dev/null
+++ dev.xenial/chrome/browser/resources/options/desktop_integration_overlay.html
@@ -0,0 +1,26 @@
+<div id="desktop-integration-area" class="page" hidden>
+
+  <div class="close-button"></div>
+  <h1 i18n-content="desktopIntegrationPage"></h1>
+  
+  <span i18n-content="desktopIntegrationInfoText"></span>
+ 
+  <div class="content-area">
+    <list id="domains-list"></list>
+
+    <div class="desktop-integration-lower-left">
+      <button id="desktop-integration-add-button"
+              i18n-content="add_button"></button>
+    </div>
+  </div>
+
+  <div class="action-area">
+    <div class="button-strip">
+      <button id="desktop-integrations-overlay-confirm"
+          i18n-content="ok">
+      </button>
+    </div>
+  </div>
+
+</div>
+
Index: dev.xenial/chrome/browser/resources/options/desktop_integration_overlay.js
===================================================================
--- /dev/null
+++ dev.xenial/chrome/browser/resources/options/desktop_integration_overlay.js
@@ -0,0 +1,207 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+cr.define('options', function() {
+  /** @const */ var ArrayDataModel = cr.ui.ArrayDataModel;
+  /** @const */ var DeletableItemList = options.DeletableItemList;
+  /** @const */ var DeletableItem = options.DeletableItem;
+
+  /**
+   * Creates a new exceptions list item.
+   * @constructor
+   * @extends {options.DeletableItem}
+   */
+  function IntegratedWebsitesListItem(domain) {
+    var el = cr.doc.createElement('div');
+    el.__proto__ = IntegratedWebsitesListItem.prototype;
+    el.domain = domain;
+    el.decorate();
+    return el;
+  }
+
+  cr.addSingletonGetter(IntegratedWebsitesListItem);
+
+  IntegratedWebsitesListItem.prototype = {
+    __proto__: DeletableItem.prototype,
+
+    /**
+     * Called when an element is decorated as a list item.
+     */
+    decorate: function() {
+      DeletableItem.prototype.decorate.call(this);
+
+      // The stored label.
+      var label = this.ownerDocument.createElement('div');
+      label.className = 'domain-name';
+      label.textContent = this.domain;
+      this.contentElement.appendChild(label);
+
+      this.deletable = true;
+    },
+  };
+
+  /**
+   * Creates a integrated websites list.
+   * @constructor
+   * @extends {cr.ui.List}
+   */
+  var IntegratedWebsitesList = cr.ui.define('list');
+
+  cr.addSingletonGetter(IntegratedWebsitesList);
+
+  IntegratedWebsitesList.prototype = {
+    __proto__: DeletableItemList.prototype,
+
+    /**
+     * Called when an element is decorated as a list.
+     */
+    decorate: function() {
+      DeletableItemList.prototype.decorate.call(this);
+      this.reset();
+    },
+
+    /** @inheritDoc */
+    createItem: function(domain) {
+      return new IntegratedWebsitesListItem(domain);
+    },
+
+    /*
+     * Adds a website domain name to the list of allowed integrated websites.
+     * @param {string} domain domain name of the website to add.
+     */
+    addWebsite: function(domain) {
+      if (!domain || this.dataModel.indexOf(domain) >= 0) {
+        return;
+      }
+      this.dataModel.push(domain);
+      this.redraw();
+      chrome.send('addIntegrationSite', [domain]);
+    },
+
+    /**
+     * Forces a revailidation of the list content. Content added while the list
+     * is hidden is not properly rendered when the list becomes visible. In
+     * addition, deleting a single item from the list results in a stale cache
+     * requiring an invalidation.
+     */
+    refresh: function() {
+      // TODO(kevers): Investigate if the root source of the problems can be
+      // fixed in cr.ui.list.
+      this.invalidate();
+      this.redraw();
+    },
+
+    /**
+     * Sets the integrated websites in the js model.
+     * @param {Object} entries A list of dictionaries of values, each dictionary
+     *     represents an exception.
+     */
+    setIntegratedWebsites: function(entries) {
+      var integratedWebsites = null;
+      try {
+        integratedWebsites = JSON.parse(entries);
+      } catch(e) {
+        console.log("Error while parsing integrated websites json: " + entries);
+        return;
+      }
+      // TODO hightlight domains differently based on permission
+      var domains = [];
+      domains = domains.concat(integratedWebsites['allowed']);
+      domains = domains.concat(integratedWebsites['dontask']);
+      this.dataModel = new ArrayDataModel(domains);
+      this.refresh();
+    },
+
+    /**
+     * Removes all integration scripts from the js model.
+     */
+    reset: function() {
+      this.dataModel = new ArrayDataModel([]);
+    },
+
+    /** @inheritDoc */
+    deleteItemAtIndex: function(index) {
+      if (index >= 0) {
+        var args = [this.dataModel.item(index)];
+        chrome.send('removeIntegrationSite', args);
+        this.dataModel.splice(index, 1);
+      }
+    },
+
+    /**
+     * The length of the list.
+     */
+    get length() {
+      return null != this.dataModel ? this.dataModel.length : 0;
+    },
+  };
+
+  var Page = cr.ui.pageManager.Page;
+  var PageManager = cr.ui.pageManager.PageManager;
+
+  /**
+   * DesktopIntegrationOverlay class
+   * Encapsulated handling of the 'Desktop integration' page.
+   * @class
+   */
+  function DesktopIntegrationOverlay() {
+    Page.call(this, 'desktopIntegrationOverlay',
+                     loadTimeData.getString('desktopIntegrationPage'),
+                     'desktop-integration-area');
+  }
+
+  cr.addSingletonGetter(DesktopIntegrationOverlay);
+
+  DesktopIntegrationOverlay.prototype = {
+    __proto__: Page.prototype,
+
+    /**
+     * Initialize the page.
+     */
+    initializePage: function() {
+      Page.prototype.initializePage.call(this);
+
+      var integrationList = $('domains-list');
+      IntegratedWebsitesList.decorate(integrationList);
+      $('desktop-integrations-overlay-confirm').onclick = function (ev) { PageManager.closeOverlay(); }
+
+      // Set up add button.
+      $('desktop-integration-add-button').onclick = function(e) {
+        PageManager.showPageByName('addDesktopIntegrationWebsite');
+      };
+
+      // Listen to add website dialog ok button.
+      var addWebsiteOkButton = $('add-integration-website-overlay-ok-button');
+      addWebsiteOkButton.addEventListener('click',
+          this.handleAddWebsiteOkButtonClick_.bind(this));
+    },
+
+    handleAddWebsiteOkButtonClick_: function () {
+      var website = $('integration-domain-url-field').value;
+      var integratedDomain = website.replace(/\s*/g, '')
+      console.log (integratedDomain);
+      $('domains-list').addWebsite(integratedDomain);
+      PageManager.closeOverlay();
+    },
+
+    /**
+     * Called by the options page when this page has been shown.
+     */
+    didShowPage: function() {
+      chrome.send('updateIntegratedWebsitesList');
+    },
+  };
+
+  DesktopIntegrationOverlay.setIntegratedWebsites = function(entries) {
+    $('domains-list').setIntegratedWebsites(entries);
+  };
+
+  // Export
+  return {
+    IntegratedWebsitesListItem: IntegratedWebsitesListItem,
+    IntegratedWebsitesList: IntegratedWebsitesList,
+    DesktopIntegrationOverlay: DesktopIntegrationOverlay
+  };
+});
+
Index: dev.xenial/chrome/browser/resources/options/options.html
===================================================================
--- dev.xenial.orig/chrome/browser/resources/options/options.html
+++ dev.xenial/chrome/browser/resources/options/options.html
@@ -25,6 +25,7 @@
 <link rel="stylesheet" href="clear_browser_data_overlay.css">
 <link rel="stylesheet" href="content_settings.css">
 <link rel="stylesheet" href="cookies_view.css">
+<link rel="stylesheet" href="desktop_integration_overlay.css">
 <link rel="stylesheet" href="do_not_track_confirm_overlay.css">
 <link rel="stylesheet" href="easy_unlock_turn_off_overlay.css">
 <link rel="stylesheet" href="font_settings.css">
@@ -118,6 +119,7 @@
   <include src="home_page_overlay.html">
   <include src="hotword_confirm_overlay.html">
   <include src="import_data_overlay.html">
+  <include src="desktop_integration_overlay.html">
   <include src="language_options.html">
   <include src="manage_profile_overlay.html">
   <include src="password_manager.html">
@@ -151,6 +153,7 @@
   <include src="clear_browser_data_history_notice_overlay.html">
   <include src="content_settings_exceptions_area.html">
   <include src="cookies_view.html">
+  <include src="desktop_integration_add_website_overlay.html">
   <include src="handler_options.html">
   <include src="language_add_language_overlay.html">
   <include src="supervised_user_import.html">
Index: dev.xenial/chrome/browser/resources/options/options.js
===================================================================
--- dev.xenial.orig/chrome/browser/resources/options/options.js
+++ dev.xenial/chrome/browser/resources/options/options.js
@@ -16,6 +16,8 @@ var ContentSettingsExceptionsArea =
     options.contentSettings.ContentSettingsExceptionsArea;
 var ContentSettings = options.ContentSettings;
 var CookiesView = options.CookiesView;
+var DesktopIntegrationOverlay = options.DesktopIntegrationOverlay;
+var AddDesktopIntegrationWebsiteOverlay = options.AddDesktopIntegrationWebsiteOverlay;
 var CreateProfileOverlay = options.CreateProfileOverlay;
 var EditDictionaryOverlay = cr.IsMac ? null : options.EditDictionaryOverlay;
 var EasyUnlockTurnOffOverlay = options.EasyUnlockTurnOffOverlay;
@@ -134,6 +136,17 @@ function load() {
   PageManager.registerOverlay(FontSettings.getInstance(),
                               BrowserOptions.getInstance(),
                               [$('fontSettingsCustomizeFontsButton')]);
+  if (cr.isLinux) {
+   try {
+    PageManager.registerOverlay(AddDesktopIntegrationWebsiteOverlay.getInstance(),
+                                DesktopIntegrationOverlay.getInstance());
+    PageManager.registerOverlay(DesktopIntegrationOverlay.getInstance(),
+                                BrowserOptions.getInstance(),
+                                [$('desktop-integration-button')]);
+   } catch (exc) {
+     console.log(exc)
+   }
+  }
   if (HandlerOptions && $('manage-handlers-button')) {
     PageManager.registerOverlay(HandlerOptions.getInstance(),
                                 ContentSettings.getInstance(),
Index: dev.xenial/chrome/browser/resources/options/options_bundle.js
===================================================================
--- dev.xenial.orig/chrome/browser/resources/options/options_bundle.js
+++ dev.xenial/chrome/browser/resources/options/options_bundle.js
@@ -84,6 +84,8 @@ var CertificateImportErrorOverlay = opti
 <include src="content_settings_ui.js">
 <include src="cookies_list.js">
 <include src="cookies_view.js">
+<include src="desktop_integration_add_website_overlay.js">
+<include src="desktop_integration_overlay.js">
 <include src="easy_unlock_turn_off_overlay.js">
 <include src="factory_reset_overlay.js">
 <include src="font_settings.js">
Index: dev.xenial/chrome/browser/ui/BUILD.gn
===================================================================
--- dev.xenial.orig/chrome/browser/ui/BUILD.gn
+++ dev.xenial/chrome/browser/ui/BUILD.gn
@@ -314,6 +314,9 @@ source_set("ui") {
     }
   }
   if (use_aura && !use_ozone && is_desktop_linux) {
+    configs += [
+      "//build/config/linux:glib",
+    ]
     deps += [
       "//build/linux:gio",
 
Index: dev.xenial/chrome/browser/ui/webui/options/browser_options_handler.cc
===================================================================
--- dev.xenial.orig/chrome/browser/ui/webui/options/browser_options_handler.cc
+++ dev.xenial/chrome/browser/ui/webui/options/browser_options_handler.cc
@@ -240,6 +240,8 @@ void BrowserOptionsHandler::GetLocalized
     { "advancedSectionTitleCloudPrint", IDS_GOOGLE_CLOUD_PRINT },
     { "advancedSectionTitleContent",
       IDS_OPTIONS_ADVANCED_SECTION_TITLE_CONTENT },
+    { "advancedSectionTitleIntegration",
+      IDS_OPTIONS_ADVANCED_SECTION_TITLE_INTEGRATION },
     { "advancedSectionTitleLanguages",
       IDS_OPTIONS_ADVANCED_SECTION_TITLE_LANGUAGES },
     { "advancedSectionTitleNetwork",
@@ -263,6 +265,10 @@ void BrowserOptionsHandler::GetLocalized
     { "defaultFontSizeLabel", IDS_OPTIONS_DEFAULT_FONT_SIZE_LABEL },
     { "defaultSearchManageEngines", IDS_OPTIONS_DEFAULTSEARCH_MANAGE_ENGINES },
     { "defaultZoomFactorLabel", IDS_OPTIONS_DEFAULT_ZOOM_LEVEL_LABEL },
+    { "desktopIntegrationExceptionsSettingsButton",
+      IDS_DESKTOP_INTEGRATION_SETTINGS_EXCEPTIONS_BUTTON_LABEL },
+    { "desktopIntegrationNoteSettingsContent",
+      IDS_DESKTOP_INTEGRATION_NOTE_DESCRIPTION },
     { "disableWebServices", IDS_OPTIONS_DISABLE_WEB_SERVICES },
     { "doNotTrack", IDS_OPTIONS_ENABLE_DO_NOT_TRACK },
     { "doNotTrackConfirmDisable",
@@ -349,6 +355,8 @@ void BrowserOptionsHandler::GetLocalized
     { "profilesManage", IDS_PROFILES_MANAGE_BUTTON_LABEL },
     { "profilesSingleUser", IDS_PROFILES_SINGLE_USER_MESSAGE,
       IDS_PRODUCT_NAME },
+    { "promptIntegrationEnableIntegration",
+      IDS_PROMPT_INTEGRATION_ENABLE_BUTTON_LABEL },
     { "proxiesLabelExtension", IDS_OPTIONS_EXTENSION_PROXIES_LABEL },
     { "proxiesLabelSystem", IDS_OPTIONS_SYSTEM_PROXIES_LABEL,
       IDS_PRODUCT_NAME },
Index: dev.xenial/chrome/browser/ui/webui/options/desktop_integration_settings_handler.cc
===================================================================
--- /dev/null
+++ dev.xenial/chrome/browser/ui/webui/options/desktop_integration_settings_handler.cc
@@ -0,0 +1,262 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "chrome/browser/ui/webui/options/desktop_integration_settings_handler.h"
+
+#include <glib.h>
+#include <dlfcn.h>
+#include <vector>
+
+#include "base/bind.h"
+#include "base/bind_helpers.h"
+#include "base/logging.h"
+#include "base/strings/string16.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/values.h"
+#include "chrome/common/url_constants.h"
+#include "content/public/browser/web_ui.h"
+#include "grit/generated_resources.h"
+#include "grit/content_resources.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/webui/web_ui_util.h"
+
+// Not a constant but preprocessor definition for easy concatenation.
+#define kLibunityWebappsEntryPointLibName "libunity-webapps.so.0"
+
+namespace options {
+
+// TODO: -> extract to function
+DesktopIntegrationSettingsHandler::DesktopIntegrationSettingsHandler()
+    : lib_unity_webapps_handle_(NULL),
+      get_all_func_handle_(NULL),
+      remove_from_permissions_func_handle_(NULL),
+      add_allowed_domain_func_handle_(NULL),
+      is_integration_allowed_func_handle_(NULL),
+      set_integration_allowed_func_handle_(NULL) {
+}
+
+DesktopIntegrationSettingsHandler::~DesktopIntegrationSettingsHandler() {
+  if (lib_unity_webapps_handle_) {
+    dlclose (lib_unity_webapps_handle_);
+    lib_unity_webapps_handle_ = NULL;
+
+    remove_from_permissions_func_handle_ = NULL;
+    get_all_func_handle_ = NULL;
+    add_allowed_domain_func_handle_ = NULL;
+    is_integration_allowed_func_handle_ = NULL;
+    set_integration_allowed_func_handle_ = NULL;
+  }
+}
+
+/////////////////////////////////////////////////////////////////////////////
+// OptionsPageUIHandler implementation:
+void DesktopIntegrationSettingsHandler::GetLocalizedValues(
+    base::DictionaryValue* localized_strings) {
+  DCHECK(localized_strings);
+
+  static OptionsStringResource resources[] = {
+    { "desktopIntegrationInfoText",
+      IDS_DESKTOP_INTEGRATION_SETTINGS_OVERLAY_DESCRIPTION }
+  };
+
+  localized_strings->SetString("add_button",
+      l10n_util::GetStringUTF16(IDS_OPTIONS_SETTINGS_INTEGRATION_WEBSITE_ADD_BUTTON));
+  localized_strings->SetString("add_desktop_website_input_label",
+      l10n_util::GetStringUTF16(IDS_OPTIONS_SETTINGS_INTEGRATION_WEBSITE_ADD_TEXT));
+  localized_strings->SetString("add_desktop_website_title",
+      l10n_util::GetStringUTF16(IDS_OPTIONS_SETTINGS_INTEGRATION_WEBSITE_ADD_TITLE));
+
+  RegisterStrings(localized_strings, resources, arraysize(resources));
+  RegisterTitle(localized_strings, "desktopIntegrationPage",
+                IDS_DESKTOP_INTEGRATION_SETTINGS_OVERLAY_TITLE);
+}
+
+void DesktopIntegrationSettingsHandler::InitializeHandler() {
+  if (!IsUnityWebappsInitialized()) {
+    LoadUnityWebappsEntryPoint();
+  }
+}
+
+void DesktopIntegrationSettingsHandler::InitializePage() {
+  DCHECK(IsUnityWebappsInitialized());
+  LoadIntegratedWebsitesData();
+}
+
+void DesktopIntegrationSettingsHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      std::string("addIntegrationSite"),
+      base::Bind(&DesktopIntegrationSettingsHandler::AddIntegrationSite,
+                 base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      std::string("setDesktopIntegrationAllowed"),
+      base::Bind(&DesktopIntegrationSettingsHandler::SetDesktopIntegrationAllowed,
+                 base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      std::string("removeIntegrationSite"),
+      base::Bind(&DesktopIntegrationSettingsHandler::RemoveIntegrationSite,
+                 base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      std::string("updateIntegratedWebsitesList"),
+      base::Bind(&DesktopIntegrationSettingsHandler::UpdateIntegratedWebsitesList,
+                 base::Unretained(this)));
+}
+
+void DesktopIntegrationSettingsHandler::LoadUnityWebappsEntryPoint() {
+  DCHECK(!IsUnityWebappsInitialized());
+
+  // TODO run on FILE thread
+  static const char* const search_paths[] = {
+    kLibunityWebappsEntryPointLibName,
+    "/usr/local/lib/" kLibunityWebappsEntryPointLibName
+    , "/usr/lib/" kLibunityWebappsEntryPointLibName
+  };
+
+  void * handle = NULL;
+  for (size_t i = 0; i < sizeof(search_paths)/sizeof(search_paths[0]); ++i) {
+    DCHECK(handle == NULL);
+    // TODO validate path?
+    handle = dlopen (search_paths[i], RTLD_LAZY|RTLD_GLOBAL);
+    if (handle) {
+      LOG(INFO) << "Found libunitywebapps entry point:" << search_paths[i];
+      break;
+    }
+  }
+  if (!handle) {
+    LOG(INFO) << "Could not load Unity Webapps entry point library";
+    return;
+  }
+
+  void * get_all_handle =
+    dlsym (handle, "unity_webapps_permissions_get_all_domains");
+  void * remove_from_permissions_handle =
+    dlsym (handle, "unity_webapps_permissions_remove_domain_from_permissions");
+  void * add_allowed_domain_handle =
+    dlsym (handle, "unity_webapps_permissions_allow_domain");
+  void * is_integration_allowed_handle =
+    dlsym (handle, "unity_webapps_permissions_is_integration_allowed");
+  void * set_integration_allowed_handle =
+    dlsym (handle, "unity_webapps_permissions_set_integration_allowed");
+
+  if (!get_all_handle ||
+      !remove_from_permissions_handle ||
+      !add_allowed_domain_handle ||
+      !is_integration_allowed_handle ||
+      !set_integration_allowed_handle) {
+    LOG(WARNING) << "Could not load Unity Webapps entry point functions";
+    dlclose(handle);
+    return;
+  }
+
+  // TODO(FIXME): cleanup that mess
+  lib_unity_webapps_handle_ = handle;
+  get_all_func_handle_ = get_all_handle;
+  remove_from_permissions_func_handle_ = remove_from_permissions_handle;
+  add_allowed_domain_func_handle_ = add_allowed_domain_handle;
+  is_integration_allowed_func_handle_ = is_integration_allowed_handle;
+  set_integration_allowed_func_handle_ = set_integration_allowed_handle;
+}
+
+bool DesktopIntegrationSettingsHandler::IsUnityWebappsInitialized() const {
+  return NULL != lib_unity_webapps_handle_
+    && NULL != get_all_func_handle_
+    && NULL != remove_from_permissions_func_handle_
+    && NULL != add_allowed_domain_func_handle_
+    && NULL != is_integration_allowed_func_handle_
+    && NULL != set_integration_allowed_func_handle_;
+}
+
+void DesktopIntegrationSettingsHandler::LoadIntegratedWebsitesData() {
+  if (!IsUnityWebappsInitialized()) {
+    web_ui()->CallJavascriptFunction(
+        "BrowserOptions.disableDesktopIntegration");
+    LOG(INFO) << "Disabling Desktop Integration options";
+    return;
+  }
+
+  typedef gboolean (* IsIntegrationAllowedFunc) (void);
+  gboolean isallowed =
+    ((IsIntegrationAllowedFunc) is_integration_allowed_func_handle_) ();
+  web_ui()->CallJavascriptFunction(
+       "BrowserOptions.setDesktopIntegrationIsAllowed",
+       base::FundamentalValue(isallowed));
+}
+
+void DesktopIntegrationSettingsHandler::UpdateIntegratedWebsitesList(
+    const base::ListValue* args) {
+  DCHECK(IsUnityWebappsInitialized());
+
+  // TODO move elsewhere
+  typedef gchar* (* GetAllDomainsFunc) (void);
+
+  gchar* all_domains = ((GetAllDomainsFunc) get_all_func_handle_) ();
+  if (all_domains) { 
+    web_ui()->CallJavascriptFunction(
+        "DesktopIntegrationOverlay.setIntegratedWebsites",
+        base::StringValue(all_domains));
+  }
+}
+
+void DesktopIntegrationSettingsHandler::SetDesktopIntegrationAllowed(
+    const base::ListValue* args) {
+  DCHECK(IsUnityWebappsInitialized());
+
+  // TODO move elsewhere
+  typedef void (* SetDesktopIntegrationAllowedFunc) (gboolean);
+
+  bool is_allowed;
+  if (!args->GetBoolean(0, &is_allowed)) {
+    NOTREACHED();
+    return;
+  }
+
+  LOG(INFO) << "Setting desktop integration:" << is_allowed;
+
+  ((SetDesktopIntegrationAllowedFunc) set_integration_allowed_func_handle_) (
+      is_allowed ? TRUE : FALSE);
+}
+
+void DesktopIntegrationSettingsHandler::AddIntegrationSite(
+    const base::ListValue* args) {
+  DCHECK(IsUnityWebappsInitialized());
+
+  // TODO move elsewhere
+  typedef void (* AddDomainFromPermissionsFunc) (gchar *);
+
+  std::string domain;
+  if (!args->GetString(0, &domain)) {
+    NOTREACHED();
+    return;
+  }
+
+  LOG(INFO) << "Adding domain:" << domain << " to the list of websites allowed";
+
+  ((AddDomainFromPermissionsFunc) add_allowed_domain_func_handle_) (
+      const_cast<gchar *>(domain.c_str()));
+
+  LoadIntegratedWebsitesData ();
+}
+
+void DesktopIntegrationSettingsHandler::RemoveIntegrationSite(
+    const base::ListValue* args) {
+  DCHECK(IsUnityWebappsInitialized());
+
+  // TODO move elsewhere
+  typedef void (* RemoveDomainFromPermissionsFunc) (gchar *);
+
+  std::string domain;
+  if (!args->GetString(0, &domain)) {
+    NOTREACHED();
+    return;
+  }
+
+  LOG(INFO) << "Removing domain:" << domain << " from the list of websites not prompting integration";
+
+  ((RemoveDomainFromPermissionsFunc) remove_from_permissions_func_handle_) (
+      const_cast<gchar *>(domain.c_str()));
+
+  LoadIntegratedWebsitesData ();
+}
+
+}  // namespace options
Index: dev.xenial/chrome/browser/ui/webui/options/desktop_integration_settings_handler.h
===================================================================
--- /dev/null
+++ dev.xenial/chrome/browser/ui/webui/options/desktop_integration_settings_handler.h
@@ -0,0 +1,76 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CHROME_BROWSER_UI_WEBUI_OPTIONS_DESKTOP_INTEGRATION_SETTINGS_HANDLER2_H_
+#define CHROME_BROWSER_UI_WEBUI_OPTIONS_DESKTOP_INTEGRATION_SETTINGS_HANDLER2_H_
+
+#include <string>
+
+#include "base/compiler_specific.h"
+#include "chrome/browser/ui/webui/options/options_ui.h"
+
+namespace base {
+class DictionaryValue;
+class ListValue;
+}
+
+namespace options {
+
+class DesktopIntegrationSettingsHandler : public OptionsPageUIHandler {
+ public:
+  DesktopIntegrationSettingsHandler();
+  virtual ~DesktopIntegrationSettingsHandler();
+
+  // OptionsPageUIHandler implementation.
+  virtual void GetLocalizedValues(
+      base::DictionaryValue* localized_strings) override;
+  virtual void InitializeHandler() override;
+  virtual void InitializePage() override;
+  virtual void RegisterMessages() override;
+
+ private:
+
+  // Loads the data associated with the currently integrated websites.
+  void LoadIntegratedWebsitesData();
+
+  // Initializes the list of website domain names that are currently
+  // either allowed or 'dontask' and send it to the overlay.
+  void LoadUnityWebappsEntryPoint();
+
+  // Removes an website from the list of integrated websites that won't prompt
+  // from integration.
+  // |args| - A string, the domain name of the website to remove.
+  void RemoveIntegrationSite(const base::ListValue* args);
+
+  // Adds an website from the list of integrated websites that won't prompt
+  // from integration. The website is being added to the list of 'allowed' sites.
+  // |args| - A string, the domain name of the website to add.
+  void AddIntegrationSite(const base::ListValue* args);
+
+  // Updates the integration allowed flag.
+  // |args| - A boolean flag indicating if integration should be allowed
+  void SetDesktopIntegrationAllowed(const base::ListValue* args);
+
+  // Updates the list of integrated websites.
+  void UpdateIntegratedWebsitesList(const base::ListValue* args);
+
+  // Predicate informing if we have been able to initialize the connection
+  // with the unity-webapps library (entry point for website integration
+  // permissions).
+  bool IsUnityWebappsInitialized() const;
+
+  // 
+  void * lib_unity_webapps_handle_;
+  void * get_all_func_handle_;
+  void * remove_from_permissions_func_handle_;
+  void * add_allowed_domain_func_handle_;
+  void * is_integration_allowed_func_handle_;
+  void * set_integration_allowed_func_handle_;
+
+  DISALLOW_COPY_AND_ASSIGN(DesktopIntegrationSettingsHandler);
+};
+
+}  // namespace options
+
+#endif  // CHROME_BROWSER_UI_WEBUI_OPTIONS_DESKTOP_INTEGRATION_SETTINGS_HANDLER2_H_
Index: dev.xenial/chrome/browser/ui/webui/options/options_ui.cc
===================================================================
--- dev.xenial.orig/chrome/browser/ui/webui/options/options_ui.cc
+++ dev.xenial/chrome/browser/ui/webui/options/options_ui.cc
@@ -108,6 +108,10 @@
 #include "chrome/browser/ui/webui/options/geolocation_options_handler.h"
 #endif
 
+#if defined(OS_LINUX)
+#include "chrome/browser/ui/webui/options/desktop_integration_settings_handler.h"
+#endif
+
 using content::RenderViewHost;
 
 namespace {
@@ -282,6 +286,8 @@ OptionsUI::OptionsUI(content::WebUI* web
   AddOptionsPageUIHandler(localized_strings, new CookiesViewHandler());
   AddOptionsPageUIHandler(localized_strings, new CreateProfileHandler());
   AddOptionsPageUIHandler(localized_strings, new EasyUnlockHandler());
+  AddOptionsPageUIHandler(localized_strings,
+                          new DesktopIntegrationSettingsHandler());
   AddOptionsPageUIHandler(localized_strings, new FontSettingsHandler());
 #if BUILDFLAG(ENABLE_GOOGLE_NOW)
   AddOptionsPageUIHandler(localized_strings, new GeolocationOptionsHandler());
Index: dev.xenial/chrome/chrome_browser_ui.gypi
===================================================================
--- dev.xenial.orig/chrome/chrome_browser_ui.gypi
+++ dev.xenial/chrome/chrome_browser_ui.gypi
@@ -1954,6 +1954,8 @@
       'browser/ui/webui/options/core_options_handler.h',
       'browser/ui/webui/options/create_profile_handler.cc',
       'browser/ui/webui/options/create_profile_handler.h',
+      'browser/ui/webui/options/desktop_integration_settings_handler.cc',
+      'browser/ui/webui/options/desktop_integration_settings_handler.h',
       'browser/ui/webui/options/easy_unlock_handler.cc',
       'browser/ui/webui/options/easy_unlock_handler.h',
       'browser/ui/webui/options/font_settings_handler.cc',
@@ -3091,6 +3093,7 @@
           ],
         }],
         ['use_aura==1 and chromeos==0 and use_ozone==0 and OS=="linux"', {
+          'cflags': [ '<!@(pkg-config --cflags glib-2.0)', ],
           'dependencies': [
             # gtk2 is the only component that can interact with gtk2 in our new
             # world.
