Index: dev.trusty/src/content/common/set_process_title_linux.cc
===================================================================
--- dev.trusty.orig/src/content/common/set_process_title_linux.cc	2014-02-28 12:46:56.726636741 -0500
+++ dev.trusty/src/content/common/set_process_title_linux.cc	2014-02-28 12:46:56.718636465 -0500
@@ -44,6 +44,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <stddef.h>
 
 extern char** environ;
 
@@ -87,7 +88,11 @@
   // Put the title in argv[0]. We have to zero out the space first since the
   // kernel doesn't actually look for a null terminator unless we make the
   // argument list longer than it started.
-  avail_size = page_end - (uintptr_t) g_main_argv[0];
+  avail_size = 0;
+  for (i = 0; g_main_argv[i]; ++i) {
+    avail_size += strlen(g_main_argv[i]) + 1;
+  }
+
   memset(g_main_argv[0], 0, avail_size);
   va_start(ap, fmt);
   if (fmt[0] == '-') {
Index: dev.trusty/src/content/common/set_process_title.cc
===================================================================
--- dev.trusty.orig/src/content/common/set_process_title.cc	2014-02-28 12:46:56.726636741 -0500
+++ dev.trusty/src/content/common/set_process_title.cc	2014-02-28 12:46:56.722636603 -0500
@@ -67,6 +67,9 @@
     // when the full command line is not being displayed in most process
     // listings.
     prctl(PR_SET_NAME, base::FilePath(title).BaseName().value().c_str());
+
+    // Limits the exe path part in the title
+    title = base::FilePath(title).BaseName().value().c_str();
   }
 #endif  // defined(OS_LINUX)
 
