Index: upstream/src/content/common/set_process_title.cc
===================================================================
--- upstream.orig/src/content/common/set_process_title.cc	2013-03-02 05:12:14.660732797 -0600
+++ upstream/src/content/common/set_process_title.cc	2013-03-03 07:21:11.721203997 -0600
@@ -65,6 +65,8 @@
     // full command line is not being displayed in most process listings.
     prctl(PR_SET_NAME, base::FilePath(title).BaseName().value().c_str());
 #endif  // defined(PR_SET_NAME)
+    // Limits the exe path part in the title
+    title = FilePath(title).BaseName().value().c_str();
   }
 #endif  // defined(OS_LINUX)
 
Index: upstream/src/content/common/set_process_title_linux.cc
===================================================================
--- upstream.orig/src/content/common/set_process_title_linux.cc	2013-03-02 05:12:12.780668108 -0600
+++ upstream/src/content/common/set_process_title_linux.cc	2013-03-03 07:22:44.516116860 -0600
@@ -44,6 +44,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <stddef.h>
 
 extern char** environ;
 
@@ -87,7 +88,11 @@
   // Put the title in argv[0]. We have to zero out the space first since the
   // kernel doesn't actually look for a null terminator unless we make the
   // argument list longer than it started.
-  avail_size = page_end - (uintptr_t) g_main_argv[0];
+  avail_size = 0;
+  for (i = 0; g_main_argv[i]; ++i) {
+    avail_size += strlen(g_main_argv[i]) + 1;
+  }
+
   memset(g_main_argv[0], 0, avail_size);
   va_start(ap, fmt);
   if (fmt[0] == '-') {
