Index: dev.vivid/chrome/browser/browser_resources.grd
===================================================================
--- dev.vivid.orig/chrome/browser/browser_resources.grd	2015-05-07 12:03:08.472338168 -0400
+++ dev.vivid/chrome/browser/browser_resources.grd	2015-05-07 12:03:08.461338319 -0400
@@ -242,6 +242,7 @@
       </if>
       <if expr="desktop_linux or (is_android and enable_plugins)">
         <include name="IDR_PLUGIN_DB_JSON" file="resources\plugin_metadata\plugins_linux.json" type="BINDATA" />
+        <include name="IDR_PLATFORM_PLUGIN_DB_JSON" file="resources\plugin_metadata\plugins_ubuntu_linux.json" type="BINDATA" />
       </if>
       <if expr="is_android">
         <include name="IDR_POLICY_CSS" file="resources\policy_android.css" type="BINDATA"/>
Index: dev.vivid/chrome/browser/plugins/plugin_finder.cc
===================================================================
--- dev.vivid.orig/chrome/browser/plugins/plugin_finder.cc	2015-05-07 12:03:08.472338168 -0400
+++ dev.vivid/chrome/browser/plugins/plugin_finder.cc	2015-05-07 12:03:08.463338291 -0400
@@ -151,44 +151,10 @@
                             static_cast<int>(PluginListError::NUM_VALUES));
 }
 
-}  // namespace
-
-// static
-void PluginFinder::RegisterPrefs(PrefRegistrySimple* registry) {
-  registry->RegisterBooleanPref(prefs::kDisablePluginFinder, false);
-}
-
-// static
-PluginFinder* PluginFinder::GetInstance() {
-  // PluginFinder::GetInstance() is the only method that's allowed to call
-  // Singleton<PluginFinder>::get().
-  return Singleton<PluginFinder>::get();
-}
-
-PluginFinder::PluginFinder() : version_(-1) {
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-}
-
-void PluginFinder::Init() {
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-  // Load the built-in plugin list first. If we have a newer version stored
-  // locally or download one, we will replace this one with it.
-  scoped_ptr<base::DictionaryValue> plugin_list(LoadBuiltInPluginList());
-
-  // Gracefully handle the case where we couldn't parse the built-in plugin list
-  // for some reason (https://crbug.com/388560). TODO(bauerb): Change back to a
-  // DCHECK once we have gathered more data about the underlying problem.
-  if (!plugin_list)
-    return;
-
-  ReinitializePlugins(plugin_list.get());
-}
-
-// static
-base::DictionaryValue* PluginFinder::LoadBuiltInPluginList() {
+base::DictionaryValue* LoadPlatformPluginListWithId(int resourceId) {
   base::StringPiece json_resource(
       ResourceBundle::GetSharedInstance().GetRawDataResource(
-          IDR_PLUGIN_DB_JSON));
+          resourceId));
   std::string error_str;
   int error_code = base::JSONReader::JSON_NO_ERROR;
   scoped_ptr<base::Value> value(base::JSONReader::ReadAndReturnError(
@@ -243,6 +209,59 @@
   return static_cast<base::DictionaryValue*>(value.release());
 }
 
+}  // namespace
+
+// static
+void PluginFinder::RegisterPrefs(PrefRegistrySimple* registry) {
+  registry->RegisterBooleanPref(prefs::kDisablePluginFinder, false);
+}
+
+void PluginFinder::MergePlatformSpecifiPlugins (base::DictionaryValue * plugin_list) {
+  if (!plugin_list)
+    return;
+  scoped_ptr<base::DictionaryValue> platform_plugin_list =
+    scoped_ptr<base::DictionaryValue>(LoadPlatformPluginList());
+  if (platform_plugin_list) {
+    plugin_list->MergeDictionary(platform_plugin_list.get());
+  }
+}
+
+// static
+PluginFinder* PluginFinder::GetInstance() {
+  // PluginFinder::GetInstance() is the only method that's allowed to call
+  // Singleton<PluginFinder>::get().
+  return Singleton<PluginFinder>::get();
+}
+
+PluginFinder::PluginFinder() : version_(-1) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+}
+
+void PluginFinder::Init() {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  // Load the built-in plugin list first. If we have a newer version stored
+  // locally or download one, we will replace this one with it.
+  scoped_ptr<base::DictionaryValue> plugin_list(LoadBuiltInPluginList());
+
+  // Gracefully handle the case where we couldn't parse the built-in plugin list
+  // for some reason (https://crbug.com/388560). TODO(bauerb): Change back to a
+  // DCHECK once we have gathered more data about the underlying problem.
+  if (!plugin_list)
+    return;
+
+  ReinitializePlugins(plugin_list.get());
+}
+
+// static
+DictionaryValue* PluginFinder::LoadPlatformPluginList() {
+  return LoadPlatformPluginListWithId(IDR_PLATFORM_PLUGIN_DB_JSON);
+}
+
+// static
+base::DictionaryValue* PluginFinder::LoadBuiltInPluginList() {
+  return LoadPlatformPluginListWithId(IDR_PLUGIN_DB_JSON);
+}
+
 PluginFinder::~PluginFinder() {
 #if defined(ENABLE_PLUGIN_INSTALLATION)
   STLDeleteValues(&installers_);
@@ -306,15 +325,20 @@
   if (version <= version_)
     return;
 
+  // Load the platform specific plugins (if any)
+  scoped_ptr<DictionaryValue> final_plugin_list(plugin_list->DeepCopy());
+  MergePlatformSpecifiPlugins(final_plugin_list.get());
+
   version_ = version;
 
   STLDeleteValues(&identifier_plugin_);
 
-  for (base::DictionaryValue::Iterator plugin_it(*plugin_list);
+  for (base::DictionaryValue::Iterator plugin_it(*final_plugin_list);
       !plugin_it.IsAtEnd(); plugin_it.Advance()) {
     const base::DictionaryValue* plugin = NULL;
     const std::string& identifier = plugin_it.key();
-    if (plugin_list->GetDictionaryWithoutPathExpansion(identifier, &plugin)) {
+    if (final_plugin_list->GetDictionaryWithoutPathExpansion(
+            identifier, &plugin)) {
       DCHECK(!identifier_plugin_[identifier]);
       identifier_plugin_[identifier] = CreatePluginMetadata(identifier, plugin);
 
Index: dev.vivid/chrome/browser/plugins/plugin_finder.h
===================================================================
--- dev.vivid.orig/chrome/browser/plugins/plugin_finder.h	2015-05-07 12:03:08.472338168 -0400
+++ dev.vivid/chrome/browser/plugins/plugin_finder.h	2015-05-07 12:03:08.466338249 -0400
@@ -80,6 +80,15 @@
   // Returns NULL if the plugin list couldn't be parsed.
   static base::DictionaryValue* LoadBuiltInPluginList();
 
+  // Loads platform specific whitelisted plugins from the browser
+  // resources and parses it.
+  // Returns NULL if the plug-in list couldn't be parsed.
+  static base::DictionaryValue* LoadPlatformPluginList();
+
+  // Merges the platform specific plugins to the list of plugins
+  // passed as parameters.
+  static void MergePlatformSpecifiPlugins(base::DictionaryValue * plugin_list);
+
 #if defined(ENABLE_PLUGIN_INSTALLATION)
   std::map<std::string, PluginInstaller*> installers_;
 #endif
Index: dev.vivid/chrome/browser/resources/plugin_metadata/plugins_ubuntu_linux.json
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dev.vivid/chrome/browser/resources/plugin_metadata/plugins_ubuntu_linux.json	2015-05-07 12:03:08.466338249 -0400
@@ -0,0 +1,18 @@
+{
+  "libunity_npapi_plugin.so": {
+    "mime_types": [
+      "application/x-unity-webapps-npapi"
+    ],
+    "versions": [
+      {
+        "version": "0",
+        "status": "up_to_date",
+        "reference": "https://launchpad.net/unity-chromium-extension/12.10"
+      }
+    ],
+    "lang": "en-US",
+    "name": "Unity WebApps plugin",
+    "url": "https://launchpad.net/ubuntu/quantal/+source/unity-chromium-extension",
+    "group_name_matcher": "Unity WebApps plugin"
+  }
+}
