Index: dev.saucy/src/chrome/browser/browser_resources.grd
===================================================================
--- dev.saucy.orig/src/chrome/browser/browser_resources.grd	2013-07-16 10:56:38.381884146 -0500
+++ dev.saucy/src/chrome/browser/browser_resources.grd	2013-07-16 10:56:38.369883722 -0500
@@ -197,6 +197,7 @@
         </if>
         <if expr="not pp_ifdef('chromeos')">
           <include name="IDR_PLUGIN_DB_JSON" file="resources\plugin_metadata\plugins_linux.json" type="BINDATA" />
+          <include name="IDR_PLATFORM_PLUGIN_DB_JSON" file="resources\plugin_metadata\plugins_ubuntu_linux.json" type="BINDATA" />
         </if>
       </if>
       <include name="IDR_POLICY_CSS" file="resources\policy.css" type="BINDATA"/>
Index: dev.saucy/src/chrome/browser/plugins/plugin_finder.cc
===================================================================
--- dev.saucy.orig/src/chrome/browser/plugins/plugin_finder.cc	2013-07-16 10:56:38.381884146 -0500
+++ dev.saucy/src/chrome/browser/plugins/plugin_finder.cc	2013-07-16 10:56:38.373883863 -0500
@@ -134,6 +134,29 @@
   return plugin;
 }
 
+base::DictionaryValue* LoadPlatformPluginListWithId(int resourceId) {
+#if defined(OS_WIN) || defined(OS_MACOSX) || defined(OS_LINUX)
+  base::StringPiece json_resource(
+      ResourceBundle::GetSharedInstance().GetRawDataResource(
+          resourceId));
+  std::string error_str;
+  scoped_ptr<base::Value> value(base::JSONReader::ReadAndReturnError(
+      json_resource,
+      base::JSON_PARSE_RFC,
+      NULL,
+      &error_str));
+  if (!value.get()) {
+    DLOG(ERROR) << error_str;
+    return NULL;
+  }
+  if (value->GetType() != base::Value::TYPE_DICTIONARY)
+    return NULL;
+  return static_cast<base::DictionaryValue*>(value.release());
+#else
+  return new DictionaryValue();
+#endif
+}
+
 }  // namespace
 
 // static
@@ -141,6 +164,16 @@
   registry->RegisterBooleanPref(prefs::kDisablePluginFinder, false);
 }
 
+void PluginFinder::MergePlatformSpecifiPlugins (base::DictionaryValue * plugin_list) {
+  if (!plugin_list)
+    return;
+  scoped_ptr<base::DictionaryValue> platform_plugin_list =
+    scoped_ptr<base::DictionaryValue>(LoadPlatformPluginList());
+  if (platform_plugin_list) {
+    plugin_list->MergeDictionary(platform_plugin_list.get());
+  }
+}
+
 // static
 PluginFinder* PluginFinder::GetInstance() {
   // PluginFinder::GetInstance() is the only method that's allowed to call
@@ -162,23 +195,13 @@
 }
 
 // static
+DictionaryValue* PluginFinder::LoadPlatformPluginList() {
+  return LoadPlatformPluginListWithId(IDR_PLATFORM_PLUGIN_DB_JSON);
+}
+
+// static
 DictionaryValue* PluginFinder::LoadBuiltInPluginList() {
-  base::StringPiece json_resource(
-      ResourceBundle::GetSharedInstance().GetRawDataResource(
-          IDR_PLUGIN_DB_JSON));
-  std::string error_str;
-  scoped_ptr<base::Value> value(base::JSONReader::ReadAndReturnError(
-      json_resource,
-      base::JSON_PARSE_RFC,
-      NULL,
-      &error_str));
-  if (!value.get()) {
-    DLOG(ERROR) << error_str;
-    return NULL;
-  }
-  if (value->GetType() != base::Value::TYPE_DICTIONARY)
-    return NULL;
-  return static_cast<base::DictionaryValue*>(value.release());
+  return LoadPlatformPluginListWithId(IDR_PLUGIN_DB_JSON);
 }
 
 PluginFinder::~PluginFinder() {
@@ -244,16 +267,21 @@
   if (version <= version_)
     return;
 
+  // Load the platform specific plugins (if any)
+  scoped_ptr<DictionaryValue> final_plugin_list(plugin_list->DeepCopy());
+  MergePlatformSpecifiPlugins(final_plugin_list.get());
+
   version_ = version;
 
   STLDeleteValues(&identifier_plugin_);
   identifier_plugin_.clear();
 
-  for (DictionaryValue::Iterator plugin_it(*plugin_list);
+  for (DictionaryValue::Iterator plugin_it(*final_plugin_list);
       !plugin_it.IsAtEnd(); plugin_it.Advance()) {
     const DictionaryValue* plugin = NULL;
     const std::string& identifier = plugin_it.key();
-    if (plugin_list->GetDictionaryWithoutPathExpansion(identifier, &plugin)) {
+    if (final_plugin_list->GetDictionaryWithoutPathExpansion(
+            identifier, &plugin)) {
       DCHECK(!identifier_plugin_[identifier]);
       identifier_plugin_[identifier] = CreatePluginMetadata(identifier, plugin);
 
Index: dev.saucy/src/chrome/browser/plugins/plugin_finder.h
===================================================================
--- dev.saucy.orig/src/chrome/browser/plugins/plugin_finder.h	2013-07-16 10:56:38.381884146 -0500
+++ dev.saucy/src/chrome/browser/plugins/plugin_finder.h	2013-07-16 10:56:38.373883863 -0500
@@ -80,6 +80,15 @@
   // Returns NULL if the plug-in list couldn't be parsed.
   static base::DictionaryValue* LoadBuiltInPluginList();
 
+  // Loads platform specific whitelisted plugins from the browser
+  // resources and parses it.
+  // Returns NULL if the plug-in list couldn't be parsed.
+  static base::DictionaryValue* LoadPlatformPluginList();
+
+  // Merges the platform specific plugins to the list of plugins
+  // passed as parameters.
+  static void MergePlatformSpecifiPlugins(base::DictionaryValue * plugin_list);
+
 #if defined(ENABLE_PLUGIN_INSTALLATION)
   std::map<std::string, PluginInstaller*> installers_;
 #endif
Index: dev.saucy/src/chrome/browser/resources/plugin_metadata/plugins_ubuntu_linux.json
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dev.saucy/src/chrome/browser/resources/plugin_metadata/plugins_ubuntu_linux.json	2013-07-16 10:56:38.373883863 -0500
@@ -0,0 +1,18 @@
+{
+  "libunity_npapi_plugin.so": {
+    "mime_types": [
+      "application/x-unity-webapps-npapi"
+    ],
+    "versions": [
+      {
+        "version": "0",
+        "status": "up_to_date",
+        "reference": "https://launchpad.net/unity-chromium-extension/12.10"
+      }
+    ],
+    "lang": "en-US",
+    "name": "Unity WebApps plugin",
+    "url": "https://launchpad.net/ubuntu/quantal/+source/unity-chromium-extension",
+    "group_name_matcher": "Unity WebApps plugin"
+  }
+}
