Index: src/chrome/browser/renderer_host/plugin_info_message_filter.cc
===================================================================
--- src.orig/chrome/browser/renderer_host/plugin_info_message_filter.cc	2012-10-01 11:04:03.000000000 -0400
+++ src/chrome/browser/renderer_host/plugin_info_message_filter.cc	2012-10-11 17:28:28.525409907 -0400
@@ -160,14 +160,8 @@
   DCHECK(plugin_setting != CONTENT_SETTING_DEFAULT);
 
 #if defined(ENABLE_PLUGIN_INSTALLATION)
-#if defined(OS_LINUX)
-  // On Linux, unknown plugins require authorization.
-  PluginInstaller::SecurityStatus plugin_status =
-      PluginInstaller::SECURITY_STATUS_REQUIRES_AUTHORIZATION;
-#else
   PluginInstaller::SecurityStatus plugin_status =
       PluginInstaller::SECURITY_STATUS_UP_TO_DATE;
-#endif
   PluginInstaller* installer =
       plugin_finder->FindPluginWithIdentifier(group->identifier());
   if (installer)
