Index: dev/components/machine_intelligence/generic_logistic_regression_inference.cc
===================================================================
--- dev.orig/components/machine_intelligence/generic_logistic_regression_inference.cc
+++ dev/components/machine_intelligence/generic_logistic_regression_inference.cc
@@ -4,6 +4,8 @@
 
 #include "components/machine_intelligence/generic_logistic_regression_inference.h"
 
+#include <math.h>
+
 #include "base/logging.h"
 #include "components/machine_intelligence/ranker_example_util.h"
 
