Index: dev.vivid/components/search_engines/prepopulated_engines.json
===================================================================
--- dev.vivid.orig/components/search_engines/prepopulated_engines.json	2015-06-02 15:12:50.067425189 -0400
+++ dev.vivid/components/search_engines/prepopulated_engines.json	2015-06-02 15:12:50.062425256 -0400
@@ -14,6 +14,7 @@
 //    88, 89, 91+
 //
 // IDs > 1000 are reserved for distribution custom engines.
+// 1001: DuckDuckGo
 //
 // NOTES:
 //   - CHANGE THE ABOVE NUMBERS IF YOU ADD A NEW ENGINE; ID conflicts = bad!
@@ -26,7 +27,7 @@
   "int_variables": {
     // The following id is for UMA stats only. Please update it if you add new
     // engines outside the current range or it will not be counted in stats.
-    "kMaxPrepopulatedEngineID": 90,
+    "kMaxPrepopulatedEngineID": 1001,
 
     // Increment this if you change the data in ways that mean users with
     // existing data should get a new version.
@@ -105,6 +106,16 @@
       "id": 68
     },
 
+    "duckduckgo": {
+      "name": "DuckDuckGo",
+      "keyword": "duckduckgo.com",
+      "favicon_url": "https://duckduckgo.com/favicon.ico",
+      "search_url": "https://duckduckgo.com/?q={searchTerms}&t=canonical",
+      "suggest_url": "https://duckduckgo.com/ac/?q={searchTerms}&type=list",
+      "type": "SEARCH_ENGINE_DUCKDUCKGO",
+      "id": 1001
+    },
+
     "google": {
       "name": "Google",
       "keyword": "google.com",
Index: dev.vivid/components/search_engines/search_engine_type.h
===================================================================
--- dev.vivid.orig/components/search_engines/search_engine_type.h	2015-06-02 15:12:50.067425189 -0400
+++ dev.vivid/components/search_engines/search_engine_type.h	2015-06-02 15:12:50.062425256 -0400
@@ -22,6 +22,7 @@
   SEARCH_ENGINE_DAUM,
   SEARCH_ENGINE_DELFI,
   SEARCH_ENGINE_DELTA,
+  SEARCH_ENGINE_DUCKDUCKGO,
   SEARCH_ENGINE_FUNMOODS,
   SEARCH_ENGINE_GOO,
   SEARCH_ENGINE_GOOGLE,
Index: dev.vivid/components/search_engines/template_url_prepopulate_data.cc
===================================================================
--- dev.vivid.orig/components/search_engines/template_url_prepopulate_data.cc	2015-06-02 15:12:50.067425189 -0400
+++ dev.vivid/components/search_engines/template_url_prepopulate_data.cc	2015-06-02 15:12:50.062425256 -0400
@@ -48,7 +48,7 @@
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* engines_default[] =
-    { &google, &bing, &yahoo, };
+    { &google, &bing, &yahoo, &duckduckgo, };
 
 // United Arab Emirates
 const PrepopulatedEngine* engines_AE[] =
@@ -140,7 +140,7 @@
 
 // Germany
 const PrepopulatedEngine* engines_DE[] =
-    { &google, &bing, &yahoo_de };
+    { &google, &bing, &yahoo_de, &duckduckgo, };
 
 // Denmark
 const PrepopulatedEngine* engines_DK[] =
@@ -180,11 +180,11 @@
 
 // France
 const PrepopulatedEngine* engines_FR[] =
-    { &google, &bing, &yahoo_fr, };
+    { &google, &bing, &yahoo_fr, &duckduckgo, };
 
 // United Kingdom
 const PrepopulatedEngine* engines_GB[] =
-    { &google, &bing, &yahoo_uk, &ask_uk, };
+    { &google, &bing, &yahoo_uk, &ask_uk, &duckduckgo, };
 
 // Greece
 const PrepopulatedEngine* engines_GR[] =
@@ -452,7 +452,7 @@
 
 // United States
 const PrepopulatedEngine* engines_US[] =
-    { &google, &bing, &yahoo, &aol, &ask, };
+    { &google, &bing, &yahoo, &aol, &ask, &duckduckgo, };
 
 // Uruguay
 const PrepopulatedEngine* engines_UY[] =
@@ -483,6 +483,7 @@
   // Prepopulated engines:
   &aol,          &ask,          &ask_br,       &ask_uk,       &baidu,
   &bing,         &daum,         &google,       &kvasir,       &mail_ru,
+  &duckduckgo,
   &najdi,        &naver,        &onet,         &seznam,       &sogou,
   &vinden,       &virgilio,     &yahoo,        &yahoo_ar,     &yahoo_at,
   &yahoo_au,     &yahoo_br,     &yahoo_ca,     &yahoo_ch,     &yahoo_cl,
