Index: dev.trusty/components/search_engines/prepopulated_engines.json
===================================================================
--- dev.trusty.orig/components/search_engines/prepopulated_engines.json	2014-07-04 08:23:32.915312906 -0400
+++ dev.trusty/components/search_engines/prepopulated_engines.json	2014-07-04 08:23:32.911312906 -0400
@@ -14,6 +14,7 @@
 //    88, 89, 91+
 //
 // IDs > 1000 are reserved for distribution custom engines.
+// 1001: DuckDuckGo
 //
 // NOTES:
 //   - CHANGE THE ABOVE NUMBERS IF YOU ADD A NEW ENGINE; ID conflicts = bad!
@@ -23,7 +24,7 @@
   "int_variables": {
     // The following id is for UMA stats only. Please update it if you add new
     // engines outside the current range or it will not be counted in stats.
-    "kMaxPrepopulatedEngineID": 90,
+    "kMaxPrepopulatedEngineID": 1001,
 
     // Increment this if you change the data in ways that mean users with
     // existing data should get a new version.
@@ -521,6 +522,16 @@
       "id": 68
     },
 
+    "duckduckgo": {
+      "name": "DuckDuckGo",
+      "keyword": "duckduckgo.com",
+      "favicon_url": "https://duckduckgo.com/favicon.ico",
+      "search_url": "https://duckduckgo.com/?q={searchTerms}&t=canonical",
+      "suggest_url": "https://duckduckgo.com/ac/?q={searchTerms}&type=list",
+      "type": "SEARCH_ENGINE_DUCKDUCKGO",
+      "id": 1001
+    },
+
     "google": {
       "name": "Google",
       "keyword": "google.com",
Index: dev.trusty/components/search_engines/search_engine_type.h
===================================================================
--- dev.trusty.orig/components/search_engines/search_engine_type.h	2014-07-04 08:23:32.915312906 -0400
+++ dev.trusty/components/search_engines/search_engine_type.h	2014-07-04 08:23:32.911312906 -0400
@@ -22,6 +22,7 @@
   SEARCH_ENGINE_DAUM,
   SEARCH_ENGINE_DELFI,
   SEARCH_ENGINE_DELTA,
+  SEARCH_ENGINE_DUCKDUCKGO,
   SEARCH_ENGINE_FUNMOODS,
   SEARCH_ENGINE_GOO,
   SEARCH_ENGINE_GOOGLE,
Index: dev.trusty/chrome/browser/search_engines/template_url_prepopulate_data.cc
===================================================================
--- dev.trusty.orig/chrome/browser/search_engines/template_url_prepopulate_data.cc	2014-07-04 08:23:32.915312906 -0400
+++ dev.trusty/chrome/browser/search_engines/template_url_prepopulate_data.cc	2014-07-04 08:23:32.912312906 -0400
@@ -49,7 +49,7 @@
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* engines_default[] =
-    { &google, &bing, &yahoo, };
+    { &google, &bing, &yahoo, &duckduckgo, };
 
 // United Arab Emirates
 const PrepopulatedEngine* engines_AE[] =
@@ -141,7 +141,7 @@
 
 // Germany
 const PrepopulatedEngine* engines_DE[] =
-    { &google, &bing_de_DE, &yahoo_de };
+    { &google, &bing_de_DE, &yahoo_de, &duckduckgo, };
 
 // Denmark
 const PrepopulatedEngine* engines_DK[] =
@@ -181,11 +181,11 @@
 
 // France
 const PrepopulatedEngine* engines_FR[] =
-    { &google, &bing_fr_FR, &yahoo_fr, };
+    { &google, &bing_fr_FR, &yahoo_fr, &duckduckgo, };
 
 // United Kingdom
 const PrepopulatedEngine* engines_GB[] =
-    { &google, &bing_en_GB, &yahoo_uk, &ask_uk, };
+    { &google, &bing_en_GB, &yahoo_uk, &ask_uk, &duckduckgo, };
 
 // Greece
 const PrepopulatedEngine* engines_GR[] =
@@ -453,7 +453,7 @@
 
 // United States
 const PrepopulatedEngine* engines_US[] =
-    { &google, &bing_en_US, &yahoo, &aol, &ask, };
+    { &google, &bing_en_US, &yahoo, &aol, &ask, &duckduckgo, };
 
 // Uruguay
 const PrepopulatedEngine* engines_UY[] =
@@ -491,7 +491,7 @@
   &bing_fr_FR,   &bing_it_IT,   &bing_ja_JP,   &bing_lv_LV,   &bing_nb_NO,
   &bing_nl_BE,   &bing_pl_PL,   &bing_pt_BR,   &bing_pt_PT,   &bing_ru_RU,
   &bing_sv_SE,   &bing_tr_TR,   &bing_zh_HK,   &bing_zh_TW,   &daum,
-  &google,       &kvasir,       &mail_ru,      &najdi,        &naver,
+  &duckduckgo, &google, &kvasir, &mail_ru,     &najdi,        &naver,
   &onet,         &seznam,       &sogou,        &vinden,       &virgilio,
   &yahoo,        &yahoo_ar,     &yahoo_at,     &yahoo_au,     &yahoo_br,
   &yahoo_ca,     &yahoo_ch,     &yahoo_cl,     &yahoo_co,     &yahoo_de,
