Index: dev.raring/src/crypto/nss_util.cc
===================================================================
--- dev.raring.orig/src/crypto/nss_util.cc	2013-05-08 17:02:56.479011811 -0500
+++ dev.raring/src/crypto/nss_util.cc	2013-05-08 17:02:56.475011673 -0500
@@ -706,7 +706,11 @@
 #elif defined(ARCH_CPU_X86)
   paths.push_back(base::FilePath("/usr/lib/i386-linux-gnu/nss"));
 #elif defined(ARCH_CPU_ARMEL)
+# ifdef __ARM_PCS_VFP
+  paths.push_back(base::FilePath("/usr/lib/arm-linux-gnueabihf/nss"));
+# else
   paths.push_back(base::FilePath("/usr/lib/arm-linux-gnueabi/nss"));
+# endif
 #endif
 
   // A list of library files to load.
