Index: dev.wily/third_party/skia/gyp/skia_lib.gyp
===================================================================
--- dev.wily.orig/third_party/skia/gyp/skia_lib.gyp
+++ dev.wily/third_party/skia/gyp/skia_lib.gyp
@@ -16,6 +16,7 @@
       'ports.gyp:ports',
       'sfnt.gyp:sfnt',
       'utils.gyp:utils',
+      "<!(echo <(skia_arch_type) <(skia_os) 1>&2)"
     ],
     'conditions': [
       [ '"x86" in skia_arch_type and skia_os != "android"', {
@@ -24,7 +25,7 @@
           'opts.gyp:opts_sse41',
         ],
       }],
-      [ 'arm_neon == 1', {
+      [ '1', {
         'component_libs': [
           'opts.gyp:opts_neon',
         ],
Index: dev.wily/ui/base/resource/data_pack.cc
===================================================================
--- dev.wily.orig/ui/base/resource/data_pack.cc
+++ dev.wily/ui/base/resource/data_pack.cc
@@ -157,9 +157,11 @@ bool DataPack::LoadImpl() {
   // 2) Verify the entries are within the appropriate bounds. There's an extra
   // entry after the last item which gives us the length of the last item.
   for (size_t i = 0; i < resource_count_ + 1; ++i) {
+    uint32_t t;
     const DataPackEntry* entry = reinterpret_cast<const DataPackEntry*>(
         mmap_->data() + kHeaderLength + (i * sizeof(DataPackEntry)));
-    if (entry->file_offset > mmap_->length()) {
+    memcpy(&t, &entry->file_offset, 32/8);
+    if (t > mmap_->length()) {
       LOG(ERROR) << "Entry #" << i << " in data pack points off end of file. "
                  << "Was the file corrupted?";
       UMA_HISTOGRAM_ENUMERATION("DataPack.Load", ENTRY_NOT_FOUND,
