Index: dev.saucy/src/ui/base/resource/data_pack.cc
===================================================================
--- dev.saucy.orig/src/ui/base/resource/data_pack.cc	2013-05-22 16:33:48.471530331 -0500
+++ dev.saucy/src/ui/base/resource/data_pack.cc	2013-05-22 16:33:48.455529776 -0500
@@ -147,9 +147,11 @@
   // 2) Verify the entries are within the appropriate bounds. There's an extra
   // entry after the last item which gives us the length of the last item.
   for (size_t i = 0; i < resource_count_ + 1; ++i) {
+    uint32 t;
     const DataPackEntry* entry = reinterpret_cast<const DataPackEntry*>(
         mmap_->data() + kHeaderLength + (i * sizeof(DataPackEntry)));
-    if (entry->file_offset > mmap_->length()) {
+    memcpy(&t, &entry->file_offset, 32/8);
+    if (t > mmap_->length()) {
       LOG(ERROR) << "Entry #" << i << " in data pack points off end of file. "
                  << "Was the file corrupted?";
       UMA_HISTOGRAM_ENUMERATION("DataPack.Load", ENTRY_NOT_FOUND,
