Index: beta/ui/base/resource/data_pack.cc
===================================================================
--- beta.orig/ui/base/resource/data_pack.cc
+++ beta/ui/base/resource/data_pack.cc
@@ -237,9 +237,11 @@ bool DataPack::LoadImpl(std::unique_ptr<
   // 2) Verify the entries are within the appropriate bounds. There's an extra
   // entry after the last item which gives us the length of the last item.
   for (size_t i = 0; i < resource_count_ + 1; ++i) {
+    uint32_t t;
     const DataPackEntry* entry = reinterpret_cast<const DataPackEntry*>(
         data_source->GetData() + kHeaderLength + (i * sizeof(DataPackEntry)));
-    if (entry->file_offset > data_source->GetLength()) {
+    memcpy(&t, &entry->file_offset, 32/8);
+    if (t > data_source->GetLength()) {
       LOG(ERROR) << "Entry #" << i << " in data pack points off end of file. "
                  << "Was the file corrupted?";
       LogDataPackError(ENTRY_NOT_FOUND);
