Index: dev.vivid/third_party/skia/gyp/skia_lib.gyp
===================================================================
--- dev.vivid.orig/third_party/skia/gyp/skia_lib.gyp	2015-06-02 15:12:48.845441259 -0400
+++ dev.vivid/third_party/skia/gyp/skia_lib.gyp	2015-06-02 15:12:48.842441298 -0400
@@ -15,6 +15,7 @@
       'ports.gyp:ports',
       'sfnt.gyp:sfnt',
       'utils.gyp:utils',
+      "<!(echo <(skia_arch_type) <(skia_os) 1>&2)"
     ],
     'conditions': [
       [ '"x86" in skia_arch_type and skia_os != "android"', {
@@ -23,7 +24,7 @@
           'opts.gyp:opts_sse41',
         ],
       }],
-      [ 'arm_neon == 1', {
+      [ '1', {
         'component_libs': [
           'opts.gyp:opts_neon',
         ],
Index: dev.vivid/ui/base/resource/data_pack.cc
===================================================================
--- dev.vivid.orig/ui/base/resource/data_pack.cc	2015-06-02 15:12:48.845441259 -0400
+++ dev.vivid/ui/base/resource/data_pack.cc	2015-06-02 15:12:48.842441298 -0400
@@ -155,9 +155,11 @@
   // 2) Verify the entries are within the appropriate bounds. There's an extra
   // entry after the last item which gives us the length of the last item.
   for (size_t i = 0; i < resource_count_ + 1; ++i) {
+    uint32 t;
     const DataPackEntry* entry = reinterpret_cast<const DataPackEntry*>(
         mmap_->data() + kHeaderLength + (i * sizeof(DataPackEntry)));
-    if (entry->file_offset > mmap_->length()) {
+    memcpy(&t, &entry->file_offset, 32/8);
+    if (t > mmap_->length()) {
       LOG(ERROR) << "Entry #" << i << " in data pack points off end of file. "
                  << "Was the file corrupted?";
       UMA_HISTOGRAM_ENUMERATION("DataPack.Load", ENTRY_NOT_FOUND,
