Index: dev.trusty/src/build/gyp_chromium
===================================================================
--- dev.trusty.orig/src/build/gyp_chromium	2014-02-23 21:47:45.178813386 -0500
+++ dev.trusty/src/build/gyp_chromium	2014-02-23 21:47:45.162812824 -0500
@@ -39,8 +39,6 @@
 sys.path.insert(1, os.path.join(chrome_src, 'third_party', 'WebKit',
     'Source', 'build', 'scripts'))
 
-import find_depot_tools
-
 # On Windows, Psyco shortens warm runs of build/gyp_chromium by about
 # 20 seconds on a z600 machine with 12 GB of RAM, from 90 down to 70
 # seconds.  Conversely, memory usage of build/gyp_chromium with Psyco
@@ -376,6 +374,7 @@
   # TODO(bradnelson): take this out once this issue is fixed:
   #    http://code.google.com/p/gyp/issues/detail?id=177
   if sys.platform == 'cygwin':
+    import find_depot_tools
     depot_tools_path = find_depot_tools.add_depot_tools_to_path()
     python_dir = sorted(glob.glob(os.path.join(depot_tools_path,
                                                'python2*_bin')))[-1]
@@ -438,6 +437,7 @@
   # depot_tools, then use it.
   vs2013_runtime_dll_dirs = None
   if sys.platform in ('win32', 'cygwin'):
+    import find_depot_tools
     depot_tools_path = find_depot_tools.add_depot_tools_to_path()
     toolchain = os.path.normpath(os.path.join(
         depot_tools_path, 'win_toolchain', 'vs2013_files'))
