Index: dev.vivid/chrome/test/chromedriver/embed_version_in_cpp.py
===================================================================
--- dev.vivid.orig/chrome/test/chromedriver/embed_version_in_cpp.py	2015-04-17 14:50:28.246334272 -0400
+++ dev.vivid/chrome/test/chromedriver/embed_version_in_cpp.py	2015-04-17 14:50:28.201334907 -0400
@@ -26,7 +26,9 @@
   options, args = parser.parse_args()
 
   version = open(options.version_file, 'r').read().strip()
-  revision = lastchange.FetchVersionInfo(None).revision
+  lastchangeFile = open(os.path.join(chrome_paths.GetSrc(), 'build', 'util', 'LASTCHANGE'), 'r')
+  lastchangeFile.seek(11)
+  revision = lastchangeFile.read()
 
   if revision:
     match = re.match('([0-9a-fA-F]+)(-refs/heads/master@{#(\d+)})?', revision)
