Index: dev.utopic/chrome/test/chromedriver/embed_version_in_cpp.py
===================================================================
--- dev.utopic.orig/chrome/test/chromedriver/embed_version_in_cpp.py	2014-08-14 09:23:29.764321495 -0400
+++ dev.utopic/chrome/test/chromedriver/embed_version_in_cpp.py	2014-08-14 09:23:29.762321494 -0400
@@ -25,7 +25,9 @@
   options, args = parser.parse_args()
 
   version = open(options.version_file, 'r').read().strip()
-  revision = lastchange.FetchVersionInfo(None).revision
+  lastchangeFile = open(os.path.join(chrome_paths.GetSrc(), 'build', 'util', 'LASTCHANGE'), 'r')
+  lastchangeFile.seek(11)
+  revision = lastchangeFile.read()
   if revision:
     version += '.' + revision.strip()
 
