Index: dev.saucy/src/chrome/test/chromedriver/embed_version_in_cpp.py
===================================================================
--- dev.saucy.orig/src/chrome/test/chromedriver/embed_version_in_cpp.py	2013-10-19 11:58:43.060373537 -0400
+++ dev.saucy/src/chrome/test/chromedriver/embed_version_in_cpp.py	2013-10-19 11:58:43.044372962 -0400
@@ -25,7 +25,9 @@
   options, args = parser.parse_args()
 
   version = open(options.version_file, 'r').read().strip()
-  revision = lastchange.FetchVersionInfo(None).revision
+  lastchangeFile = open(os.path.join(chrome_paths.GetSrc(), 'build', 'util', 'LASTCHANGE'), 'r')
+  lastchangeFile.seek(11)
+  revision = lastchangeFile.read()
   if revision:
     version += '.' + revision.strip()
 
