Index: dev.trusty/src/chrome/test/chromedriver/embed_version_in_cpp.py
===================================================================
--- dev.trusty.orig/src/chrome/test/chromedriver/embed_version_in_cpp.py	2014-02-05 23:42:06.413321604 -0500
+++ dev.trusty/src/chrome/test/chromedriver/embed_version_in_cpp.py	2014-02-05 23:42:06.401321151 -0500
@@ -25,7 +25,9 @@
   options, args = parser.parse_args()
 
   version = open(options.version_file, 'r').read().strip()
-  revision = lastchange.FetchVersionInfo(None).revision
+  lastchangeFile = open(os.path.join(chrome_paths.GetSrc(), 'build', 'util', 'LASTCHANGE'), 'r')
+  lastchangeFile.seek(11)
+  revision = lastchangeFile.read()
   if revision:
     version += '.' + revision.strip()
 
