Index: beta/build/toolchain/gcc_toolchain.gni
===================================================================
--- beta.orig/build/toolchain/gcc_toolchain.gni
+++ beta/build/toolchain/gcc_toolchain.gni
@@ -584,12 +584,11 @@ template("clang_toolchain") {
   }
 
   gcc_toolchain(target_name) {
-    prefix = rebase_path("$clang_base_path/bin", root_build_dir)
-    cc = "$prefix/clang"
-    cxx = "$prefix/clang++"
+    cc = "clang-4.0"
+    cxx = "clang++-4.0"
     ld = cxx
     readelf = "${toolprefix}readelf"
-    ar = "${prefix}/llvm-ar"
+    ar = "${toolprefix}ar"
     nm = "${toolprefix}nm"
 
     forward_variables_from(invoker,
