Index: dev/build/toolchain/gcc_toolchain.gni
===================================================================
--- dev.orig/build/toolchain/gcc_toolchain.gni
+++ dev/build/toolchain/gcc_toolchain.gni
@@ -548,9 +548,8 @@ template("clang_toolchain") {
   }
 
   gcc_toolchain(target_name) {
-    prefix = rebase_path("$clang_base_path/bin", root_build_dir)
-    cc = "$prefix/clang"
-    cxx = "$prefix/clang++"
+    cc = "clang"
+    cxx = "clang++"
     ld = cxx
     readelf = "${toolprefix}readelf"
     ar = "${toolprefix}ar"
