Index: beta/build/toolchain/gcc_toolchain.gni
===================================================================
--- beta.orig/build/toolchain/gcc_toolchain.gni
+++ beta/build/toolchain/gcc_toolchain.gni
@@ -483,9 +483,8 @@ template("clang_toolchain") {
   }
 
   gcc_toolchain(target_name) {
-    prefix = rebase_path("$clang_base_path/bin", root_build_dir)
-    cc = "$prefix/clang"
-    cxx = "$prefix/clang++"
+    cc = "clang"
+    cxx = "clang++"
     ld = cxx
 
     if (use_clang_static_analyzer) {
