Index: beta/mojo/public/tools/mojom/mojom_parser.py
===================================================================
--- beta.orig/mojo/public/tools/mojom/mojom_parser.py
+++ beta/mojo/public/tools/mojom/mojom_parser.py
@@ -30,7 +30,12 @@ from mojom.parse import conditional_feat
 
 
 # Disable this for easier debugging.
-_ENABLE_MULTIPROCESSING = True
+#
+# When doing a local armhf compilation (through qemu), this script seems to
+# hang, possibly due to multiprocessing, so just disable it. There's other
+# processes running anyways in a regular build that can use the other CPU
+# cores.
+_ENABLE_MULTIPROCESSING = False
 
 # https://docs.python.org/3/library/multiprocessing.html#:~:text=bpo-33725
 if __name__ == '__main__' and sys.platform == 'darwin':
