---
 base/logging.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/base/logging.h
===================================================================
--- src.orig/base/logging.h
+++ src/base/logging.h
@@ -451,7 +451,7 @@
 // uses OFFICIAL_BUILD and other platforms use the branding flag when NDEBUG is
 // defined.
 #if ( defined(OS_WIN) && defined(OFFICIAL_BUILD)) || \
-    (!defined(OS_WIN) && defined(NDEBUG) && defined(GOOGLE_CHROME_BUILD))
+    (!defined(OS_WIN) && defined(NDEBUG))
 #define LOGGING_IS_OFFICIAL_BUILD 1
 #else
 #define LOGGING_IS_OFFICIAL_BUILD 0
