Index: dev.saucy/src/base/logging.h
===================================================================
--- dev.saucy.orig/src/base/logging.h	2013-05-22 16:33:47.327490637 -0500
+++ dev.saucy/src/base/logging.h	2013-05-22 16:33:47.315490220 -0500
@@ -451,7 +451,7 @@
 // uses OFFICIAL_BUILD and other platforms use the branding flag when NDEBUG is
 // defined.
 #if ( defined(OS_WIN) && defined(OFFICIAL_BUILD)) || \
-    (!defined(OS_WIN) && defined(NDEBUG) && defined(GOOGLE_CHROME_BUILD))
+    (!defined(OS_WIN) && defined(NDEBUG))
 #define LOGGING_IS_OFFICIAL_BUILD 1
 #else
 #define LOGGING_IS_OFFICIAL_BUILD 0
