Index: dev.raring/src/base/logging.h
===================================================================
--- dev.raring.orig/src/base/logging.h	2013-05-11 12:56:32.797331080 -0500
+++ dev.raring/src/base/logging.h	2013-05-11 12:56:32.793330941 -0500
@@ -451,7 +451,7 @@
 // uses OFFICIAL_BUILD and other platforms use the branding flag when NDEBUG is
 // defined.
 #if ( defined(OS_WIN) && defined(OFFICIAL_BUILD)) || \
-    (!defined(OS_WIN) && defined(NDEBUG) && defined(GOOGLE_CHROME_BUILD))
+    (!defined(OS_WIN) && defined(NDEBUG))
 #define LOGGING_IS_OFFICIAL_BUILD 1
 #else
 #define LOGGING_IS_OFFICIAL_BUILD 0
