Index: dev.trusty/src/build/gyp_chromium
===================================================================
--- dev.trusty.orig/src/build/gyp_chromium	2013-12-08 09:29:16.680693231 -0500
+++ dev.trusty/src/build/gyp_chromium	2013-12-08 09:29:23.004915702 -0500
@@ -537,8 +537,6 @@
           'GYP_CROSSCOMPILE' not in os.environ)):
     os.environ['GYP_CROSSCOMPILE'] = '1'
 
-  if not RunGN(gn_vars_dict):
-    sys.exit(1)
   args.extend(
       ['-I' + i for i in additional_include_files(supplemental_includes, args)])
 
