Index: dev/BUILD.gn
===================================================================
--- dev.orig/BUILD.gn
+++ dev/BUILD.gn
@@ -95,7 +95,6 @@ group("gn_all") {
       "//codelabs",
       "//components:components_unittests",
       "//components/gwp_asan:gwp_asan_unittests",
-      "//infra/orchestrator:orchestrator_all",
       "//net:net_unittests",
       "//sandbox:sandbox_unittests",
       "//services:services_unittests",
