Index: dev/chrome/test/BUILD.gn
===================================================================
--- dev.orig/chrome/test/BUILD.gn
+++ dev/chrome/test/BUILD.gn
@@ -570,7 +570,11 @@ if (!is_android) {
           ]
           if (!use_ozone) {
             sources += [ "../browser/ui/libgtk2ui/select_file_dialog_interactive_uitest.cc" ]
-            configs += [ "//build/config/linux/gtk2:gtk2_internal_config" ]
+            if (use_gtk3) {
+                configs += [ "//build/config/linux/gtk3:gtk3_internal_config" ]
+            } else {
+                configs += [ "//build/config/linux/gtk2:gtk2_internal_config" ]
+            }
           }
         }
       }
Index: dev/remoting/test/BUILD.gn
===================================================================
--- dev.orig/remoting/test/BUILD.gn
+++ dev/remoting/test/BUILD.gn
@@ -196,7 +196,11 @@ if (enable_remoting_host && !is_android
     ]
 
     if (is_desktop_linux) {
-      deps += [ "//build/config/linux/gtk2" ]
+      if (use_gtk3) {
+        deps += [ "//build/config/linux/gtk3" ]
+      } else {
+        deps += [ "//build/config/linux/gtk2" ]
+      }
     }
   }
 }
Index: dev/remoting/host/it2me/BUILD.gn
===================================================================
--- dev.orig/remoting/host/it2me/BUILD.gn
+++ dev/remoting/host/it2me/BUILD.gn
@@ -210,7 +210,11 @@ if (!is_chromeos && !is_android && enabl
       }
 
       if (is_desktop_linux) {
-        deps += [ "//build/config/linux/gtk2" ]
+        if (use_gtk3) {
+          deps += [ "//build/config/linux/gtk3" ]
+        } else {
+          deps += [ "//build/config/linux/gtk2" ]
+        }
       }
     }
   }
Index: dev/remoting/host/BUILD.gn
===================================================================
--- dev.orig/remoting/host/BUILD.gn
+++ dev/remoting/host/BUILD.gn
@@ -413,7 +413,11 @@ static_library("host") {
       "//build/config/linux:xrandr",
     ]
     if (is_desktop_linux) {
-      deps += [ "//build/config/linux/gtk2" ]
+      if (use_gtk3) {
+        deps += [ "//build/config/linux/gtk3" ]
+      } else {
+        deps += [ "//build/config/linux/gtk2" ]
+      }
     }
   } else {
     sources -= [
@@ -1157,7 +1161,11 @@ if (enable_me2me_host) {
     }
 
     if (is_desktop_linux) {
-      deps += [ "//build/config/linux/gtk2" ]
+      if (use_gtk3) {
+        deps += [ "//build/config/linux/gtk3" ]
+      } else {
+        deps += [ "//build/config/linux/gtk2" ]
+      }
     }
     if ((is_linux && !is_chromeos) || is_mac) {
       libs = [ "pam" ]
Index: dev/build/config/linux/gtk3/BUILD.gn
===================================================================
--- dev.orig/build/config/linux/gtk3/BUILD.gn
+++ dev/build/config/linux/gtk3/BUILD.gn
@@ -30,6 +30,7 @@ group("gtk3") {
     "//remoting/host",
     "//remoting/host/it2me:remote_assistance_host",
     "//remoting/host:remoting_me2me_host_static",
+    "//remoting/test:it2me_standalone_host_main",
   ]
   public_configs = [ ":gtk3_internal_config" ]
 }
