Index: beta/third_party/blink/renderer/bindings/scripts/bind_gen/style_format.py
===================================================================
--- beta.orig/third_party/blink/renderer/bindings/scripts/bind_gen/style_format.py
+++ beta/third_party/blink/renderer/bindings/scripts/bind_gen/style_format.py
@@ -70,15 +70,10 @@ def gn_format(contents, filename=None):
 
 
 def _invoke_format_command(command_line, filename, contents):
-    proc = subprocess.Popen(
-        command_line, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
-    stdout_output, stderr_output = proc.communicate(input=contents)
-    exit_code = proc.wait()
-
     return StyleFormatResult(
-        stdout_output=stdout_output,
-        stderr_output=stderr_output,
-        exit_code=exit_code,
+        stdout_output=contents,
+        stderr_output="",
+        exit_code=0,
         filename=filename)
 
 
