Index: beta/chrome/test/BUILD.gn
===================================================================
--- beta.orig/chrome/test/BUILD.gn
+++ beta/chrome/test/BUILD.gn
@@ -438,7 +438,11 @@ if (!is_android) {
           ]
           if (!use_ozone) {
             sources += [ "../browser/ui/libgtk2ui/select_file_dialog_interactive_uitest.cc" ]
-            configs += [ "//build/config/linux/gtk2:gtk2_internal_config" ]
+            if (use_gtk3) {
+                configs += [ "//build/config/linux/gtk3:gtk3_internal_config" ]
+            } else {
+                configs += [ "//build/config/linux/gtk2:gtk2_internal_config" ]
+            }
           }
         }
       }
Index: beta/remoting/test/BUILD.gn
===================================================================
--- beta.orig/remoting/test/BUILD.gn
+++ beta/remoting/test/BUILD.gn
@@ -191,7 +191,11 @@ if (enable_remoting_host && !is_android
     ]
 
     if (is_desktop_linux) {
-      deps += [ "//build/config/linux/gtk2" ]
+      if (use_gtk3) {
+        deps += [ "//build/config/linux/gtk3" ]
+      } else {
+        deps += [ "//build/config/linux/gtk2" ]
+      }
     }
   }
 }
Index: beta/remoting/host/it2me/BUILD.gn
===================================================================
--- beta.orig/remoting/host/it2me/BUILD.gn
+++ beta/remoting/host/it2me/BUILD.gn
@@ -93,7 +93,11 @@ if (!is_chromeos && !is_android && enabl
       }
 
       if (is_desktop_linux) {
-        deps += [ "//build/config/linux/gtk2" ]
+        if (use_gtk3) {
+          deps += [ "//build/config/linux/gtk3" ]
+        } else {
+          deps += [ "//build/config/linux/gtk2" ]
+        }
       }
     }
   }
Index: beta/remoting/host/BUILD.gn
===================================================================
--- beta.orig/remoting/host/BUILD.gn
+++ beta/remoting/host/BUILD.gn
@@ -128,7 +128,11 @@ if (is_mac) {
         "//build/config/linux:xrandr",
       ]
       if (is_desktop_linux) {
-        deps += [ "//build/config/linux/gtk2" ]
+        if (use_gtk3) {
+          deps += [ "//build/config/linux/gtk3" ]
+        } else {
+          deps += [ "//build/config/linux/gtk2" ]
+        }
       }
     } else {
       sources -= [
@@ -805,7 +809,11 @@ if (is_mac) {
       }
 
       if (is_desktop_linux) {
-        deps += [ "//build/config/linux/gtk2" ]
+        if (use_gtk3) {
+          deps += [ "//build/config/linux/gtk3" ]
+        } else {
+          deps += [ "//build/config/linux/gtk2" ]
+        }
       }
       if ((is_linux && !is_chromeos) || is_mac) {
         libs = [ "pam" ]
Index: beta/build/config/linux/gtk3/BUILD.gn
===================================================================
--- beta.orig/build/config/linux/gtk3/BUILD.gn
+++ beta/build/config/linux/gtk3/BUILD.gn
@@ -30,6 +30,7 @@ group("gtk3") {
     "//remoting/host",
     "//remoting/host/it2me:remote_assistance_host",
     "//remoting/host:remoting_me2me_host_static",
+    "//remoting/test:it2me_standalone_host_main",
   ]
   public_configs = [ ":gtk3_internal_config" ]
 }
