Index: dev/chrome/test/chromedriver/BUILD.gn
===================================================================
--- dev.orig/chrome/test/chromedriver/BUILD.gn
+++ dev/chrome/test/chromedriver/BUILD.gn
@@ -4,9 +4,6 @@
 
 import("//build/config/python.gni")
 import("//build/config/ui.gni")
-if (is_linux) {
-  import("//build/linux/strip_binary.gni")
-}
 import("//build/util/lastchange.gni")
 import("//testing/test.gni")
 import("//third_party/ffmpeg/ffmpeg_options.gni")
@@ -368,10 +365,10 @@ source_set("lib") {
 
 if (is_linux) {
   # Linux
-  executable("chromedriver_server.unstripped") {
+  executable("chromedriver_server") {
     testonly = true
     sources = [ "server/chromedriver_server.cc" ]
-    output_name = "chromedriver.unstripped"
+    output_name = "chromedriver"
 
     deps = [
       ":automation_client_lib",
@@ -382,14 +379,6 @@ if (is_linux) {
       "//services/network/public/mojom",
     ]
   }
-
-  strip_binary("chromedriver_server") {
-    testonly = true
-    binary_input = "$root_out_dir/chromedriver.unstripped"
-    symbol_output = "$root_out_dir/chromedriver.debug"
-    stripped_binary_output = "$root_out_dir/chromedriver"
-    deps = [ ":chromedriver_server.unstripped" ]
-  }
 } else {
   # Non-Linux
   executable("chromedriver_server") {
