Index: dev/base/allocator/debugallocation_shim.cc
===================================================================
--- dev.orig/base/allocator/debugallocation_shim.cc
+++ dev/base/allocator/debugallocation_shim.cc
@@ -7,7 +7,7 @@
 // AFDO can mess with them. Better not to use AFDO there.  This is a
 // temporary hack. We will add a mechanism in the build system to
 // avoid using -fauto-profile for tcmalloc files.
-#if !defined(__clang__)
+#if !defined(__clang__) && __GNUC__ >= 5
 #pragma GCC optimize ("no-auto-profile")
 #endif
 
