Index: dev/chrome/chrome_exe.gypi
===================================================================
--- dev.orig/chrome/chrome_exe.gypi
+++ dev/chrome/chrome_exe.gypi
@@ -379,7 +379,13 @@
               'dependencies': [
                 '../native_client/src/trusted/service_runtime/linux/nacl_bootstrap.gyp:nacl_helper_bootstrap',
                 '../components/nacl.gyp:nacl_helper',
-                '../components/nacl_nonsfi.gyp:nacl_helper_nonsfi',
+              ],
+              'conditions': [
+                ['disable_nacl_untrusted==0', {
+                  'dependencies': [
+                    '../components/nacl_nonsfi.gyp:nacl_helper_nonsfi',
+                  ],
+                }],
               ],
             }],
           ],
Index: dev/chrome/chrome_tests.gypi
===================================================================
--- dev.orig/chrome/chrome_tests.gypi
+++ dev/chrome/chrome_tests.gypi
@@ -2395,7 +2395,13 @@
               'dependencies': [
                 '../native_client/src/trusted/service_runtime/linux/nacl_bootstrap.gyp:nacl_helper_bootstrap',
                 '../components/nacl.gyp:nacl_helper',
-                '../components/nacl_nonsfi.gyp:nacl_helper_nonsfi',
+              ],
+              'conditions': [
+                ['disable_nacl_untrusted==0', {
+                  'dependencies': [
+                    '../components/nacl_nonsfi.gyp:nacl_helper_nonsfi',
+                  ],
+                }],
               ],
             }],
             ['chromeos==0', {
