Index: dev/chrome/test/BUILD.gn
===================================================================
--- dev.orig/chrome/test/BUILD.gn
+++ dev/chrome/test/BUILD.gn
@@ -473,7 +473,11 @@ if (!is_android) {
           ]
           if (!use_ozone) {
             sources += [ "../browser/ui/libgtk2ui/select_file_dialog_interactive_uitest.cc" ]
-            configs += [ "//build/config/linux/gtk2:gtk2_internal_config" ]
+            if (use_gtk3) {
+                configs += [ "//build/config/linux/gtk3:gtk3_internal_config" ]
+            } else {
+                configs += [ "//build/config/linux/gtk2:gtk2_internal_config" ]
+            }
           }
         }
       }
Index: dev/remoting/test/BUILD.gn
===================================================================
--- dev.orig/remoting/test/BUILD.gn
+++ dev/remoting/test/BUILD.gn
@@ -192,7 +192,11 @@ if (enable_remoting_host && !is_android
     ]
 
     if (is_desktop_linux) {
-      deps += [ "//build/config/linux/gtk2" ]
+      if (use_gtk3) {
+        deps += [ "//build/config/linux/gtk3" ]
+      } else {
+        deps += [ "//build/config/linux/gtk2" ]
+      }
     }
   }
 }
