Index: beta/third_party/boringssl/BUILD.gn
===================================================================
--- beta.orig/third_party/boringssl/BUILD.gn
+++ beta/third_party/boringssl/BUILD.gn
@@ -69,7 +69,7 @@ if (is_win && !is_msan) {
 
     if ((current_cpu == "arm" || current_cpu == "arm64") && is_clang &&
         !is_ios) {
-      if (current_cpu == "arm" && arm_version != 6) {
+      if ((current_cpu == "arm" && arm_version != 6) || (current_cpu == "arm64" && is_linux)) {
         # TODO(hans) Enable integrated-as (crbug.com/124610).
         asmflags += [ "-fno-integrated-as" ]
       }
