Bug-Chromium: https://code.google.com/p/chromium/issues/detail?id=159349
                                                                       
Index: dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp
===================================================================
--- dev.saucy.orig/src/third_party/ffmpeg/ffmpeg.gyp	2013-10-19 14:55:03.295138953 -0400
+++ dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp	2013-10-19 14:55:03.275138210 -0400
@@ -125,6 +125,10 @@
             ],
             'yasm_output_path': '<(shared_generated_dir)/yasm'
           },
+          'include_dirs': [
+            '<(platform_config_root)',
+            '.',
+          ],
         },
       ] # targets
     }], # arch != arm
@@ -493,6 +497,9 @@
   'targets': [
     {
       'target_name': 'ffmpeg',
+      'include_dirs': [
+        '<(platform_config_root)',
+      ],
       'sources': [
         # Files needed for stub generation rules.
         '<@(sig_files)',
@@ -580,6 +587,7 @@
           ],
           'type': '<(component)',
           'include_dirs': [
+	    '<(platform_config_root)',
             '<(output_root)',
             '../..',  # The chromium 'src' directory.
             '<(platform_config_root)',
