Bug-Chromium: https://code.google.com/p/chromium/issues/detail?id=159349
                                                                       
Index: dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp
===================================================================
--- dev.saucy.orig/src/third_party/ffmpeg/ffmpeg.gyp	2013-11-02 21:29:52.968810409 -0400
+++ dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp	2013-11-02 21:29:52.964810266 -0400
@@ -132,6 +132,10 @@
             ],
             'yasm_output_path': '<(shared_generated_dir)/yasm'
           },
+          'include_dirs': [
+            '<(platform_config_root)',
+            '.',
+          ],
         },
       ] # targets
     }], # arch != arm
@@ -501,6 +505,9 @@
   'targets': [
     {
       'target_name': 'ffmpeg',
+      'include_dirs': [
+        '<(platform_config_root)',
+      ],
       'sources': [
         # Files needed for stub generation rules.
         '<@(sig_files)',
@@ -588,6 +595,7 @@
           ],
           'type': '<(component)',
           'include_dirs': [
+	    '<(platform_config_root)',
             '<(output_root)',
             '../..',  # The chromium 'src' directory.
             '<(platform_config_root)',
