Bug-Chromium: https://code.google.com/p/chromium/issues/detail?id=159349
                                                                       
Index: dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp
===================================================================
--- dev.saucy.orig/src/third_party/ffmpeg/ffmpeg.gyp	2013-07-12 13:12:54.296996147 -0500
+++ dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp	2013-07-12 13:12:54.288995868 -0500
@@ -431,6 +431,9 @@
   'targets': [
     {
       'target_name': 'ffmpeg',
+      'include_dirs': [
+        '<(platform_config_root)',
+      ],
       'sources': [
         # Files needed for stub generation rules.
         '<@(sig_files)',
@@ -518,6 +521,7 @@
           ],
           'type': '<(component)',
           'include_dirs': [
+	    '<(platform_config_root)',
             '<(output_root)',
             '../..',  # The chromium 'src' directory.
             '<(platform_config_root)',
@@ -640,6 +644,10 @@
         ],
         'yasm_output_path': '<(shared_generated_dir)/yasm'
       },
+      'include_dirs': [
+        '<(platform_config_root)',
+        '.',
+      ],
     },
   ],  # targets
 }
