Bug-Chromium: https://code.google.com/p/chromium/issues/detail?id=159349
                                                                       
Index: dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp
===================================================================
--- dev.saucy.orig/src/third_party/ffmpeg/ffmpeg.gyp	2013-09-12 17:40:18.121923793 -0400
+++ dev.saucy/src/third_party/ffmpeg/ffmpeg.gyp	2013-09-12 17:40:18.113923831 -0400
@@ -436,6 +436,9 @@
   'targets': [
     {
       'target_name': 'ffmpeg',
+      'include_dirs': [
+        '<(platform_config_root)',
+      ],
       'sources': [
         # Files needed for stub generation rules.
         '<@(sig_files)',
@@ -523,6 +526,7 @@
           ],
           'type': '<(component)',
           'include_dirs': [
+	    '<(platform_config_root)',
             '<(output_root)',
             '../..',  # The chromium 'src' directory.
             '<(platform_config_root)',
@@ -645,6 +649,10 @@
         ],
         'yasm_output_path': '<(shared_generated_dir)/yasm'
       },
+      'include_dirs': [
+        '<(platform_config_root)',
+        '.',
+      ],
     },
   ],  # targets
 }
