Bug-Chromium: https://code.google.com/p/chromium/issues/detail?id=159349
                                                                       
Index: dev.trusty/src/third_party/ffmpeg/ffmpeg.gyp
===================================================================
--- dev.trusty.orig/src/third_party/ffmpeg/ffmpeg.gyp	2014-03-29 23:18:19.950693596 -0400
+++ dev.trusty/src/third_party/ffmpeg/ffmpeg.gyp	2014-03-29 23:18:19.874690840 -0400
@@ -148,6 +148,10 @@
             ],
             'yasm_output_path': '<(shared_generated_dir)/yasm'
           },
+          'include_dirs': [
+            '<(platform_config_root)',
+            '.',
+          ],
         },
       ] # targets
     }], # arch != arm
@@ -516,6 +520,9 @@
   'targets': [
     {
       'target_name': 'ffmpeg',
+      'include_dirs': [
+        '<(platform_config_root)',
+      ],
       'sources': [
         # Files needed for stub generation rules.
         '<@(sig_files)',
@@ -603,6 +610,7 @@
           ],
           'type': '<(component)',
           'include_dirs': [
+	    '<(platform_config_root)',
             '<(output_root)',
             '../..',  # The chromium 'src' directory.
             '<(platform_config_root)',
Index: dev.trusty/src/third_party/icu/icu.gyp
===================================================================
--- dev.trusty.orig/src/third_party/icu/icu.gyp	2014-03-29 23:18:19.950693596 -0400
+++ dev.trusty/src/third_party/icu/icu.gyp	2014-03-29 23:18:19.946693451 -0400
@@ -34,8 +34,11 @@
          target_arch=="mipsel")', {
         'target_conditions': [
           ['_toolset=="host"', {
-            'cflags': [ '-m32' ],
-            'ldflags': [ '-m32' ],
+            'variables': {
+              'm32flag': '<!((echo | $(echo ${CXX_host:-$(which g++)}) -m32 -E - > /dev/null 2>&1) && echo "-m32" || true)',
+            },
+            'cflags': [ '<(m32flag)' ],
+            'ldflags': [ '<(m32flag)' ],
             'asflags': [ '-32' ],
             'xcode_settings': {
               'ARCHS': [ 'i386' ],
