Index: dev/tools/gn/bootstrap/bootstrap.py
===================================================================
--- dev.orig/tools/gn/bootstrap/bootstrap.py
+++ dev/tools/gn/bootstrap/bootstrap.py
@@ -192,7 +192,8 @@ def build_gn_with_ninja_manually(tempdir
           'ENABLE_LOCATION_SOURCE': 'false',
           'ENABLE_PROFILING': 'false',
           'CAN_UNWIND_WITH_FRAME_POINTERS': 'false',
-          'UNSAFE_DEVELOPER_BUILD': 'false'
+          'UNSAFE_DEVELOPER_BUILD': 'false',
+          'CAN_UNWIND_WITH_CFI_TABLE': 'false',
       })
 
   write_buildflag_header_manually(root_gen_dir,
