Index: beta/tools/gn/bootstrap/bootstrap.py
===================================================================
--- beta.orig/tools/gn/bootstrap/bootstrap.py
+++ beta/tools/gn/bootstrap/bootstrap.py
@@ -534,7 +534,6 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/trace_event/heap_profiler_allocation_context_tracker.cc',
       'base/trace_event/heap_profiler_allocation_register.cc',
       'base/trace_event/heap_profiler_event_filter.cc',
-      'base/trace_event/heap_profiler_event_writer.cc',
       'base/trace_event/heap_profiler_serialization_state.cc',
       'base/trace_event/heap_profiler_stack_frame_deduplicator.cc',
       'base/trace_event/heap_profiler_type_name_deduplicator.cc',
