Index: beta/media/gpu/vaapi/vaapi_mjpeg_decode_accelerator.cc
===================================================================
--- beta.orig/media/gpu/vaapi/vaapi_mjpeg_decode_accelerator.cc
+++ beta/media/gpu/vaapi/vaapi_mjpeg_decode_accelerator.cc
@@ -89,7 +89,7 @@ static bool VerifyDataSize(const VAImage
   const gfx::Size dimensions(base::strict_cast<int>(image->width),
                              base::strict_cast<int>(image->height));
   size_t min_size = 0;
-  if (image->format.fourcc == VA_FOURCC_I420) {
+  if (image->format.fourcc == VA_FOURCC('I', '4', '2', '0')) {
     min_size = VideoFrame::AllocationSize(PIXEL_FORMAT_I420, dimensions);
   } else if (image->format.fourcc == VA_FOURCC_NV12) {
     min_size = VideoFrame::AllocationSize(PIXEL_FORMAT_NV12, dimensions);
@@ -269,7 +269,7 @@ bool VaapiMjpegDecodeAccelerator::Output
   }
 
   switch (image->format.fourcc) {
-    case VA_FOURCC_I420:
+    case VA_FOURCC('I', '4', '2', '0'):
       DCHECK_EQ(image->num_planes, 3u);
       switch (video_frame->format()) {
         case PIXEL_FORMAT_I420:
Index: beta/media/gpu/vaapi/vaapi_video_encode_accelerator.h
===================================================================
--- beta.orig/media/gpu/vaapi/vaapi_video_encode_accelerator.h
+++ beta/media/gpu/vaapi/vaapi_video_encode_accelerator.h
@@ -51,7 +51,9 @@ class MEDIA_GPU_EXPORT VaapiVideoEncodeA
  private:
   class H264Accelerator;
   class VP8Accelerator;
+#if VA_CHECK_VERSION(0, 39, 2)
   class VP9Accelerator;
+#endif
 
   // Encoder state.
   enum State {
Index: beta/media/gpu/vaapi/vaapi_video_encode_accelerator.cc
===================================================================
--- beta.orig/media/gpu/vaapi/vaapi_video_encode_accelerator.cc
+++ beta/media/gpu/vaapi/vaapi_video_encode_accelerator.cc
@@ -222,6 +222,7 @@ class VaapiVideoEncodeAccelerator::VP8Ac
   VaapiVideoEncodeAccelerator* const vea_;
 };
 
+#if VA_CHECK_VERSION(0, 39, 2)
 class VaapiVideoEncodeAccelerator::VP9Accelerator
     : public VP9Encoder::Accelerator {
  public:
@@ -243,6 +244,7 @@ class VaapiVideoEncodeAccelerator::VP9Ac
  private:
   VaapiVideoEncodeAccelerator* const vea_;
 };
+#endif
 
 VaapiVideoEncodeAccelerator::VaapiVideoEncodeAccelerator()
     : output_buffer_byte_size_(0),
@@ -374,10 +376,12 @@ void VaapiVideoEncodeAccelerator::Initia
           std::make_unique<VP8Encoder>(std::make_unique<VP8Accelerator>(this));
       break;
 
+#if VA_CHECK_VERSION(0, 39, 2)
     case kCodecVP9:
       encoder_ =
           std::make_unique<VP9Encoder>(std::make_unique<VP9Accelerator>(this));
       break;
+#endif
 
     default:
       NOTREACHED() << "Unsupported codec type " << GetCodecName(codec);
@@ -1133,7 +1137,9 @@ bool VaapiVideoEncodeAccelerator::H264Ac
   rate_control_param.window_size = encode_params.cpb_window_size_ms;
   rate_control_param.initial_qp = pic_param.pic_init_qp;
   rate_control_param.min_qp = encode_params.scaling_settings.min_qp;
+#if VA_CHECK_VERSION(1, 1, 0)
   rate_control_param.max_qp = encode_params.scaling_settings.max_qp;
+#endif
   rate_control_param.rc_flags.bits.disable_frame_skip = true;
 
   VAEncMiscParameterFrameRate framerate_param = {};
@@ -1345,7 +1351,9 @@ bool VaapiVideoEncodeAccelerator::VP8Acc
   rate_control_param.window_size = encode_params.cpb_window_size_ms;
   rate_control_param.initial_qp = encode_params.initial_qp;
   rate_control_param.min_qp = encode_params.scaling_settings.min_qp;
+#if VA_CHECK_VERSION(1, 1, 0)
   rate_control_param.max_qp = encode_params.scaling_settings.max_qp;
+#endif
   rate_control_param.rc_flags.bits.disable_frame_skip = true;
 
   VAEncMiscParameterFrameRate framerate_param = {};
@@ -1388,6 +1396,7 @@ bool VaapiVideoEncodeAccelerator::VP8Acc
   return true;
 }
 
+#if VA_CHECK_VERSION(0, 39, 2)
 scoped_refptr<VP9Picture>
 VaapiVideoEncodeAccelerator::VP9Accelerator::GetPicture(
     AcceleratedVideoEncoder::EncodeJob* job) {
@@ -1520,5 +1529,6 @@ bool VaapiVideoEncodeAccelerator::VP9Acc
 
   return true;
 }
+#endif
 
 }  // namespace media
Index: beta/media/gpu/vaapi/vaapi_wrapper.cc
===================================================================
--- beta.orig/media/gpu/vaapi/vaapi_wrapper.cc
+++ beta/media/gpu/vaapi/vaapi_wrapper.cc
@@ -575,10 +575,16 @@ std::vector<VAEntrypoint> GetEntryPoints
   std::vector<VAEntrypoint>
       whitelisted_entrypoints[VaapiWrapper::CodecMode::kCodecModeMax] = {
           {VAEntrypointVLD},  // For kDecode.
-          {VAEntrypointEncSlice, VAEntrypointEncPicture,
-           VAEntrypointEncSliceLP},  // For kEncode.
-          {VAEntrypointEncSlice,
-           VAEntrypointEncSliceLP},  // For
+          {VAEntrypointEncSlice, VAEntrypointEncPicture
+#if VA_CHECK_VERSION(0, 39, 1)
+           , VAEntrypointEncSliceLP
+#endif
+          },  // For kEncode.
+          {VAEntrypointEncSlice
+#if VA_CHECK_VERSION(0, 39, 1)
+           , VAEntrypointEncSliceLP
+#endif
+                                 },  // For
                                      // kEncodeConstantQuantizationParameter.
           {VAEntrypointVideoProc}    // For kVideoProcess.
       };
@@ -1175,10 +1181,10 @@ bool VASupportedImageFormats::InitSuppor
     // Remove this workaround once b/128340287 is resolved.
     if (std::find_if(supported_formats_.cbegin(), supported_formats_.cend(),
                      [](const VAImageFormat& format) {
-                       return format.fourcc == VA_FOURCC_I420;
+                       return format.fourcc == VA_FOURCC('I', '4', '2', '0');
                      }) == supported_formats_.cend()) {
       VAImageFormat i420_format{};
-      i420_format.fourcc = VA_FOURCC_I420;
+      i420_format.fourcc = VA_FOURCC('I', '4', '2', '0');
       supported_formats_.push_back(i420_format);
     }
   }
@@ -1354,7 +1360,7 @@ bool VaapiWrapper::GetJpegDecodeSuitable
     // The VAAPI mesa state tracker only supports conversion from NV12 to YV12
     // and IYUV (synonym of I420).
     if (rt_format == VA_RT_FORMAT_YUV420) {
-      if (preferred_fourcc != VA_FOURCC_I420 &&
+      if (preferred_fourcc != VA_FOURCC('I', '4', '2', '0') &&
           preferred_fourcc != VA_FOURCC_YV12) {
         preferred_fourcc = VA_FOURCC_NV12;
       }
@@ -1384,18 +1390,18 @@ bool VaapiWrapper::GetJpegDecodeSuitable
     //
     if (preferred_fourcc == VA_FOURCC_422H ||
         preferred_fourcc == VA_FOURCC_P010) {
-      preferred_fourcc = VA_FOURCC_I420;
+      preferred_fourcc = VA_FOURCC('I', '4', '2', '0');
     }
   } else if (GetImplementationType() == VAImplementation::kIntelIHD) {
     // TODO(b/155939640): iHD v19.4 fails to allocate AYUV surfaces for the VPP
     // on gen 9.5.
     if (preferred_fourcc == VA_FOURCC_AYUV)
-      preferred_fourcc = VA_FOURCC_I420;
+      preferred_fourcc = VA_FOURCC('I', '4', '2', '0');
   }
 
   if (!VASupportedImageFormats::Get().IsImageFormatSupported(
           VAImageFormat{.fourcc = preferred_fourcc})) {
-    preferred_fourcc = VA_FOURCC_I420;
+    preferred_fourcc = VA_FOURCC('I', '4', '2', '0');
   }
 
   // After workarounds, assume the conversion is supported.
@@ -1613,7 +1619,11 @@ scoped_refptr<VASurface> VaapiWrapper::C
   }
   // We only have to pass the first file descriptor to a driver. A VA-API driver
   // shall create a VASurface from the single fd correctly.
+#if VA_CHECK_VERSION(1, 0, 0)
   uintptr_t fd = base::checked_cast<uintptr_t>(pixmap->GetDmaBufFd(0));
+#else
+  unsigned long fd = base::checked_cast<unsigned long>(pixmap->GetDmaBufFd(0));
+#endif
   va_attrib_extbuf.buffers = &fd;
   va_attrib_extbuf.num_buffers = 1u;
 
@@ -1663,6 +1673,7 @@ VaapiWrapper::ExportVASurfaceAsNativePix
     return nullptr;
   }
 
+#if VA_CHECK_VERSION(1, 0, 0)
   VADRMPRIMESurfaceDescriptor descriptor;
   {
     base::AutoLock auto_lock(*va_lock_);
@@ -1755,6 +1766,10 @@ VaapiWrapper::ExportVASurfaceAsNativePix
   exported_pixmap->pixmap = base::MakeRefCounted<gfx::NativePixmapDmaBuf>(
       scoped_va_surface.size(), buffer_format, std::move(handle));
   return exported_pixmap;
+#else
+  LOG(ERROR) << "Exporting a surface with FOURCC has been disabled";
+  return nullptr;
+#endif
 }
 
 bool VaapiWrapper::SyncSurface(VASurfaceID va_surface_id) {
@@ -2240,11 +2255,14 @@ bool VaapiWrapper::Initialize(CodecMode
     TryToSetVADisplayAttributeToLocalGPU();
 
   VAEntrypoint entrypoint = GetDefaultVaEntryPoint(mode, va_profile);
+
+#if VA_CHECK_VERSION(0, 39, 1)
   if (IsModeEncoding(mode) && IsLowPowerEncSupported(va_profile, mode) &&
       base::FeatureList::IsEnabled(kVaapiLowPowerEncoder)) {
     entrypoint = VAEntrypointEncSliceLP;
     DVLOG(2) << "Enable VA-API Low-Power Encode Entrypoint";
   }
+#endif
 
   base::AutoLock auto_lock(*va_lock_);
   std::vector<VAConfigAttrib> required_attribs;
@@ -2475,6 +2493,7 @@ void VaapiWrapper::TryToSetVADisplayAttr
     DVLOG(2) << "vaSetDisplayAttributes unsupported, ignoring by default.";
 }
 
+#if VA_CHECK_VERSION(0, 39, 1)
 // Check the support for low-power encode
 bool VaapiWrapper::IsLowPowerEncSupported(VAProfile va_profile,
                                           CodecMode mode) const {
@@ -2511,5 +2530,6 @@ bool VaapiWrapper::IsLowPowerEncSupporte
   }
   return true;
 }
+#endif
 
 }  // namespace media
Index: beta/media/gpu/vaapi/vaapi_wrapper.h
===================================================================
--- beta.orig/media/gpu/vaapi/vaapi_wrapper.h
+++ beta/media/gpu/vaapi/vaapi_wrapper.h
@@ -429,8 +429,10 @@ class MEDIA_GPU_EXPORT VaapiWrapper
   // Attempt to set render mode to "render to texture.". Failure is non-fatal.
   void TryToSetVADisplayAttributeToLocalGPU();
 
+#if VA_CHECK_VERSION(0, 39, 1)
   // Check low-power encode support for |profile| and |mode|.
   bool IsLowPowerEncSupported(VAProfile va_profile, CodecMode mode) const;
+#endif
 
   const CodecMode mode_;
 
Index: beta/media/gpu/chromeos/fourcc.cc
===================================================================
--- beta.orig/media/gpu/chromeos/fourcc.cc
+++ beta/media/gpu/chromeos/fourcc.cc
@@ -210,11 +210,11 @@ uint32_t Fourcc::ToV4L2PixFmt() const {
 // static
 base::Optional<Fourcc> Fourcc::FromVAFourCC(uint32_t va_fourcc) {
   switch (va_fourcc) {
-    case VA_FOURCC_I420:
+    case VA_FOURCC('I', '4', '2', '0'):
       return Fourcc(YU12);
     case VA_FOURCC_NV12:
       return Fourcc(NV12);
-    case VA_FOURCC_NV21:
+    case VA_FOURCC('N', 'V', '2', '1'):
       return Fourcc(NV21);
     case VA_FOURCC_YV12:
       return Fourcc(YV12);
@@ -238,11 +238,11 @@ base::Optional<Fourcc> Fourcc::FromVAFou
 base::Optional<uint32_t> Fourcc::ToVAFourCC() const {
   switch (value_) {
     case YU12:
-      return VA_FOURCC_I420;
+      return VA_FOURCC('I', '4', '2', '0');
     case NV12:
       return VA_FOURCC_NV12;
     case NV21:
-      return VA_FOURCC_NV21;
+      return VA_FOURCC('N', 'V', '2', '1');
     case YV12:
       return VA_FOURCC_YV12;
     case YUYV:
Index: beta/ui/gfx/x/gen_xproto.py
===================================================================
--- beta.orig/ui/gfx/x/gen_xproto.py
+++ beta/ui/gfx/x/gen_xproto.py
@@ -356,7 +356,7 @@ class GenXproto:
             return 'Request'
         elif t.is_reply:
             return 'Reply'
-        elif t.is_event:
+        elif isinstance(t, self.xcbgen.xtypes.Event):
             return 'Event'
         return ''
 
@@ -525,7 +525,8 @@ class GenXproto:
             self.copy_primitive(name)
 
     def declare_case(self, case):
-        assert case.type.is_case != case.type.is_bitcase
+        if hasattr(case.type, 'is_case'):
+            assert case.type.is_case != case.type.is_bitcase
 
         with (Indent(self, 'struct {', '} %s;' % safe_name(case.field_name))
               if case.field_name else NullContext()):
@@ -533,7 +534,10 @@ class GenXproto:
                 self.declare_field(case_field)
 
     def copy_case(self, case, switch_var):
-        op = 'CaseEq' if case.type.is_case else 'BitAnd'
+        if hasattr(case.type, "is_case"):
+            op = 'CaseEq' if case.type.is_case else 'BitAnd'
+        else:
+            op = 'BitAnd'
         condition = ' || '.join([
             '%s(%s, %s)' % (op, switch_var, self.expr(expr))
             for expr in case.type.expr
@@ -851,7 +855,7 @@ class GenXproto:
         for field in fields.values():
             field.parent = t
             field.for_list = None
-            if field.type.is_switch or field.type.is_case_or_bitcase:
+            if field.type.is_switch or (field.type.is_case_or_bitcase if hasattr(field.type, 'is_case_or_bitcase') else field.type.is_bitcase):
                 self.resolve_type(field.type, field.field_type)
             elif field.type.is_list:
                 self.resolve_type(field.type.member, field.type.member.name)
@@ -887,7 +891,7 @@ class GenXproto:
             l = list(self.types[t])
             # For some reason, FDs always have distint types so they appear
             # duplicated in the set.  If the set contains only FDs, then bail.
-            if all(x.is_fd for x in l):
+            if all(hasattr(x, 'is_fd') and x.is_fd for x in l):
                 continue
             if len(l) == 1:
                 continue
Index: beta/ui/gtk/gtk_util.cc
===================================================================
--- beta.orig/ui/gtk/gtk_util.cc
+++ beta/ui/gtk/gtk_util.cc
@@ -32,7 +32,9 @@
 #include "ui/ozone/public/ozone_platform.h"
 #include "ui/views/linux_ui/linux_ui.h"
 
+#if GTK_CHECK_VERSION(3, 20, 0)
 WEAK_GTK_FN(gtk_widget_path_iter_set_object_name);
+#endif
 WEAK_GTK_FN(gtk_widget_path_iter_set_state);
 
 namespace {
@@ -342,8 +344,10 @@ ScopedStyleContext AppendCssNodeToStyleC
           break;
         case CSS_OBJECT_NAME:
           if (GtkCheckVersion(3, 20)) {
+#if GTK_CHECK_VERSION(3, 20, 0)
             DCHECK(gtk_widget_path_iter_set_object_name);
             gtk_widget_path_iter_set_object_name(path, -1, t.token().c_str());
+#endif
           } else {
             gtk_widget_path_iter_add_class(path, -1, t.token().c_str());
           }
@@ -353,8 +357,10 @@ ScopedStyleContext AppendCssNodeToStyleC
           DCHECK(type);
           gtk_widget_path_append_type(path, type);
           if (GtkCheckVersion(3, 20) && t.token() == "GtkLabel") {
+#if GTK_CHECK_VERSION(3, 20, 0)
             DCHECK(gtk_widget_path_iter_set_object_name);
             gtk_widget_path_iter_set_object_name(path, -1, "label");
+#endif
           }
           break;
         }
