Index: dev/third_party/webrtc/modules/audio_processing/aec3/reverb_model_estimator.cc
===================================================================
--- dev.orig/third_party/webrtc/modules/audio_processing/aec3/reverb_model_estimator.cc
+++ dev/third_party/webrtc/modules/audio_processing/aec3/reverb_model_estimator.cc
@@ -14,6 +14,7 @@
 #include <array>
 #include <memory>
 #include <numeric>
+#include <math.h>
 
 #include "api/array_view.h"
 #include "api/audio/echo_canceller3_config.h"
