Index: dev/tools/gn/bootstrap/bootstrap.py
===================================================================
--- dev.orig/tools/gn/bootstrap/bootstrap.py
+++ dev/tools/gn/bootstrap/bootstrap.py
@@ -526,6 +526,7 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/trace_event/process_memory_totals.cc',
       'base/trace_event/trace_buffer.cc',
       'base/trace_event/trace_config.cc',
+      'base/trace_event/trace_config_category_filter.cc',
       'base/trace_event/trace_event_argument.cc',
       'base/trace_event/trace_event_filter.cc',
       'base/trace_event/trace_event_impl.cc',
