Index: dev/components/language/core/browser/language_prefs.cc
===================================================================
--- dev.orig/components/language/core/browser/language_prefs.cc
+++ dev/components/language/core/browser/language_prefs.cc
@@ -85,7 +85,7 @@ void LanguagePrefs::ResetEmptyFluentLang
 }
 
 base::Value LanguagePrefs::GetDefaultFluentLanguages() {
-  std::set<const std::string> languages;
+  std::set<std::string> languages;
 #if defined(OS_CHROMEOS)
   // Preferred languages.
   std::string language = language::kFallbackInputMethodLocale;
