Index: dev/cc/layers/picture_layer_impl.cc
===================================================================
--- dev.orig/cc/layers/picture_layer_impl.cc
+++ dev/cc/layers/picture_layer_impl.cc
@@ -738,7 +738,7 @@ void PictureLayerImpl::UpdateViewportRec
           0,                                                   // left
           0,                                                   // top,
           0,                                                   // right,
-          std::ceilf(-total_controls_height * hidden_ratio));  // bottom
+          std::ceil(-total_controls_height * hidden_ratio));  // bottom
     }
   }
 }
