Index: dev.wily/native_client/build/untrusted.gypi
===================================================================
--- dev.wily.orig/native_client/build/untrusted.gypi
+++ dev.wily/native_client/build/untrusted.gypi
@@ -338,7 +338,7 @@
                     '>@(extra_deps)',
                     '>@(extra_deps_newlib64)',
                     '^(source_list_newlib64)',
-                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_x86_newlib/nacl_x86_newlib.json',
+                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                     '<(PRODUCT_DIR)/tls_edit<(EXECUTABLE_SUFFIX)',
                  ],
                  'outputs': ['>(out_newlib64)'],
@@ -382,7 +382,7 @@
                     '>@(extra_deps)',
                     '>@(extra_deps_newlib64)',
                     '^(source_list_newlib64)',
-                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_x86_newlib/nacl_x86_newlib.json',
+                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                  ],
                  'outputs': ['>(out_newlib64)'],
                  'action': [
@@ -506,7 +506,7 @@
                     '>@(extra_deps)',
                     '>@(extra_deps_newlib32)',
                     '^(source_list_newlib32)',
-                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_x86_newlib/nacl_x86_newlib.json',
+                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                     '<(PRODUCT_DIR)/tls_edit<(EXECUTABLE_SUFFIX)',
                  ],
                  'outputs': ['>(out_newlib32)'],
@@ -550,7 +550,7 @@
                     '>@(extra_deps)',
                     '>@(extra_deps_newlib32)',
                     '^(source_list_newlib32)',
-                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_x86_newlib/nacl_x86_newlib.json',
+                    '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                  ],
                  'outputs': ['>(out_newlib32)'],
                  'action': [
@@ -598,7 +598,7 @@
                   '>@(extra_deps)',
                   '>@(extra_deps_newlib32_nonsfi)',
                   '^(source_list_newlib32_nonsfi)',
-                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_x86_newlib/nacl_x86_newlib.json',
+                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                   '>(tc_lib_dir_nonsfi_helper32)/libnacl_sys_private.a',
                 ],
                 'outputs': ['>(out_newlib32_nonsfi)'],
@@ -656,7 +656,7 @@
                   '>@(extra_deps)',
                   '>@(extra_deps_newlib32_nonsfi)',
                   '^(source_list_newlib32_nonsfi)',
-                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_x86_newlib/nacl_x86_newlib.json',
+                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                 ],
                 'outputs': ['>(out_newlib32_nonsfi)'],
                 'action': [
@@ -1032,7 +1032,7 @@
                   '>@(extra_deps)',
                   '>@(extra_deps_newlib_arm_nonsfi)',
                   '^(source_list_newlib_arm_nonsfi)',
-                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_arm_newlib/nacl_arm_newlib.json',
+                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                   '>(tc_lib_dir_nonsfi_helper_arm)/libnacl_sys_private.a',
                 ],
                 'outputs': ['>(out_newlib_arm_nonsfi)'],
@@ -1088,7 +1088,7 @@
                   '>@(extra_deps)',
                   '>@(extra_deps_newlib_arm_nonsfi)',
                   '^(source_list_newlib_arm_nonsfi)',
-                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/nacl_arm_newlib/nacl_arm_newlib.json',
+                  '<(DEPTH)/native_client/toolchain/<(TOOLCHAIN_OS)_x86/pnacl_newlib/pnacl_newlib.json',
                 ],
                 'outputs': ['>(out_newlib_arm_nonsfi)'],
                 'action': [
