Index: dev/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
===================================================================
--- dev.orig/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
+++ dev/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
@@ -20,7 +20,8 @@
 #include <sys/cdefs.h>
 
 // https://sourceware.org/bugzilla/show_bug.cgi?id=22433
-#if !defined(PTRACE_GET_THREAD_AREA) && defined(__GLIBC__)
+#if !defined(PTRACE_GET_THREAD_AREA) && !defined(PT_GET_THREAD_AREA) && \
+    defined(__GLIBC__)
 #if defined(__i386__) || defined(__x86_64__)
 static constexpr __ptrace_request PTRACE_GET_THREAD_AREA =
     static_cast<__ptrace_request>(25);
@@ -30,7 +31,7 @@ static constexpr __ptrace_request PTRACE
     static_cast<__ptrace_request>(22);
 #define PTRACE_GET_THREAD_AREA PTRACE_GET_THREAD_AREA
 #endif
-#endif  // !PTRACE_GET_THREAD_AREA && defined(__GLIBC__)
+#endif  // !PTRACE_GET_THREAD_AREA && !PT_GET_THREAD_AREA && defined(__GLIBC__)
 
 // https://sourceware.org/bugzilla/show_bug.cgi?id=22433
 #if !defined(PTRACE_GETVFPREGS) && \
