Index: dev/tools/gn/bootstrap/bootstrap.py
===================================================================
--- dev.orig/tools/gn/bootstrap/bootstrap.py
+++ dev/tools/gn/bootstrap/bootstrap.py
@@ -503,6 +503,8 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/threading/thread_restrictions.cc',
       'base/threading/thread_task_runner_handle.cc',
       'base/threading/worker_pool.cc',
+      'base/time/default_tick_clock.cc',
+      'base/time/tick_clock.cc',
       'base/time/time.cc',
       'base/timer/elapsed_timer.cc',
       'base/timer/timer.cc',
