Index: dev/sandbox/policy/linux/bpf_audio_policy_linux.cc
===================================================================
--- dev.orig/sandbox/policy/linux/bpf_audio_policy_linux.cc
+++ dev/sandbox/policy/linux/bpf_audio_policy_linux.cc
@@ -75,6 +75,9 @@ ResultExpr AudioProcessPolicy::EvaluateS
 #if defined(__NR_uname)
     case __NR_uname:
 #endif
+#if defined(__NR_sched_getaffinity)
+    case __NR_sched_getaffinity:
+#endif
       return Allow();
 #if defined(__NR_futex)
     case __NR_futex:
