Index: dev/tools/gn/build/gen.py
===================================================================
--- dev.orig/tools/gn/build/gen.py
+++ dev/tools/gn/build/gen.py
@@ -334,7 +334,7 @@ def WriteGNNinja(path, platform, host, o
       ldflags.append('-maix64')
 
     if platform.is_posix():
-      ldflags.append('-lpthread')
+      ldflags.append('-pthread')
 
     if options.use_lto:
       cflags.extend(['-flto', '-fwhole-program-vtables'])
