Index: beta/third_party/webrtc/webrtc.gni
===================================================================
--- beta.orig/third_party/webrtc/webrtc.gni
+++ beta/third_party/webrtc/webrtc.gni
@@ -117,6 +117,9 @@ declare_args() {
   # Set this to link PipeWire directly instead of using the dlopen.
   rtc_link_pipewire = false
 
+  # Use libpipewire-0.3 instead of libpipewire-0.2, for newer distros.
+  rtc_use_new_pipewire_version = false
+
   # Enable to use the Mozilla internal settings.
   build_with_mozilla = false
 
Index: beta/third_party/webrtc/modules/desktop_capture/BUILD.gn
===================================================================
--- beta.orig/third_party/webrtc/modules/desktop_capture/BUILD.gn
+++ beta/third_party/webrtc/modules/desktop_capture/BUILD.gn
@@ -201,8 +201,14 @@ if (is_linux || is_chromeos) {
     }
 
     if (rtc_link_pipewire) {
-      pkg_config("pipewire") {
-        packages = [ "libpipewire-0.2" ]
+      if (rtc_use_new_pipewire_version) {
+        pkg_config("pipewire") {
+          packages = [ "libpipewire-0.3" ]
+        }
+      } else {
+        pkg_config("pipewire") {
+          packages = [ "libpipewire-0.2" ]
+        }
       }
     } else {
       # When libpipewire is not directly linked, use stubs to allow for dlopening of
