Index: dev/v8/src/objects/js-display-names.cc
===================================================================
--- dev.orig/v8/src/objects/js-display-names.cc
+++ dev/v8/src/objects/js-display-names.cc
@@ -129,9 +129,7 @@ class LanguageNames : public LocaleDispl
     // 1.b If IsStructurallyValidLanguageTag(code) is false, throw a RangeError
     // exception.
     icu::Locale l =
-        icu::Locale(icu::Locale::forLanguageTag(code, status).getBaseName());
-    // 1.c Set code to CanonicalizeUnicodeLocaleId(code).
-    l.canonicalize(status);
+        icu::Locale::createCanonical(icu::Locale::forLanguageTag(code, status).getBaseName());
     std::string checked = l.toLanguageTag<std::string>(status);
 
     if (U_FAILURE(status)) {
