Index: dev/sandbox/policy/linux/bpf_audio_policy_linux.cc
===================================================================
--- dev.orig/sandbox/policy/linux/bpf_audio_policy_linux.cc
+++ dev/sandbox/policy/linux/bpf_audio_policy_linux.cc
@@ -83,14 +83,8 @@ ResultExpr AudioProcessPolicy::EvaluateS
     case __NR_futex_time64:
 #endif
     {
-      const Arg<int> op(1);
 #if defined(USE_PULSEAUDIO)
-      return Switch(op & ~FUTEX_PRIVATE_FLAG)
-          .SANDBOX_BPF_DSL_CASES(
-              (FUTEX_CMP_REQUEUE, FUTEX_LOCK_PI, FUTEX_UNLOCK_PI, FUTEX_WAIT,
-               FUTEX_WAIT_BITSET, FUTEX_WAKE),
-              Allow())
-          .Default(Error(EPERM));
+      return Allow();
 #else
       return RestrictFutex();
 #endif
