Index: dev/media/gpu/vaapi/vaapi_mjpeg_decode_accelerator.cc
===================================================================
--- dev.orig/media/gpu/vaapi/vaapi_mjpeg_decode_accelerator.cc
+++ dev/media/gpu/vaapi/vaapi_mjpeg_decode_accelerator.cc
@@ -89,7 +89,7 @@ static bool VerifyDataSize(const VAImage
   const gfx::Size dimensions(base::strict_cast<int>(image->width),
                              base::strict_cast<int>(image->height));
   size_t min_size = 0;
-  if (image->format.fourcc == VA_FOURCC_I420) {
+  if (image->format.fourcc == VA_FOURCC('I', '4', '2', '0')) {
     min_size = VideoFrame::AllocationSize(PIXEL_FORMAT_I420, dimensions);
   } else if (image->format.fourcc == VA_FOURCC_NV12) {
     min_size = VideoFrame::AllocationSize(PIXEL_FORMAT_NV12, dimensions);
@@ -269,7 +269,7 @@ bool VaapiMjpegDecodeAccelerator::Output
   }
 
   switch (image->format.fourcc) {
-    case VA_FOURCC_I420:
+    case VA_FOURCC('I', '4', '2', '0'):
       DCHECK_EQ(image->num_planes, 3u);
       switch (video_frame->format()) {
         case PIXEL_FORMAT_I420:
Index: dev/media/gpu/vaapi/vaapi_video_encode_accelerator.h
===================================================================
--- dev.orig/media/gpu/vaapi/vaapi_video_encode_accelerator.h
+++ dev/media/gpu/vaapi/vaapi_video_encode_accelerator.h
@@ -52,7 +52,9 @@ class MEDIA_GPU_EXPORT VaapiVideoEncodeA
   friend class VaapiVideoEncodeAcceleratorTest;
   class H264Accelerator;
   class VP8Accelerator;
+#if VA_CHECK_VERSION(0, 39, 2)
   class VP9Accelerator;
+#endif
 
   // Encoder state.
   enum State {
Index: dev/media/gpu/vaapi/vaapi_video_encode_accelerator.cc
===================================================================
--- dev.orig/media/gpu/vaapi/vaapi_video_encode_accelerator.cc
+++ dev/media/gpu/vaapi/vaapi_video_encode_accelerator.cc
@@ -219,6 +219,7 @@ class VaapiVideoEncodeAccelerator::VP8Ac
   VaapiVideoEncodeAccelerator* const vea_;
 };
 
+#if VA_CHECK_VERSION(0, 39, 2)
 class VaapiVideoEncodeAccelerator::VP9Accelerator
     : public VP9Encoder::Accelerator {
  public:
@@ -240,6 +241,7 @@ class VaapiVideoEncodeAccelerator::VP9Ac
  private:
   VaapiVideoEncodeAccelerator* const vea_;
 };
+#endif
 
 VaapiVideoEncodeAccelerator::VaapiVideoEncodeAccelerator()
     : output_buffer_byte_size_(0),
@@ -384,6 +386,7 @@ void VaapiVideoEncodeAccelerator::Initia
       DCHECK_EQ(ave_config.bitrate_control,
                 AcceleratedVideoEncoder::BitrateControl::kConstantBitrate);
       break;
+#if VA_CHECK_VERSION(0, 39, 2)
     case kCodecVP9:
       if (!IsConfiguredForTesting()) {
         encoder_ = std::make_unique<VP9Encoder>(
@@ -392,6 +395,7 @@ void VaapiVideoEncodeAccelerator::Initia
       ave_config.bitrate_control = AcceleratedVideoEncoder::BitrateControl::
           kConstantQuantizationParameter;
       break;
+#endif
     default:
       NOTREACHED() << "Unsupported codec type " << GetCodecName(codec);
       return;
@@ -1167,7 +1171,9 @@ bool VaapiVideoEncodeAccelerator::H264Ac
   rate_control_param.window_size = encode_params.cpb_window_size_ms;
   rate_control_param.initial_qp = pic_param.pic_init_qp;
   rate_control_param.min_qp = encode_params.scaling_settings.min_qp;
+#if VA_CHECK_VERSION(1, 1, 0)
   rate_control_param.max_qp = encode_params.scaling_settings.max_qp;
+#endif
   rate_control_param.rc_flags.bits.disable_frame_skip = true;
 
   VAEncMiscParameterFrameRate framerate_param = {};
@@ -1379,7 +1385,9 @@ bool VaapiVideoEncodeAccelerator::VP8Acc
   rate_control_param.window_size = encode_params.cpb_window_size_ms;
   rate_control_param.initial_qp = encode_params.initial_qp;
   rate_control_param.min_qp = encode_params.scaling_settings.min_qp;
+#if VA_CHECK_VERSION(1, 1, 0)
   rate_control_param.max_qp = encode_params.scaling_settings.max_qp;
+#endif
   rate_control_param.rc_flags.bits.disable_frame_skip = true;
 
   VAEncMiscParameterFrameRate framerate_param = {};
@@ -1422,6 +1430,7 @@ bool VaapiVideoEncodeAccelerator::VP8Acc
   return true;
 }
 
+#if VA_CHECK_VERSION(0, 39, 2)
 scoped_refptr<VP9Picture>
 VaapiVideoEncodeAccelerator::VP9Accelerator::GetPicture(
     AcceleratedVideoEncoder::EncodeJob* job) {
@@ -1563,5 +1572,6 @@ bool VaapiVideoEncodeAccelerator::VP9Acc
 
   return true;
 }
+#endif
 
 }  // namespace media
Index: dev/media/gpu/vaapi/vaapi_wrapper.cc
===================================================================
--- dev.orig/media/gpu/vaapi/vaapi_wrapper.cc
+++ dev/media/gpu/vaapi/vaapi_wrapper.cc
@@ -10,8 +10,10 @@
 #include <va/va.h>
 #include <va/va_drm.h>
 #include <va/va_drmcommon.h>
-#include <va/va_str.h>
 #include <va/va_version.h>
+#if VA_CHECK_VERSION(1, 0, 0)
+#include <va/va_str.h>
+#endif
 
 #include <algorithm>
 #include <string>
@@ -312,6 +314,62 @@ bool IsBlockedDriver(VaapiWrapper::Codec
   return false;
 }
 
+#if !VA_CHECK_VERSION(1, 0, 0)
+#define TOSTR(enumCase) case enumCase: return #enumCase
+
+const char *vaProfileStr(VAProfile profile)
+{
+  switch (profile) {
+    TOSTR(VAProfileNone);
+    TOSTR(VAProfileMPEG2Simple);
+    TOSTR(VAProfileMPEG2Main);
+    TOSTR(VAProfileMPEG4Simple);
+    TOSTR(VAProfileMPEG4AdvancedSimple);
+    TOSTR(VAProfileMPEG4Main);
+    TOSTR(VAProfileH264Baseline);
+    TOSTR(VAProfileH264Main);
+    TOSTR(VAProfileH264High);
+    TOSTR(VAProfileVC1Simple);
+    TOSTR(VAProfileVC1Main);
+    TOSTR(VAProfileVC1Advanced);
+    TOSTR(VAProfileH263Baseline);
+    TOSTR(VAProfileH264ConstrainedBaseline);
+    TOSTR(VAProfileJPEGBaseline);
+    TOSTR(VAProfileVP8Version0_3);
+    TOSTR(VAProfileH264MultiviewHigh);
+    TOSTR(VAProfileH264StereoHigh);
+    TOSTR(VAProfileHEVCMain);
+    TOSTR(VAProfileHEVCMain10);
+    TOSTR(VAProfileVP9Profile0);
+    TOSTR(VAProfileVP9Profile1);
+    TOSTR(VAProfileVP9Profile2);
+    TOSTR(VAProfileVP9Profile3);
+  }
+  return "<unknown profile>";
+}
+
+
+const char *vaEntrypointStr(VAEntrypoint entrypoint)
+{
+  switch (entrypoint) {
+    TOSTR(VAEntrypointVLD);
+    TOSTR(VAEntrypointIZZ);
+    TOSTR(VAEntrypointIDCT);
+    TOSTR(VAEntrypointMoComp);
+    TOSTR(VAEntrypointDeblocking);
+    TOSTR(VAEntrypointEncSlice);
+    TOSTR(VAEntrypointEncPicture);
+#if VA_CHECK_VERSION(0, 39, 1)
+    TOSTR(VAEntrypointEncSliceLP);
+#endif
+    TOSTR(VAEntrypointVideoProc);
+  }
+  return "<unknown entrypoint>";
+}
+
+#undef TOSTR
+#endif
+
 // This class is a wrapper around its |va_display_| (and its associated
 // |va_lock_|) to guarantee mutual exclusion and singleton behaviour.
 class VADisplayState {
@@ -566,10 +624,16 @@ std::vector<VAEntrypoint> GetEntryPoints
 
   const std::vector<VAEntrypoint> kAllowedEntryPoints[] = {
       {VAEntrypointVLD},  // kDecode.
-      {VAEntrypointEncSlice, VAEntrypointEncPicture,
-       VAEntrypointEncSliceLP},  // kEncode.
-      {VAEntrypointEncSlice,
-       VAEntrypointEncSliceLP},  // kEncodeConstantQuantizationParameter.
+      {VAEntrypointEncSlice, VAEntrypointEncPicture
+#if VA_CHECK_VERSION(0, 39, 1)
+       , VAEntrypointEncSliceLP
+#endif
+      },  // kEncode.
+      {VAEntrypointEncSlice
+#if VA_CHECK_VERSION(0, 39, 1)
+       , VAEntrypointEncSliceLP
+#endif
+      },  // kEncodeConstantQuantizationParameter.
       {VAEntrypointVideoProc}    // kVideoProcess.
   };
   static_assert(base::size(kAllowedEntryPoints) == VaapiWrapper::kCodecModeMax,
@@ -1065,16 +1129,17 @@ bool VASupportedImageFormats::InitSuppor
     // Remove this workaround once b/128340287 is resolved.
     if (std::find_if(supported_formats_.cbegin(), supported_formats_.cend(),
                      [](const VAImageFormat& format) {
-                       return format.fourcc == VA_FOURCC_I420;
+                       return format.fourcc == VA_FOURCC('I', '4', '2', '0');
                      }) == supported_formats_.cend()) {
       VAImageFormat i420_format{};
-      i420_format.fourcc = VA_FOURCC_I420;
+      i420_format.fourcc = VA_FOURCC('I', '4', '2', '0');
       supported_formats_.push_back(i420_format);
     }
   }
   return true;
 }
 
+#if VA_CHECK_VERSION(0, 39, 1)
 bool IsLowPowerEncSupported(VAProfile va_profile) {
   constexpr VAProfile kSupportedLowPowerEncodeProfiles[] = {
       VAProfileH264ConstrainedBaseline,
@@ -1098,6 +1163,7 @@ bool IsLowPowerEncSupported(VAProfile va
   }
   return false;
 }
+#endif
 
 }  // namespace
 
@@ -1268,7 +1334,7 @@ bool VaapiWrapper::GetJpegDecodeSuitable
     // The VAAPI mesa state tracker only supports conversion from NV12 to YV12
     // and IYUV (synonym of I420).
     if (rt_format == VA_RT_FORMAT_YUV420) {
-      if (preferred_fourcc != VA_FOURCC_I420 &&
+      if (preferred_fourcc != VA_FOURCC('I', '4', '2', '0') &&
           preferred_fourcc != VA_FOURCC_YV12) {
         preferred_fourcc = VA_FOURCC_NV12;
       }
@@ -1298,7 +1364,7 @@ bool VaapiWrapper::GetJpegDecodeSuitable
     //
     if (preferred_fourcc == VA_FOURCC_422H ||
         preferred_fourcc == VA_FOURCC_P010) {
-      preferred_fourcc = VA_FOURCC_I420;
+      preferred_fourcc = VA_FOURCC('I', '4', '2', '0');
     }
   } else if (GetImplementationType() == VAImplementation::kIntelIHD) {
     // TODO(b/155939640): iHD v19.4 fails to allocate AYUV surfaces for the VPP
@@ -1306,16 +1372,16 @@ bool VaapiWrapper::GetJpegDecodeSuitable
     // (b/159896972): iHD v20.1.1 cannot create Y216 and Y416 images from a
     // decoded JPEG on gen 12. It is also failing to support Y800 format.
     if (preferred_fourcc == VA_FOURCC_AYUV ||
-        preferred_fourcc == VA_FOURCC_Y216 ||
-        preferred_fourcc == VA_FOURCC_Y416 ||
+        preferred_fourcc == VA_FOURCC('Y', '2', '1', '6') ||
+        preferred_fourcc == VA_FOURCC('Y', '4', '1', '6') ||
         preferred_fourcc == VA_FOURCC_Y800) {
-      preferred_fourcc = VA_FOURCC_I420;
+      preferred_fourcc = VA_FOURCC('I', '4', '2', '0');
     }
   }
 
   if (!VASupportedImageFormats::Get().IsImageFormatSupported(
           VAImageFormat{.fourcc = preferred_fourcc})) {
-    preferred_fourcc = VA_FOURCC_I420;
+    preferred_fourcc = VA_FOURCC('I', '4', '2', '0');
   }
 
   // After workarounds, assume the conversion is supported.
@@ -1412,8 +1478,10 @@ VAEntrypoint VaapiWrapper::GetDefaultVaE
       if (profile == VAProfileJPEGBaseline)
         return VAEntrypointEncPicture;
       DCHECK(IsModeEncoding(mode));
+#if VA_CHECK_VERSION(0, 39, 1)
       if (IsLowPowerEncSupported(profile))
         return VAEntrypointEncSliceLP;
+#endif
       return VAEntrypointEncSlice;
     case VaapiWrapper::kVideoProcess:
       return VAEntrypointVideoProc;
@@ -1545,7 +1613,11 @@ scoped_refptr<VASurface> VaapiWrapper::C
   }
   // We only have to pass the first file descriptor to a driver. A VA-API driver
   // shall create a VASurface from the single fd correctly.
+#if VA_CHECK_VERSION(1, 0, 0)
   uintptr_t fd = base::checked_cast<uintptr_t>(pixmap->GetDmaBufFd(0));
+#else
+  unsigned long fd = base::checked_cast<unsigned long>(pixmap->GetDmaBufFd(0));
+#endif
   va_attrib_extbuf.buffers = &fd;
   va_attrib_extbuf.num_buffers = 1u;
 
@@ -1595,6 +1667,7 @@ VaapiWrapper::ExportVASurfaceAsNativePix
     return nullptr;
   }
 
+#if VA_CHECK_VERSION(1, 0, 0)
   VADRMPRIMESurfaceDescriptor descriptor;
   {
     base::AutoLock auto_lock(*va_lock_);
@@ -1687,6 +1760,10 @@ VaapiWrapper::ExportVASurfaceAsNativePix
   exported_pixmap->pixmap = base::MakeRefCounted<gfx::NativePixmapDmaBuf>(
       scoped_va_surface.size(), buffer_format, std::move(handle));
   return exported_pixmap;
+#else
+  LOG(ERROR) << "Exporting a surface with FOURCC has been disabled";
+  return nullptr;
+#endif
 }
 
 bool VaapiWrapper::SyncSurface(VASurfaceID va_surface_id) {
@@ -2060,6 +2137,7 @@ void VaapiWrapper::DestroyVABuffers() {
 }
 
 bool VaapiWrapper::IsRotationSupported() {
+#if VA_CHECK_VERSION(1, 1, 0)
   base::AutoLock auto_lock(*va_lock_);
   VAProcPipelineCaps pipeline_caps;
   memset(&pipeline_caps, 0, sizeof(pipeline_caps));
@@ -2074,6 +2152,10 @@ bool VaapiWrapper::IsRotationSupported()
     return false;
   }
   return true;
+#else
+  DVLOG(2) << "VA-API driver doesn't support any rotation";
+  return false;
+#endif
 }
 
 bool VaapiWrapper::BlitSurface(const VASurface& va_surface_src,
@@ -2129,6 +2211,7 @@ bool VaapiWrapper::BlitSurface(const VAS
     pipeline_param->output_color_standard = VAProcColorStandardNone;
     pipeline_param->filter_flags = VA_FILTER_SCALING_DEFAULT;
 
+#if VA_CHECK_VERSION(1, 1, 0)
     switch (rotation) {
       case VIDEO_ROTATION_0:
         pipeline_param->rotation_state = VA_ROTATION_NONE;
@@ -2143,6 +2226,7 @@ bool VaapiWrapper::BlitSurface(const VAS
         pipeline_param->rotation_state = VA_ROTATION_270;
         break;
     }
+#endif
 
     VA_SUCCESS_OR_RETURN(mapping.Unmap(), "Vpp Buffer unmapping", false);
   }
Index: dev/media/gpu/chromeos/fourcc.cc
===================================================================
--- dev.orig/media/gpu/chromeos/fourcc.cc
+++ dev/media/gpu/chromeos/fourcc.cc
@@ -215,11 +215,11 @@ uint32_t Fourcc::ToV4L2PixFmt() const {
 // static
 base::Optional<Fourcc> Fourcc::FromVAFourCC(uint32_t va_fourcc) {
   switch (va_fourcc) {
-    case VA_FOURCC_I420:
+    case VA_FOURCC('I', '4', '2', '0'):
       return Fourcc(YU12);
     case VA_FOURCC_NV12:
       return Fourcc(NV12);
-    case VA_FOURCC_NV21:
+    case VA_FOURCC('N', 'V', '2', '1'):
       return Fourcc(NV21);
     case VA_FOURCC_YV12:
       return Fourcc(YV12);
@@ -245,11 +245,11 @@ base::Optional<Fourcc> Fourcc::FromVAFou
 base::Optional<uint32_t> Fourcc::ToVAFourCC() const {
   switch (value_) {
     case YU12:
-      return VA_FOURCC_I420;
+      return VA_FOURCC('I', '4', '2', '0');
     case NV12:
       return VA_FOURCC_NV12;
     case NV21:
-      return VA_FOURCC_NV21;
+      return VA_FOURCC('N', 'V', '2', '1');
     case YV12:
       return VA_FOURCC_YV12;
     case YUYV:
Index: dev/ui/gfx/x/gen_xproto.py
===================================================================
--- dev.orig/ui/gfx/x/gen_xproto.py
+++ dev/ui/gfx/x/gen_xproto.py
@@ -417,7 +417,7 @@ class GenXproto(FileWriter):
             return 'Request'
         elif t.is_reply:
             return 'Reply'
-        elif t.is_event:
+        elif isinstance(t, self.xcbgen.xtypes.Event):
             return 'Event'
         return ''
 
@@ -629,7 +629,7 @@ class GenXproto(FileWriter):
                 copy_basic()
             else:
                 container_type, container_name = field.parent
-                assert container_type.is_event
+                assert isinstance(container_type, self.xcbgen.xtypes.Event)
                 opcode = container_type.opcodes[container_name]
                 self.write('%s %s = %s;' % (type_name, name, opcode))
                 self.copy_primitive(name)
@@ -650,7 +650,8 @@ class GenXproto(FileWriter):
             self.copy_primitive(name)
 
     def declare_case(self, case):
-        assert case.type.is_case != case.type.is_bitcase
+        if hasattr(case.type, 'is_case'):
+            assert case.type.is_case != case.type.is_bitcase
 
         fields = [
             field for case_field in case.type.fields
@@ -666,7 +667,10 @@ class GenXproto(FileWriter):
         return [(typename, name)]
 
     def copy_case(self, case, switch_name):
-        op = 'CaseEq' if case.type.is_case else 'CaseAnd'
+        if hasattr(case.type, "is_case"):
+            op = 'CaseEq' if case.type.is_case else 'CaseAnd'
+        else:
+            op = 'CaseAnd'
         condition = ' || '.join([
             '%s(%s_expr, %s)' % (op, switch_name, self.expr(expr))
             for expr in case.type.expr
@@ -1075,7 +1079,7 @@ class GenXproto(FileWriter):
             self.write_special_container(item, name)
         if isinstance(item, self.xcbgen.xtypes.Request):
             self.define_request(item)
-        elif item.is_event:
+        elif isinstance(item, self.xcbgen.xtypes.Event):
             self.define_event(item, name)
 
     def declare_type(self, item, name):
@@ -1083,7 +1087,7 @@ class GenXproto(FileWriter):
             self.declare_union(item)
         elif isinstance(item, self.xcbgen.xtypes.Request):
             self.declare_request(item)
-        elif item.is_event:
+        elif isinstance(item, self.xcbgen.xtypes.Event):
             self.declare_event(item, name)
         elif item.is_container:
             self.declare_container(item, name)
@@ -1163,7 +1167,7 @@ class GenXproto(FileWriter):
                         and field_name in fields):
                     setattr(fields[field_name], for_type, field)
 
-            if field.type.is_switch or field.type.is_case_or_bitcase:
+            if field.type.is_switch or (field.type.is_case_or_bitcase if hasattr(field.type, 'is_case_or_bitcase') else field.type.is_bitcase):
                 self.resolve_type(field.type, field.field_type)
 
         if isinstance(t, self.xcbgen.xtypes.Request) and t.reply:
@@ -1324,6 +1328,11 @@ class GenXproto(FileWriter):
             self.write('#include "%s.h"' % direct_import[-1])
         self.write('#include "%s_undef.h"' % self.module.namespace.header)
         self.write()
+        self.write('#define X11_%s_MAJOR_VERSION %s' %
+                (self.class_name.upper(), self.module.namespace.major_version))
+        self.write('#define X11_%s_MINOR_VERSION %s' %
+                (self.class_name.upper(), self.module.namespace.minor_version))
+        self.write()
         self.write('namespace x11 {')
         self.write()
         self.write('class Connection;')
Index: dev/ui/gtk/gtk_util.cc
===================================================================
--- dev.orig/ui/gtk/gtk_util.cc
+++ dev/ui/gtk/gtk_util.cc
@@ -32,7 +32,9 @@
 #include "ui/ozone/public/ozone_platform.h"
 #include "ui/views/linux_ui/linux_ui.h"
 
+#if GTK_CHECK_VERSION(3, 20, 0)
 WEAK_GTK_FN(gtk_widget_path_iter_set_object_name);
+#endif
 WEAK_GTK_FN(gtk_widget_path_iter_set_state);
 
 namespace {
@@ -344,8 +346,10 @@ ScopedStyleContext AppendCssNodeToStyleC
           break;
         case CSS_OBJECT_NAME:
           if (GtkCheckVersion(3, 20)) {
+#if GTK_CHECK_VERSION(3, 20, 0)
             DCHECK(gtk_widget_path_iter_set_object_name);
             gtk_widget_path_iter_set_object_name(path, -1, t.token().c_str());
+#endif
           } else {
             gtk_widget_path_iter_add_class(path, -1, t.token().c_str());
           }
@@ -355,8 +359,10 @@ ScopedStyleContext AppendCssNodeToStyleC
           DCHECK(type);
           gtk_widget_path_append_type(path, type);
           if (GtkCheckVersion(3, 20) && t.token() == "GtkLabel") {
+#if GTK_CHECK_VERSION(3, 20, 0)
             DCHECK(gtk_widget_path_iter_set_object_name);
             gtk_widget_path_iter_set_object_name(path, -1, "label");
+#endif
           }
           break;
         }
Index: dev/ui/gfx/linux/drm_util_linux.cc
===================================================================
--- dev.orig/ui/gfx/linux/drm_util_linux.cc
+++ dev/ui/gfx/linux/drm_util_linux.cc
@@ -8,6 +8,15 @@
 
 #include "base/notreached.h"
 
+#ifndef DRM_FORMAT_INVALID
+// TODO(mcasas): Remove when uprevving //third_party/libdrm.
+#define DRM_FORMAT_INVALID 0
+#endif
+
+#ifndef DRM_FORMAT_P010
+#define DRM_FORMAT_P010 fourcc_code('P', '0', '1', '0')
+#endif
+
 namespace ui {
 
 int GetFourCCFormatFromBufferFormat(gfx::BufferFormat format) {
Index: dev/ui/base/x/x11_display_util.cc
===================================================================
--- dev.orig/ui/base/x/x11_display_util.cc
+++ dev/ui/base/x/x11_display_util.cc
@@ -34,6 +34,7 @@ std::map<x11::RandR::Output, int> GetMon
                                               x11::RandR* randr,
                                               x11::Window window) {
   std::map<x11::RandR::Output, int> output_to_monitor;
+#if X11_RANDR_MINOR_VERSION >= 5
   if (version >= 105) {
     if (auto reply = randr->GetMonitors({window}).Sync()) {
       for (size_t monitor = 0; monitor < reply->monitors.size(); monitor++) {
@@ -42,6 +43,7 @@ std::map<x11::RandR::Output, int> GetMon
       }
     }
   }
+#endif
   return output_to_monitor;
 }
 
Index: dev/third_party/nearby/src/cpp/core_v2/internal/payload_manager.cc
===================================================================
--- dev.orig/third_party/nearby/src/cpp/core_v2/internal/payload_manager.cc
+++ dev/third_party/nearby/src/cpp/core_v2/internal/payload_manager.cc
@@ -946,7 +946,7 @@ PayloadManager::PendingPayload::PendingP
   for (const auto& id : endpoint_ids) {
     endpoints_.emplace(id, EndpointInfo{
                                .id = id,
-                               .status {EndpointInfo::Status::kAvailable},
+                               .status = AtomicReference<EndpointInfo::Status>(EndpointInfo::Status::kAvailable),
                            });
   }
 }
Index: dev/ui/gfx/x/xproto_internal.cc
===================================================================
--- dev.orig/ui/gfx/x/xproto_internal.cc
+++ dev/ui/gfx/x/xproto_internal.cc
@@ -15,7 +15,7 @@
 
 // Declare the function in case this is a packager build on an older distro with
 // use_sysroot=false.
-unsigned int xcb_send_request_with_fds(xcb_connection_t* c,
+extern "C" unsigned int xcb_send_request_with_fds(xcb_connection_t* c,
                                        int flags,
                                        struct iovec* vector,
                                        const xcb_protocol_request_t* request,
Index: dev/ui/gfx/x/event.cc
===================================================================
--- dev.orig/ui/gfx/x/event.cc
+++ dev/ui/gfx/x/event.cc
@@ -63,6 +63,10 @@ Event::Event(scoped_refptr<base::RefCoun
     if ((xcb_event->response_type & ~kSendEventMask) ==
         x11::GeGenericEvent::opcode) {
       auto* ge = reinterpret_cast<xcb_ge_event_t*>(xcb_event);
+#if 1
+      // Xenial doesn't know what xcb_raw_generic_event_t is
+      memmove(&ge->full_sequence, &ge[1], ge->length * 4);
+#else
       constexpr size_t ge_length = sizeof(xcb_raw_generic_event_t);
       constexpr size_t offset = sizeof(ge->full_sequence);
       size_t extended_length = ge->length * 4;
@@ -78,6 +82,7 @@ Event::Event(scoped_refptr<base::RefCoun
             event_bytes, offset, ge_length + extended_length);
         xcb_event = reinterpret_cast<xcb_generic_event_t*>(addr + offset);
       }
+#endif
     }
   }
 
