Index: dev/third_party/libpng/BUILD.gn
===================================================================
--- dev.orig/third_party/libpng/BUILD.gn
+++ dev/third_party/libpng/BUILD.gn
@@ -67,7 +67,7 @@ source_set("libpng_sources") {
       "contrib/intel/intel_init.c",
     ]
     defines += [ "PNG_INTEL_SSE_OPT=1" ]
-  } else if (current_cpu == "arm" || current_cpu == "arm64") {
+  } else if ((current_cpu == "arm" || current_cpu == "arm64") && arm_use_neon) {
     sources += [
       "arm/arm_init.c",
       "arm/filter_neon_intrinsics.c",
