Index: dev.vivid/third_party/ffmpeg/ffmpeg.gyp
===================================================================
--- dev.vivid.orig/third_party/ffmpeg/ffmpeg.gyp	2015-06-19 16:20:23.000000000 -0400
+++ dev.vivid/third_party/ffmpeg/ffmpeg.gyp	2015-07-02 11:11:29.950783357 -0400
@@ -167,15 +167,15 @@
       'platform_config_root': 'chromium/config/<(ffmpeg_branding)/<(os_config)/<(ffmpeg_config)',
     },
     'conditions': [
+      'direct_dependent_settings': {
+        'include_dirs': [
+          '../..',  # The chromium 'src' directory.
+          '<(platform_config_root)',
+          '.',
+        ],
+      },
       ['build_ffmpegsumo == 1',
         {
-          'direct_dependent_settings': {
-            'include_dirs': [
-              '../..',  # The chromium 'src' directory.
-              '<(platform_config_root)',
-              '.',
-            ],
-          },
           'includes': [
             'ffmpeg_generated.gypi',
             '../../build/util/version.gypi',
