Index: beta/ui/ozone/platform/wayland/BUILD.gn
===================================================================
--- beta.orig/ui/ozone/platform/wayland/BUILD.gn
+++ beta/ui/ozone/platform/wayland/BUILD.gn
@@ -5,11 +5,18 @@
 visibility = [ "//ui/ozone/*" ]
 
 import("//build/config/linux/pkg_config.gni")
+import("//ui/base/ui_features.gni")
 
 pkg_config("wayland-egl") {
   packages = [ "wayland-egl" ]
 }
 
+if (use_xkbcommon) {
+  pkg_config("xkbcommon") {
+    packages = [ "xkbcommon" ]
+  }
+}
+
 source_set("wayland") {
   sources = [
     "client_native_pixmap_factory_wayland.cc",
@@ -66,6 +73,10 @@ source_set("wayland") {
     ":wayland-egl",
     "//third_party/khronos:khronos_headers",
   ]
+
+  if (use_xkbcommon) {
+    configs += [ ":xkbcommon" ]
+  }
 }
 
 source_set("wayland_unittests") {
