Index: beta/tools/gn/bootstrap/bootstrap.py
===================================================================
--- beta.orig/tools/gn/bootstrap/bootstrap.py
+++ beta/tools/gn/bootstrap/bootstrap.py
@@ -761,7 +761,6 @@ def write_gn_ninja(path, root_gen_dir, o
         'base/sync_socket_win.cc',
         'base/synchronization/condition_variable_win.cc',
         'base/synchronization/lock_impl_win.cc',
-        'base/synchronization/read_write_lock_win.cc',
         'base/synchronization/waitable_event_watcher_win.cc',
         'base/synchronization/waitable_event_win.cc',
         'base/sys_info_win.cc',
