Index: dev/chrome/common/chrome_content_client.cc
===================================================================
--- dev.orig/chrome/common/chrome_content_client.cc
+++ dev/chrome/common/chrome_content_client.cc
@@ -238,10 +238,7 @@ void ComputeBuiltInPlugins(std::vector<c
     info.is_out_of_process = true;
     info.path = adapter_path;
     info.name = kWidevineCdmDisplayName;
-    info.description =
-        base::StringPrintf("%s (version: " WIDEVINE_CDM_VERSION_STRING ")",
-                           kWidevineCdmDescription);
-    info.version = WIDEVINE_CDM_VERSION_STRING;
+    info.description = kWidevineCdmDescription;
     info.permissions = kWidevineCdmPluginPermissions;
 
     content::WebPluginMimeType mime_type(kWidevineCdmPluginMimeType,
@@ -581,8 +578,7 @@ void ChromeContentClient::AddContentDecr
       // not the adapter, so adjust as necessary. It will be in the
       // same directory as the installed adapter.
       // TODO(xhwang): Pass |is_persistent_license_supported| to CdmInfo.
-      const base::Version version(WIDEVINE_CDM_VERSION_STRING);
-      DCHECK(version.IsValid());
+      const base::Version version;
       cdms->push_back(
           content::CdmInfo(kWidevineCdmDisplayName, kWidevineCdmGuid, version,
                            cdm_path, kWidevineCdmFileSystemId, codecs_supported,
