Index: dev.wily/chrome/common/chrome_content_client.cc
===================================================================
--- dev.wily.orig/chrome/common/chrome_content_client.cc	2015-07-17 00:00:44.563433805 -0400
+++ dev.wily/chrome/common/chrome_content_client.cc	2015-07-17 00:00:44.560433844 -0400
@@ -172,10 +172,7 @@
       widevine_cdm.is_out_of_process = true;
       widevine_cdm.path = path;
       widevine_cdm.name = kWidevineCdmDisplayName;
-      widevine_cdm.description = kWidevineCdmDescription +
-                                 std::string(" (version: ") +
-                                 WIDEVINE_CDM_VERSION_STRING + ")";
-      widevine_cdm.version = WIDEVINE_CDM_VERSION_STRING;
+      widevine_cdm.description = kWidevineCdmDescription;
       content::WebPluginMimeType widevine_cdm_mime_type(
           kWidevineCdmPluginMimeType,
           kWidevineCdmPluginExtension,
