Index: beta/chrome/common/chrome_content_client.cc
===================================================================
--- beta.orig/chrome/common/chrome_content_client.cc
+++ beta/chrome/common/chrome_content_client.cc
@@ -216,9 +216,7 @@ void ComputeBuiltInPlugins(std::vector<c
     widevine_cdm.path = adapter_path;
     widevine_cdm.name = kWidevineCdmDisplayName;
     widevine_cdm.description =
-        base::StringPrintf("%s (version: " WIDEVINE_CDM_VERSION_STRING ")",
-                           kWidevineCdmDescription);
-    widevine_cdm.version = WIDEVINE_CDM_VERSION_STRING;
+    widevine_cdm.description = kWidevineCdmDescription;
     content::WebPluginMimeType widevine_cdm_mime_type(
         kWidevineCdmPluginMimeType, kWidevineCdmPluginExtension,
         kWidevineCdmPluginMimeTypeDescription);
@@ -558,8 +556,7 @@ void ChromeContentClient::AddContentDecr
       // CdmInfo needs |path| to be the actual Widevine library,
       // not the adapter, so adjust as necessary. It will be in the
       // same directory as the installed adapter.
-      const base::Version version(WIDEVINE_CDM_VERSION_STRING);
-      DCHECK(version.IsValid());
+      const base::Version version;
       cdms->push_back(content::CdmInfo(kWidevineCdmType, version, cdm_path,
                                        codecs_supported));
     }
