Index: dev.xenial/chrome/common/chrome_content_client.cc
===================================================================
--- dev.xenial.orig/chrome/common/chrome_content_client.cc
+++ dev.xenial/chrome/common/chrome_content_client.cc
@@ -200,9 +200,7 @@ void ComputeBuiltInPlugins(std::vector<c
     widevine_cdm.path = adapter_path;
     widevine_cdm.name = kWidevineCdmDisplayName;
     widevine_cdm.description =
-        base::StringPrintf("%s (version: " WIDEVINE_CDM_VERSION_STRING ")",
-                           kWidevineCdmDescription);
-    widevine_cdm.version = WIDEVINE_CDM_VERSION_STRING;
+    widevine_cdm.description = kWidevineCdmDescription;
     content::WebPluginMimeType widevine_cdm_mime_type(
         kWidevineCdmPluginMimeType, kWidevineCdmPluginExtension,
         kWidevineCdmPluginMimeTypeDescription);
