Index: dev/build/config/compiler/BUILD.gn
===================================================================
--- dev.orig/build/config/compiler/BUILD.gn
+++ dev/build/config/compiler/BUILD.gn
@@ -311,6 +311,10 @@ config("compiler") {
     }
   }
 
+  if (current_cpu == "x86") {
+    cflags += [ "-msse2", "-mfpmath=sse", "-mmmx" ]
+  }
+
   if (is_clang && is_debug) {
     # Allow comparing the address of references and 'this' against 0
     # in debug builds. Technically, these can never be null in
