Index: dev.xenial/build/toolchain/gcc_toolchain.gni
===================================================================
--- dev.xenial.orig/build/toolchain/gcc_toolchain.gni
+++ dev.xenial/build/toolchain/gcc_toolchain.gni
@@ -511,12 +511,13 @@ template("clang_toolchain") {
     ar = "${toolprefix}ar"
     nm = "${toolprefix}nm"
 
-    forward_variables_from(invoker,
-                           [
-                             "toolchain_cpu",
-                             "toolchain_os",
-                             "use_gold",
-                             "strip",
-                           ])
+    toolchain_cpu = invoker.toolchain_cpu
+    toolchain_os = invoker.toolchain_os
+    if (defined(invoker.use_gold)) {
+      use_gold = invoker.use_gold
+    }
+    if (defined(invoker.strip)) {
+      strip = invoker.strip
+    }
   }
 }
