Index: beta/build/toolchain/gcc_toolchain.gni
===================================================================
--- beta.orig/build/toolchain/gcc_toolchain.gni
+++ beta/build/toolchain/gcc_toolchain.gni
@@ -520,13 +520,14 @@ template("clang_toolchain") {
     ar = "${toolprefix}ar"
     nm = "${toolprefix}nm"
 
-    forward_variables_from(invoker,
-                           [
-                             "toolchain_cpu",
-                             "toolchain_os",
-                             "use_gold",
-                             "strip",
-                           ])
+    toolchain_cpu = invoker.toolchain_cpu
+    toolchain_os = invoker.toolchain_os
+    if (defined(invoker.use_gold)) {
+      use_gold = invoker.use_gold
+    }
+    if (defined(invoker.strip)) {
+      strip = invoker.strip
+    }
 
     if (defined(invoker.use_debug_fission)) {
       use_debug_fission = invoker.use_debug_fission
