Index: beta/v8/snapshot_toolchain.gni
===================================================================
--- beta.orig/v8/snapshot_toolchain.gni
+++ beta/v8/snapshot_toolchain.gni
@@ -31,7 +31,7 @@
 # TODO(GYP): For now we only support 32-bit little-endian target builds from an
 # x64 Linux host. Eventually we need to support all of the host/target
 # configurations v8 runs on.
-if (host_cpu == "x64" && host_os == "linux") {
+if (host_cpu == "x64" && host_os == "linux" && is_clang) {
   if (target_cpu == "arm" || target_cpu == "mipsel" || target_cpu == "x86") {
     snapshot_toolchain = "//build/toolchain/linux:clang_x86"
   } else if (target_cpu == "x64" || target_cpu == "arm64" || target_cpu == "mips64el") {
