Index: beta/tools/gn/bootstrap/bootstrap.py
===================================================================
--- beta.orig/tools/gn/bootstrap/bootstrap.py
+++ beta/tools/gn/bootstrap/bootstrap.py
@@ -100,7 +100,7 @@ def main(argv):
 
     append_to_env('LDFLAGS', [
         '-nodefaultlibs', 'libc++.gn.so',
-        '-lpthread', '-lc', '-lm',
+        '-lpthread', '-lc', '-lgcc_s', '-lm',
         '-Wl,-rpath="\$$ORIGIN/."', '-Wl,-rpath-link=.'
     ])
     append_to_env('CXXFLAGS', [
