=== modified file 'src/chrome/app/chromium_strings.grd'
Index: src/chrome/app/chromium_strings.grd
===================================================================
--- src.orig/chrome/app/chromium_strings.grd	2012-10-01 14:10:41.902592085 -0400
+++ src/chrome/app/chromium_strings.grd	2012-10-01 14:13:29.262597588 -0400
@@ -45,12 +45,8 @@
     <output filename="chromium_strings_fake-bidi.pak" type="data_package" lang="fake-bidi" />
     <output filename="chromium_strings_fi.pak" type="data_package" lang="fi" />
     <output filename="chromium_strings_fil.pak" type="data_package" lang="fil" />
-    <if expr="pp_ifdef('use_third_party_translations')">
-      <output filename="chromium_strings_fo.pak" type="data_package" lang="fo" />
-    </if>
     <output filename="chromium_strings_fr.pak" type="data_package" lang="fr" />
     <if expr="pp_ifdef('use_third_party_translations')">
-      <output filename="chromium_strings_gd.pak" type="data_package" lang="gd" />
       <output filename="chromium_strings_gl.pak" type="data_package" lang="gl" />
     </if>
     <output filename="chromium_strings_gu.pak" type="data_package" lang="gu" />
@@ -81,12 +77,6 @@
     <output filename="chromium_strings_ms.pak" type="data_package" lang="ms" />
     <output filename="chromium_strings_nl.pak" type="data_package" lang="nl" />
     <output filename="chromium_strings_nb.pak" type="data_package" lang="no" />
-    <if expr="pp_ifdef('use_third_party_translations')">
-      <output filename="chromium_strings_nn.pak" type="data_package" lang="nn" />
-    </if>
-    <if expr="pp_ifdef('use_third_party_translations')">
-      <output filename="chromium_strings_os.pak" type="data_package" lang="os" />
-    </if>
     <!-- 'no' for Norwegian Bokmål. It should be 'nb'. -->
     <output filename="chromium_strings_pl.pak" type="data_package" lang="pl" />
     <output filename="chromium_strings_pt-BR.pak" type="data_package" lang="pt-BR" />
@@ -98,9 +88,6 @@
     <output filename="chromium_strings_sr.pak" type="data_package" lang="sr" />
     <output filename="chromium_strings_sv.pak" type="data_package" lang="sv" />
     <output filename="chromium_strings_sw.pak" type="data_package" lang="sw" />
-    <if expr="pp_ifdef('use_third_party_translations')">
-      <output filename="chromium_strings_szl.pak" type="data_package" lang="szl" />
-    </if>
     <output filename="chromium_strings_ta.pak" type="data_package" lang="ta" />
     <output filename="chromium_strings_te.pak" type="data_package" lang="te" />
     <output filename="chromium_strings_th.pak" type="data_package" lang="th" />
@@ -109,89 +96,106 @@
       <output filename="chromium_strings_ug.pak" type="data_package" lang="ug" />
     </if>
     <output filename="chromium_strings_uk.pak" type="data_package" lang="uk" />
-    <if expr="pp_ifdef('use_third_party_translations')">
-      <output filename="chromium_strings_uz.pak" type="data_package" lang="uz" />
-    </if>
     <output filename="chromium_strings_vi.pak" type="data_package" lang="vi" />
     <output filename="chromium_strings_zh-CN.pak" type="data_package" lang="zh-CN" />
-    <if expr="pp_ifdef('use_third_party_translations')">
-      <output filename="chromium_strings_zh-HK.pak" type="data_package" lang="zh-HK" />
-    </if>
     <output filename="chromium_strings_zh-TW.pak" type="data_package" lang="zh-TW" />
   </outputs>
   <translations>
-    <file path="resources/chromium_strings_am.xtb" lang="am" />
-    <file path="resources/chromium_strings_ar.xtb" lang="ar" />
+    <if expr="not pp_ifdef('use_third_party_translations')">
+      <file path="resources/chromium_strings_am.xtb" lang="am" />
+      <file path="resources/chromium_strings_ar.xtb" lang="ar" />
+      <file path="resources/chromium_strings_bg.xtb" lang="bg" />
+      <file path="resources/chromium_strings_bn.xtb" lang="bn" />
+      <file path="resources/chromium_strings_ca.xtb" lang="ca" />
+      <file path="resources/chromium_strings_cs.xtb" lang="cs" />
+      <file path="resources/chromium_strings_da.xtb" lang="da" />
+      <file path="resources/chromium_strings_de.xtb" lang="de" />
+      <file path="resources/chromium_strings_el.xtb" lang="el" />
+      <file path="resources/chromium_strings_en-GB.xtb" lang="en-GB" />
+      <file path="resources/chromium_strings_es.xtb" lang="es" />
+      <file path="resources/chromium_strings_es-419.xtb" lang="es-419" />
+      <file path="resources/chromium_strings_et.xtb" lang="et" />
+      <file path="resources/chromium_strings_fa.xtb" lang="fa" />
+      <file path="resources/chromium_strings_fi.xtb" lang="fi" />
+      <file path="resources/chromium_strings_fil.xtb" lang="fil" />
+      <file path="resources/chromium_strings_fr.xtb" lang="fr" />
+      <file path="resources/chromium_strings_gu.xtb" lang="gu" />
+      <file path="resources/chromium_strings_hi.xtb" lang="hi" />
+      <file path="resources/chromium_strings_hr.xtb" lang="hr" />
+      <file path="resources/chromium_strings_hu.xtb" lang="hu" />
+      <file path="resources/chromium_strings_id.xtb" lang="id" />
+      <file path="resources/chromium_strings_it.xtb" lang="it" />
+      <!-- The translation console uses 'iw' for Hebrew, but we use 'he'. -->
+      <file path="resources/chromium_strings_iw.xtb" lang="he" />
+      <file path="resources/chromium_strings_ja.xtb" lang="ja" />
+      <file path="resources/chromium_strings_kn.xtb" lang="kn" />
+      <file path="resources/chromium_strings_ko.xtb" lang="ko" />
+      <file path="resources/chromium_strings_lt.xtb" lang="lt" />
+      <file path="resources/chromium_strings_lv.xtb" lang="lv" />
+      <file path="resources/chromium_strings_ml.xtb" lang="ml" />
+      <file path="resources/chromium_strings_mr.xtb" lang="mr" />
+      <file path="resources/chromium_strings_ms.xtb" lang="ms" />
+      <file path="resources/chromium_strings_nl.xtb" lang="nl" />
+      <file path="resources/chromium_strings_no.xtb" lang="no" />
+      <file path="resources/chromium_strings_pl.xtb" lang="pl" />
+      <file path="resources/chromium_strings_pt-BR.xtb" lang="pt-BR" />
+      <file path="resources/chromium_strings_pt-PT.xtb" lang="pt-PT" />
+      <file path="resources/chromium_strings_ro.xtb" lang="ro" />
+      <file path="resources/chromium_strings_ru.xtb" lang="ru" />
+      <file path="resources/chromium_strings_sk.xtb" lang="sk" />
+      <file path="resources/chromium_strings_sl.xtb" lang="sl" />
+      <file path="resources/chromium_strings_sr.xtb" lang="sr" />
+      <file path="resources/chromium_strings_sv.xtb" lang="sv" />
+      <file path="resources/chromium_strings_sw.xtb" lang="sw" />
+      <file path="resources/chromium_strings_ta.xtb" lang="ta" />
+      <file path="resources/chromium_strings_te.xtb" lang="te" />
+      <file path="resources/chromium_strings_th.xtb" lang="th" />
+      <file path="resources/chromium_strings_tr.xtb" lang="tr" />
+      <file path="resources/chromium_strings_uk.xtb" lang="uk" />
+      <file path="resources/chromium_strings_vi.xtb" lang="vi" />
+      <file path="resources/chromium_strings_zh-CN.xtb" lang="zh-CN" />
+      <file path="resources/chromium_strings_zh-TW.xtb" lang="zh-TW" />
+    </if>
+    <if expr="pp_ifdef('use_third_party_translations')">
       <file path="../../third_party/launchpad_translations/chromium_strings_ast.xtb" lang="ast" />
       <file path="../../third_party/launchpad_translations/chromium_strings_bg.xtb" lang="bg" />
-      <file path="resources/chromium_strings_bn.xtb" lang="bn" />
       <file path="../../third_party/launchpad_translations/chromium_strings_bs.xtb" lang="bs" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ca.xtb" lang="ca" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ca-valencia.xtb" lang="ca@valencia" />
       <file path="../../third_party/launchpad_translations/chromium_strings_cs.xtb" lang="cs" />
       <file path="../../third_party/launchpad_translations/chromium_strings_da.xtb" lang="da" />
       <file path="../../third_party/launchpad_translations/chromium_strings_de.xtb" lang="de" />
-      <file path="resources/chromium_strings_el.xtb" lang="el" />
       <file path="../../third_party/launchpad_translations/chromium_strings_en-AU.xtb" lang="en-AU" />
-      <file path="resources/chromium_strings_en-GB.xtb" lang="en-GB" />
       <file path="../../third_party/launchpad_translations/chromium_strings_eo.xtb" lang="eo" />
       <file path="../../third_party/launchpad_translations/chromium_strings_es.xtb" lang="es" />
-      <file path="resources/chromium_strings_es-419.xtb" lang="es-419" />
       <file path="../../third_party/launchpad_translations/chromium_strings_et.xtb" lang="et" />
       <file path="../../third_party/launchpad_translations/chromium_strings_eu.xtb" lang="eu" />
-      <file path="resources/chromium_strings_fa.xtb" lang="fa" />
       <file path="../../third_party/launchpad_translations/chromium_strings_fi.xtb" lang="fi" />
-      <file path="resources/chromium_strings_fil.xtb" lang="fil" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_fo.xtb" lang="fo" />
       <file path="../../third_party/launchpad_translations/chromium_strings_fr.xtb" lang="fr" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_gd.xtb" lang="gd" />
       <file path="../../third_party/launchpad_translations/chromium_strings_gl.xtb" lang="gl" />
-      <file path="resources/chromium_strings_gu.xtb" lang="gu" />
       <file path="../../third_party/launchpad_translations/chromium_strings_he.xtb" lang="he" />
-      <file path="resources/chromium_strings_hr.xtb" lang="hr" />
-      <file path="resources/chromium_strings_hu.xtb" lang="hu" />
       <file path="../../third_party/launchpad_translations/chromium_strings_hy.xtb" lang="hy" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ia.xtb" lang="ia" />
       <file path="../../third_party/launchpad_translations/chromium_strings_id.xtb" lang="id" />
       <file path="../../third_party/launchpad_translations/chromium_strings_it.xtb" lang="it" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_iw.xtb" lang="he" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ja.xtb" lang="ja" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ka.xtb" lang="ka" />
-      <file path="resources/chromium_strings_kn.xtb" lang="kn" />
-      <file path="resources/chromium_strings_ko.xtb" lang="ko" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ku.xtb" lang="ku" />
       <file path="../../third_party/launchpad_translations/chromium_strings_kw.xtb" lang="kw" />
       <file path="../../third_party/launchpad_translations/chromium_strings_lt.xtb" lang="lt" />
-      <file path="resources/chromium_strings_lv.xtb" lang="lv" />
-      <file path="resources/chromium_strings_ml.xtb" lang="ml" />
-      <file path="resources/chromium_strings_mr.xtb" lang="mr" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ms.xtb" lang="ms" />
       <file path="../../third_party/launchpad_translations/chromium_strings_nl.xtb" lang="nl" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_nn.xtb" lang="nn" />
       <file path="../../third_party/launchpad_translations/chromium_strings_no.xtb" lang="no" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_os.xtb" lang="os" />
-      <file path="resources/chromium_strings_pl.xtb" lang="pl" />
       <file path="../../third_party/launchpad_translations/chromium_strings_pt-BR.xtb" lang="pt-BR" />
-      <file path="resources/chromium_strings_pt-PT.xtb" lang="pt-PT" />
-      <file path="resources/chromium_strings_ro.xtb" lang="ro" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ru.xtb" lang="ru" />
-      <file path="resources/chromium_strings_sk.xtb" lang="sk" />
       <file path="../../third_party/launchpad_translations/chromium_strings_sl.xtb" lang="sl" />
       <file path="../../third_party/launchpad_translations/chromium_strings_sr.xtb" lang="sr" />
       <file path="../../third_party/launchpad_translations/chromium_strings_sv.xtb" lang="sv" />
-      <file path="resources/chromium_strings_sw.xtb" lang="sw" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_szl.xtb" lang="szl" />
-      <file path="resources/chromium_strings_ta.xtb" lang="ta" />
-      <file path="resources/chromium_strings_te.xtb" lang="te" />
-      <file path="resources/chromium_strings_th.xtb" lang="th" />
-      <file path="resources/chromium_strings_tr.xtb" lang="tr" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ug.xtb" lang="ug" />
-      <file path="resources/chromium_strings_uk.xtb" lang="uk" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_uz.xtb" lang="uz" />
       <file path="../../third_party/launchpad_translations/chromium_strings_vi.xtb" lang="vi" />
       <file path="../../third_party/launchpad_translations/chromium_strings_zh-CN.xtb" lang="zh-CN" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_zh-HK.xtb" lang="zh-HK" />
       <file path="../../third_party/launchpad_translations/chromium_strings_zh-TW.xtb" lang="zh-TW" />
+    </if>
   </translations>
   <release seq="1" allow_pseudo="false">
     <messages fallback_to_english="true">
