Index: dev.saucy/src/build/common.gypi
===================================================================
--- dev.saucy.orig/src/build/common.gypi	2013-10-19 11:58:44.464424049 -0400
+++ dev.saucy/src/build/common.gypi	2013-10-19 11:58:44.444423330 -0400
@@ -3393,7 +3393,7 @@
           ['linux_dump_symbols==1', {
             'cflags': [ '-g' ],
             'conditions': [
-              ['target_arch=="ia32" and OS!="android"', {
+              ['OS!="android"', {
                 'target_conditions': [
                   ['_toolset=="target"', {
                     'ldflags': [
