Index: dev.saucy/src/build/common.gypi
===================================================================
--- dev.saucy.orig/src/build/common.gypi	2013-09-29 08:50:36.576184163 -0400
+++ dev.saucy/src/build/common.gypi	2013-09-29 08:50:36.560183584 -0400
@@ -3386,7 +3386,7 @@
           ['linux_dump_symbols==1', {
             'cflags': [ '-g' ],
             'conditions': [
-              ['target_arch=="ia32" and OS!="android"', {
+              ['OS!="android"', {
                 'target_conditions': [
                   ['_toolset=="target"', {
                     'ldflags': [
