Index: dev.saucy/src/build/common.gypi
===================================================================
--- dev.saucy.orig/src/build/common.gypi	2013-09-21 22:13:17.042847857 -0400
+++ dev.saucy/src/build/common.gypi	2013-09-21 22:13:17.034848126 -0400
@@ -3387,7 +3387,7 @@
           ['linux_dump_symbols==1', {
             'cflags': [ '-g' ],
             'conditions': [
-              ['target_arch=="ia32" and OS!="android"', {
+              ['OS!="android"', {
                 'target_conditions': [
                   ['_toolset=="target"', {
                     'ldflags': [
