Index: dev.saucy/src/build/common.gypi
===================================================================
--- dev.saucy.orig/src/build/common.gypi	2013-11-03 07:15:41.103320639 -0500
+++ dev.saucy/src/build/common.gypi	2013-11-03 07:15:41.087320244 -0500
@@ -3386,7 +3386,7 @@
           ['linux_dump_symbols==1', {
             'cflags': [ '-g' ],
             'conditions': [
-              ['target_arch=="ia32" and OS!="android"', {
+              ['OS!="android"', {
                 'target_conditions': [
                   ['_toolset=="target"', {
                     'ldflags': [
