Index: upstream/src/ui/base/strings/app_locale_settings.grd
===================================================================
--- upstream.orig/src/ui/base/strings/app_locale_settings.grd	2013-03-04 18:42:08.431751343 -0600
+++ upstream/src/ui/base/strings/app_locale_settings.grd	2013-03-04 18:42:14.379758370 -0600
@@ -94,6 +94,7 @@
       <output filename="app_locale_settings_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="app_locale_settings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="app_locale_settings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
Index: upstream/src/chrome/app/resources/locale_settings.grd
===================================================================
--- upstream.orig/src/chrome/app/resources/locale_settings.grd	2013-03-04 18:40:52.459859900 -0600
+++ upstream/src/chrome/app/resources/locale_settings.grd	2013-03-04 18:42:14.379758370 -0600
@@ -94,6 +94,7 @@
       <output filename="locale_settings_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="locale_settings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="locale_settings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
Index: upstream/src/chrome/app/chromium_strings.grd
===================================================================
--- upstream.orig/src/chrome/app/chromium_strings.grd	2013-03-04 18:28:22.000000000 -0600
+++ upstream/src/chrome/app/chromium_strings.grd	2013-03-04 18:42:35.223800200 -0600
@@ -98,6 +98,7 @@
     </if>
     <!-- 'no' for Norwegian Bokmål. It should be 'nb'. -->
     <output filename="chromium_strings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="chromium_strings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
@@ -169,7 +170,7 @@
       <file path="../../third_party/launchpad_translations/chromium_strings_ia.xtb" lang="ia" />
       <file path="../../third_party/launchpad_translations/chromium_strings_id.xtb" lang="id" />
       <file path="../../third_party/launchpad_translations/chromium_strings_it.xtb" lang="it" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_iw.xtb" lang="he" />
+      <file path="../../third_party/launchpad_translations/chromium_strings_iw.xtb" lang="iw" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ja.xtb" lang="ja" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ka.xtb" lang="ka" />
       <file path="resources/chromium_strings_kn.xtb" lang="kn" />
Index: upstream/src/chrome/app/generated_resources.grd
===================================================================
--- upstream.orig/src/chrome/app/generated_resources.grd	2013-03-04 07:44:02.000000000 -0600
+++ upstream/src/chrome/app/generated_resources.grd	2013-03-04 18:42:14.507758546 -0600
@@ -103,6 +103,7 @@
       <output filename="generated_resources_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="generated_resources_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="generated_resources_pt.pak" type="data_package" lang="pt-BR" />
     </if>
Index: upstream/src/ui/base/strings/ui_strings.grd
===================================================================
--- upstream.orig/src/ui/base/strings/ui_strings.grd	2013-03-04 07:44:02.000000000 -0600
+++ upstream/src/ui/base/strings/ui_strings.grd	2013-03-04 18:42:14.543758595 -0600
@@ -105,6 +105,7 @@
       <output filename="ui_strings_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="ui_strings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="ui_strings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
