Index: src/src/ui/base/strings/app_locale_settings.grd
===================================================================
--- src.orig/src/ui/base/strings/app_locale_settings.grd	2013-03-15 13:39:09.679070995 -0500
+++ src/src/ui/base/strings/app_locale_settings.grd	2013-03-15 13:39:23.043484745 -0500
@@ -97,6 +97,7 @@
       <output filename="app_locale_settings_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="app_locale_settings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="app_locale_settings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
Index: src/src/chrome/app/resources/locale_settings.grd
===================================================================
--- src.orig/src/chrome/app/resources/locale_settings.grd	2013-03-15 13:39:09.679070995 -0500
+++ src/src/chrome/app/resources/locale_settings.grd	2013-03-15 13:39:23.043484745 -0500
@@ -97,6 +97,7 @@
       <output filename="locale_settings_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="locale_settings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="locale_settings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
Index: src/src/chrome/app/chromium_strings.grd
===================================================================
--- src.orig/src/chrome/app/chromium_strings.grd	2013-03-15 13:39:09.679070995 -0500
+++ src/src/chrome/app/chromium_strings.grd	2013-03-15 13:39:23.051484994 -0500
@@ -101,6 +101,7 @@
     </if>
     <!-- 'no' for Norwegian Bokmål. It should be 'nb'. -->
     <output filename="chromium_strings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="chromium_strings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
@@ -172,7 +173,7 @@
       <file path="../../third_party/launchpad_translations/chromium_strings_ia.xtb" lang="ia" />
       <file path="../../third_party/launchpad_translations/chromium_strings_id.xtb" lang="id" />
       <file path="../../third_party/launchpad_translations/chromium_strings_it.xtb" lang="it" />
-      <file path="../../third_party/launchpad_translations/chromium_strings_iw.xtb" lang="he" />
+      <file path="../../third_party/launchpad_translations/chromium_strings_iw.xtb" lang="iw" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ja.xtb" lang="ja" />
       <file path="../../third_party/launchpad_translations/chromium_strings_ka.xtb" lang="ka" />
       <file path="resources/chromium_strings_kn.xtb" lang="kn" />
Index: src/src/chrome/app/generated_resources.grd
===================================================================
--- src.orig/src/chrome/app/generated_resources.grd	2013-03-15 13:39:09.679070995 -0500
+++ src/src/chrome/app/generated_resources.grd	2013-03-15 13:39:23.067485488 -0500
@@ -106,6 +106,7 @@
       <output filename="generated_resources_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="generated_resources_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="generated_resources_pt.pak" type="data_package" lang="pt-BR" />
     </if>
Index: src/src/ui/base/strings/ui_strings.grd
===================================================================
--- src.orig/src/ui/base/strings/ui_strings.grd	2013-03-15 13:39:09.679070995 -0500
+++ src/src/ui/base/strings/ui_strings.grd	2013-03-15 13:39:23.083485984 -0500
@@ -108,6 +108,7 @@
       <output filename="ui_strings_os.pak" type="data_package" lang="os" />
     </if>
     <output filename="ui_strings_pl.pak" type="data_package" lang="pl" />
+    <if expr="pp_ifdef('ios')">
       <!-- iOS uses pt for pt-BR -->
       <output filename="ui_strings_pt.pak" type="data_package" lang="pt-BR" />
     </if>
