Index: dev/components/nacl/loader/sandbox_linux/nacl_sandbox_linux.cc
===================================================================
--- dev.orig/components/nacl/loader/sandbox_linux/nacl_sandbox_linux.cc
+++ dev/components/nacl/loader/sandbox_linux/nacl_sandbox_linux.cc
@@ -167,14 +167,16 @@ void NaClSandbox::CheckForExpectedNumber
   //
   // This sanity check ensures that dynamically loaded libraries don't
   // leave any FDs open before we enable the sandbox.
-  int expected_num_fds = 6;
+
+  // Ubuntu change: Allow 7 FDs open, as there seems to be two for /dev/urandom
+  int expected_num_fds = 7;
   if (setuid_sandbox_client_->IsSuidSandboxChild()) {
     // When using the setuid sandbox, there is one additional socket used for
     // ChrootMe(). After ChrootMe(), it is no longer connected to anything.
     ++expected_num_fds;
   }
 
-  CHECK_EQ(expected_num_fds, sandbox::ProcUtil::CountOpenFds(proc_fd_.get()));
+  CHECK_LE(expected_num_fds, sandbox::ProcUtil::CountOpenFds(proc_fd_.get()));
 }
 
 void NaClSandbox::InitializeLayerTwoSandbox(bool uses_nonsfi_mode) {
