Index: dev/build/config/compiler/BUILD.gn
===================================================================
--- dev.orig/build/config/compiler/BUILD.gn
+++ dev/build/config/compiler/BUILD.gn
@@ -394,7 +394,7 @@ config("compiler") {
     ldflags += [ "-fuse-ld=bfd" ]
   }
 
-  if (is_posix && (use_gold || (use_lld && !is_nacl)) && !using_sanitizer &&
+  if (is_posix && !is_nacl && (use_gold || use_lld) && !using_sanitizer &&
       !(is_android && use_order_profiling)) {
     # TODO(thakis): Remove `!is_android` below once NDK gold has been rolled
     # with the fix for https://sourceware.org/bugzilla/show_bug.cgi?id=17704
