Index: beta/build/config/compiler/BUILD.gn
===================================================================
--- beta.orig/build/config/compiler/BUILD.gn
+++ beta/build/config/compiler/BUILD.gn
@@ -1196,14 +1196,6 @@ config("compiler_deterministic") {
         "-Xclang",
         ".",
       ]
-    } else {
-      # -ffile-compilation-dir is an alias for both -fdebug-compilation-dir=
-      # and -fcoverage-compilation-dir=.
-      cflags += [ "-ffile-compilation-dir=." ]
-    }
-    if (!is_win) {
-      # We don't use clang -cc1as on Windows (yet? https://crbug.com/762167)
-      asmflags = [ "-Wa,-fdebug-compilation-dir,." ]
     }
 
     if (is_win && use_lld) {
