Index: beta/tools/gn/bootstrap/bootstrap.py
===================================================================
--- beta.orig/tools/gn/bootstrap/bootstrap.py
+++ beta/tools/gn/bootstrap/bootstrap.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2014 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: beta/tools/generate_stubs/generate_stubs.py
===================================================================
--- beta.orig/tools/generate_stubs/generate_stubs.py
+++ beta/tools/generate_stubs/generate_stubs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: beta/tools/gn/build/gen.py
===================================================================
--- beta.orig/tools/gn/build/gen.py
+++ beta/tools/gn/build/gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2014 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: beta/tools/gn/src/gn/setup.cc
===================================================================
--- beta.orig/tools/gn/src/gn/setup.cc
+++ beta/tools/gn/src/gn/setup.cc
@@ -717,7 +717,7 @@ bool Setup::FillPythonPath(const base::C
     }
     build_settings_.set_python_path(python_path.NormalizePathSeparatorsTo('/'));
 #else
-    build_settings_.set_python_path(base::FilePath("python"));
+    build_settings_.set_python_path(base::FilePath("python2"));
 #endif
   }
   return true;
Index: beta/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
===================================================================
--- beta.orig/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
+++ beta/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: beta/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
===================================================================
--- beta.orig/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
+++ beta/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
@@ -83,7 +83,7 @@ def _MinifyJS(input_js):
 
   with tempfile.NamedTemporaryFile() as _:
     args = [
-        'python',
+        'python3',
         rjsmin_path
     ]
     p = subprocess.Popen(args,
@@ -203,7 +203,7 @@ def _MinifyCSS(css_text):
       os.path.join(py_vulcanize_path, 'third_party', 'rcssmin', 'rcssmin.py'))
 
   with tempfile.NamedTemporaryFile() as _:
-    rcssmin_args = ['python', rcssmin_path]
+    rcssmin_args = ['python3', rcssmin_path]
     p = subprocess.Popen(rcssmin_args,
                          stdin=subprocess.PIPE,
                          stdout=subprocess.PIPE,
Index: beta/ui/gfx/x/gen_xproto.py
===================================================================
--- beta.orig/ui/gfx/x/gen_xproto.py
+++ beta/ui/gfx/x/gen_xproto.py
@@ -173,6 +173,8 @@ import re
 import sys
 import types
 
+from functools import cmp_to_key
+
 # __main__.output must be defined before importing xcbgen,
 # so this global is unavoidable.
 output = collections.defaultdict(int)
@@ -852,7 +854,7 @@ class GenXproto:
 
         self.resolve_element(t.elt, fields)
 
-        for field in fields.values():
+        for field in list(fields.values()):
             field.parent = t
             field.for_list = None
             if field.type.is_switch or (field.type.is_case_or_bitcase if hasattr(field.type, 'is_case_or_bitcase') else field.type.is_bitcase):
@@ -923,11 +925,13 @@ class GenXproto:
                 return 3
             return 2
 
-        def cmp((_1, item1), (_2, item2)):
+        def cmp_func(tuple1, tuple2):
+            (_1, item1) = tuple1
+            (_2, item2) = tuple2
             return type_order_priority(item1) - type_order_priority(item2)
 
         # sort() is guaranteed to be stable.
-        self.module.all.sort(cmp=cmp)
+        self.module.all.sort(key=cmp_to_key(cmp_func))
 
     def gen_header(self):
         self.file = self.args.headerfile
@@ -1019,8 +1023,14 @@ def main():
                             'dist-packages')
         sys.path.insert(1, path)
 
-    import xcbgen.xtypes
-    import xcbgen.state
+    try:
+        import xcbgen.xtypes
+        import xcbgen.state
+    except:
+        if sys.version_info < (3, 0):
+            import subprocess
+            subprocess.check_call(["python3"] + sys.argv)
+            return 0
 
     generator = GenXproto(args, xcbgen)
     generator.generate()
