Index: dev/tools/gn/bootstrap/bootstrap.py
===================================================================
--- dev.orig/tools/gn/bootstrap/bootstrap.py
+++ dev/tools/gn/bootstrap/bootstrap.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2014 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: dev/tools/generate_stubs/generate_stubs.py
===================================================================
--- dev.orig/tools/generate_stubs/generate_stubs.py
+++ dev/tools/generate_stubs/generate_stubs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: dev/tools/gn/build/gen.py
===================================================================
--- dev.orig/tools/gn/build/gen.py
+++ dev/tools/gn/build/gen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2014 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: dev/tools/gn/src/gn/setup.cc
===================================================================
--- dev.orig/tools/gn/src/gn/setup.cc
+++ dev/tools/gn/src/gn/setup.cc
@@ -795,7 +795,7 @@ bool Setup::FillPythonPath(const base::C
     }
     build_settings_.set_python_path(python_path);
 #else
-    build_settings_.set_python_path(base::FilePath("python"));
+    build_settings_.set_python_path(base::FilePath("python2"));
 #endif
   }
   return true;
Index: dev/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
===================================================================
--- dev.orig/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
+++ dev/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
Index: dev/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
===================================================================
--- dev.orig/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
+++ dev/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
@@ -83,7 +83,7 @@ def _MinifyJS(input_js):
 
   with tempfile.NamedTemporaryFile() as _:
     args = [
-        'python',
+        'python3',
         rjsmin_path
     ]
     p = subprocess.Popen(args,
@@ -203,7 +203,7 @@ def _MinifyCSS(css_text):
       os.path.join(py_vulcanize_path, 'third_party', 'rcssmin', 'rcssmin.py'))
 
   with tempfile.NamedTemporaryFile() as _:
-    rcssmin_args = ['python', rcssmin_path]
+    rcssmin_args = ['python3', rcssmin_path]
     p = subprocess.Popen(rcssmin_args,
                          stdin=subprocess.PIPE,
                          stdout=subprocess.PIPE,
Index: dev/ui/gfx/x/gen_xproto.py
===================================================================
--- dev.orig/ui/gfx/x/gen_xproto.py
+++ dev/ui/gfx/x/gen_xproto.py
@@ -1020,7 +1020,7 @@ class GenXproto(FileWriter):
 
         self.resolve_element(t.elt, fields)
 
-        for field in fields.values():
+        for field in list(fields.values()):
             if field.field_name == 'sequence':
                 field.visible = True
             field.parent = (t, name)
@@ -1205,10 +1205,11 @@ class GenXproto(FileWriter):
         for direct_import in sorted(list(imports)):
             self.write('#include "%s.h"' % direct_import[-1])
         self.write()
-        self.write('#define X11_%s_MAJOR_VERSION %s' %
-                (self.class_name.upper(), self.module.namespace.major_version))
-        self.write('#define X11_%s_MINOR_VERSION %s' %
-                (self.class_name.upper(), self.module.namespace.minor_version))
+        if self.module.namespace.is_ext:
+            self.write('#define X11_%s_MAJOR_VERSION %s' %
+                    (self.class_name.upper(), self.module.namespace.major_version))
+            self.write('#define X11_%s_MINOR_VERSION %s' %
+                    (self.class_name.upper(), self.module.namespace.minor_version))
         self.write()
         self.write('namespace x11 {')
         self.write()
@@ -1593,8 +1594,14 @@ def main():
     args = parser.parse_args()
 
     sys.path.insert(1, args.xcbproto_dir)
-    import xcbgen.xtypes
-    import xcbgen.state
+    try:
+        import xcbgen.xtypes
+        import xcbgen.state
+    except:
+        if sys.version_info < (3, 0):
+            import subprocess
+            subprocess.check_call(["python3"] + sys.argv)
+            return 0
 
     all_types = {}
     proto_src_dir = os.path.join(args.xcbproto_dir, 'src')
