Description: Avoid SIGBUS on ARM
Author: Chris Coulson <chris.coulson@canonical.com>
Bug: http://crbug.com/230063
--- a/src/chrome/browser/safe_browsing/protocol_parser.cc
+++ b/src/chrome/browser/safe_browsing/protocol_parser.cc
@@ -402,10 +402,14 @@
       return false;
     if (entry->IsPrefix()) {
       DCHECK_EQ(hash_len, (int)sizeof(SBPrefix));
-      entry->SetPrefixAt(i, *reinterpret_cast<const SBPrefix*>(*data));
+      SBPrefix aligned;
+      memcpy(&aligned, *data, sizeof(SBPrefix)); 
+      entry->SetPrefixAt(i, aligned);
     } else {
       DCHECK_EQ(hash_len, (int)sizeof(SBFullHash));
-      entry->SetFullHashAt(i, *reinterpret_cast<const SBFullHash*>(*data));
+      SBFullHash aligned;
+      memcpy(&aligned, *data, sizeof(SBFullHash));
+      entry->SetFullHashAt(i, aligned);
     }
     *data += hash_len;
     *remaining -= hash_len;
