Index: dev/third_party/angle/third_party/vulkan-validation-layers/BUILD.gn
===================================================================
--- dev.orig/third_party/angle/third_party/vulkan-validation-layers/BUILD.gn
+++ dev/third_party/angle/third_party/vulkan-validation-layers/BUILD.gn
@@ -555,6 +555,7 @@ source_set("vulkan_layer_utils") {
     ":vulkan_generate_helper_files",
   ]
   configs -= vulkan_undefine_configs
+  cflags_c = [ "-std=gnu11" ]
 }
 
 config("vulkan_core_validation_config") {
