Index: dev/build/toolchain/gcc_toolchain.gni
===================================================================
--- dev.orig/build/toolchain/gcc_toolchain.gni
+++ dev/build/toolchain/gcc_toolchain.gni
@@ -28,6 +28,10 @@ declare_args() {
       is_official_build &&
       # Don't enable for Android-on-Chrome OS.
       (target_os == "android" || target_os == "win")
+  
+  clang_use_compiler_version_suffix = false
+
+  clang_compiler_version_suffix = ""
 }
 
 # When the arg is set via args.gn, it applies to all toolchains. In order to not
@@ -737,13 +741,20 @@ template("gcc_toolchain") {
 # use_gold too.
 template("clang_toolchain") {
   gcc_toolchain(target_name) {
-    prefix = rebase_path("$clang_base_path/bin", root_build_dir)
-    cc = "${prefix}/clang"
-    cxx = "${prefix}/clang++"
+    if (clang_use_compiler_version_suffix) {
+      cc = "clang-${clang_compiler_version_suffix}"
+      cxx = "clang++-${clang_compiler_version_suffix}"
+      ar = "llvm-ar-${clang_compiler_version_suffix}"
+      nm = "llvm-nm-${clang_compiler_version_suffix}"
+      readelf = "llvm-readelf-${clang_compiler_version_suffix}"
+    } else {
+      cc = "clang"
+      cxx = "clang++"
+      ar = "llvm-ar"
+      nm = "llvm-nm"
+      readelf = "llvm-readelf"
+    }
     ld = cxx
-    readelf = "${prefix}/llvm-readelf"
-    ar = "${prefix}/llvm-ar"
-    nm = "${prefix}/llvm-nm"
 
     forward_variables_from(invoker,
                            [
