Index: dev.wily/ozone/ui/webui/ozone_webui.cc
===================================================================
--- dev.wily.orig/ozone/ui/webui/ozone_webui.cc
+++ dev.wily/ozone/ui/webui/ozone_webui.cc
@@ -64,6 +64,10 @@ gfx::Image OzoneWebUI::GetThemeImageName
   return gfx::Image();
 }
 
+bool OzoneWebUI::GetTint(int id, color_utils::HSL* tint) const {
+  return false;
+}
+
 bool OzoneWebUI::GetColor(int id, SkColor* color) const {
   return false;
 }
Index: dev.wily/ozone/ui/webui/ozone_webui.h
===================================================================
--- dev.wily.orig/ozone/ui/webui/ozone_webui.h
+++ dev.wily/ozone/ui/webui/ozone_webui.h
@@ -74,6 +74,7 @@ class OZONE_WAYLAND_EXPORT OzoneWebUI :
 
   // These methods are not needed
   gfx::Image GetThemeImageNamed(int id) const override;
+  bool GetTint(int id, color_utils::HSL* tint) const override;
   bool GetColor(int id, SkColor* color) const override;
   bool HasCustomImage(int id) const override;
   SkColor GetFocusRingColor() const override;
