Index: dev.xenial/ozone/wayland/data_device.h
===================================================================
--- dev.xenial.orig/ozone/wayland/data_device.h
+++ dev.xenial/ozone/wayland/data_device.h
@@ -8,8 +8,8 @@
 #include <wayland-client.h>
 
 #include <string>
+#include <stdint.h>
 
-#include "base/basictypes.h"
 #include "base/memory/scoped_ptr.h"
 #include "ozone/wayland/window.h"
 
Index: dev.xenial/ozone/wayland/data_offer.h
===================================================================
--- dev.xenial.orig/ozone/wayland/data_offer.h
+++ dev.xenial/ozone/wayland/data_offer.h
@@ -8,10 +8,9 @@
 #include <wayland-client.h>
 
 #include <string>
+#include <stdint.h>
 #include <vector>
 
-#include "base/basictypes.h"
-
 namespace ozonewayland {
 
 // The WaylandDataOffer represents copy-and-paste or drag-and-drop data sent to
Index: dev.xenial/ozone/wayland/display.cc
===================================================================
--- dev.xenial.orig/ozone/wayland/display.cc
+++ dev.xenial/ozone/wayland/display.cc
@@ -209,8 +209,8 @@ bool WaylandDisplay::LoadEGLGLES2Binding
   return true;
 }
 
-const int32*
-WaylandDisplay::GetEGLSurfaceProperties(const int32* desired_list) {
+const int32_t*
+WaylandDisplay::GetEGLSurfaceProperties(const int32_t* desired_list) {
   static const EGLint kConfigAttribs[] = {
     EGL_BUFFER_SIZE, 32,
     EGL_ALPHA_SIZE, 8,
Index: dev.xenial/ozone/wayland/display.h
===================================================================
--- dev.xenial.orig/ozone/wayland/display.h
+++ dev.xenial/ozone/wayland/display.h
@@ -21,9 +21,9 @@
 #include <map>
 #include <queue>
 #include <string>
+#include <stdint.h>
 #include <vector>
 
-#include "base/basictypes.h"
 #include "base/memory/shared_memory.h"
 #include "base/memory/weak_ptr.h"
 #include "ozone/platform/window_constants.h"
@@ -115,7 +115,7 @@ class WaylandDisplay : public ui::Surfac
       ui::SurfaceFactoryOzone::AddGLLibraryCallback add_gl_library,
       ui::SurfaceFactoryOzone::SetGLGetProcAddressProcCallback
       proc_address) override;
-  const int32* GetEGLSurfaceProperties(const int32* desired_list) override;
+  const int32_t* GetEGLSurfaceProperties(const int32_t* desired_list) override;
   scoped_refptr<ui::NativePixmap> CreateNativePixmap(
       gfx::AcceleratedWidget widget, gfx::Size size, gfx::BufferFormat format,
           gfx::BufferUsage usage) override;
Index: dev.xenial/ozone/wayland/egl/egl_window.h
===================================================================
--- dev.xenial.orig/ozone/wayland/egl/egl_window.h
+++ dev.xenial/ozone/wayland/egl/egl_window.h
@@ -6,8 +6,7 @@
 #define OZONE_WAYLAND_EGL_EGL_WINDOW_H_
 
 #include <wayland-client.h>
-
-#include "base/basictypes.h"
+#include <stdint.h>
 
 struct wl_egl_window;
 
Index: dev.xenial/ozone/wayland/egl/surface_ozone_wayland.h
===================================================================
--- dev.xenial.orig/ozone/wayland/egl/surface_ozone_wayland.h
+++ dev.xenial/ozone/wayland/egl/surface_ozone_wayland.h
@@ -5,7 +5,8 @@
 #ifndef OZONE_WAYLAND_EGL_SURFACE_OZONE_WAYLAND
 #define OZONE_WAYLAND_EGL_SURFACE_OZONE_WAYLAND
 
-#include "base/basictypes.h"
+#include <stdint.h>
+
 #include "base/memory/scoped_ptr.h"
 #include "ui/gfx/gfx_export.h"
 #include "ui/ozone/public/surface_ozone_egl.h"
Index: dev.xenial/ozone/wayland/input/cursor.h
===================================================================
--- dev.xenial.orig/ozone/wayland/input/cursor.h
+++ dev.xenial/ozone/wayland/input/cursor.h
@@ -6,9 +6,9 @@
 #define OZONE_WAYLAND_INPUT_CURSOR_H_
 
 #include <wayland-client.h>
+#include <stdint.h>
 #include <vector>
 
-#include "base/basictypes.h"
 #include "third_party/skia/include/core/SkBitmap.h"
 
 namespace base {
Index: dev.xenial/ozone/wayland/screen.h
===================================================================
--- dev.xenial.orig/ozone/wayland/screen.h
+++ dev.xenial/ozone/wayland/screen.h
@@ -8,7 +8,6 @@
 
 #include <stdint.h>
 
-#include "base/basictypes.h"
 #include "ui/gfx/geometry/rect.h"
 
 struct wl_output;
Index: dev.xenial/ozone/wayland/seat.h
===================================================================
--- dev.xenial.orig/ozone/wayland/seat.h
+++ dev.xenial/ozone/wayland/seat.h
@@ -6,9 +6,9 @@
 #define OZONE_WAYLAND_SEAT_H_
 
 #include <wayland-client.h>
+#include <stdint.h>
 #include <vector>
 
-#include "base/basictypes.h"
 #include "third_party/skia/include/core/SkBitmap.h"
 #include "ui/gfx/geometry/rect.h"
 
Index: dev.xenial/ozone/wayland/shell/shell.h
===================================================================
--- dev.xenial.orig/ozone/wayland/shell/shell.h
+++ dev.xenial/ozone/wayland/shell/shell.h
@@ -6,8 +6,8 @@
 #define OZONE_WAYLAND_SHELL_SHELL_H_
 
 #include <wayland-client.h>
+#include <stdint.h>
 
-#include "base/basictypes.h"
 #include "ozone/wayland/window.h"
 
 struct xdg_shell;
Index: dev.xenial/ozone/wayland/shell/shell_surface.h
===================================================================
--- dev.xenial.orig/ozone/wayland/shell/shell_surface.h
+++ dev.xenial/ozone/wayland/shell/shell_surface.h
@@ -6,8 +6,8 @@
 #define OZONE_WAYLAND_SHELL_SHELL_SURFACE_H_
 
 #include <wayland-client.h>
+#include <stdint.h>
 
-#include "base/basictypes.h"
 #include "ozone/wayland/window.h"
 
 namespace ozonewayland {
