Index: dev.xenial/ui/views/views.gyp
===================================================================
--- dev.xenial.orig/ui/views/views.gyp
+++ dev.xenial/ui/views/views.gyp
@@ -763,6 +763,7 @@
             '../aura/aura.gyp:aura',
             '../touch_selection/ui_touch_selection.gyp:ui_touch_selection',
             '../wm/wm.gyp:wm',
+            '../../content/content.gyp:content_browser',
           ],
         }],
         ['use_aura and chromeos == 0', {
