Index: src/src/webkit/build/webkit_version.py
===================================================================
--- src.orig/src/webkit/build/webkit_version.py	2013-03-15 11:02:24.372129196 -0500
+++ src/src/webkit/build/webkit_version.py	2013-03-15 11:03:10.353617735 -0500
@@ -62,6 +62,15 @@
       directory=os.path.join(webkit_dir, version_file_dir),
       directory_regex_prior_to_src_url='webkit')
 
+  if version_info.url is None or version_info.url == 'unknown':
+    # parse the DEPS file
+    data = open("../../DEPS", "rb").read()
+    x = re.search(r'"webkit_revision": "(\d+)",', data)
+    rev = x.group(1) if x else "n-a"
+    x = re.search(r'"webkit_trunk": "(.*?)",', data)
+    branch = os.path.basename(x.group(1)) if x else "n-a"
+    return "%s@%s" % (branch, rev)
+
   if version_info.url == None:
     version_info.url = 'Unknown URL'
   version_info.url = version_info.url.strip('/')
