Index: beta/third_party/widevine/cdm/widevine.gni
===================================================================
--- beta.orig/third_party/widevine/cdm/widevine.gni
+++ beta/third_party/widevine/cdm/widevine.gni
@@ -31,7 +31,7 @@ enable_library_widevine_cdm =
 # and Windows. The CDM can be bundled regardless whether it's a component. See
 # below.
 enable_widevine_cdm_component =
-    enable_library_widevine_cdm && (is_win || is_mac)
+    enable_library_widevine_cdm && (is_win || is_mac || is_desktop_linux)
 
 # Widevine CDM is bundled as part of Google Chrome builds.
 bundle_widevine_cdm = enable_library_widevine_cdm && is_chrome_branded
